---
title: "Hi-C Basic Analysis"
author: "Ittai Eres"
date: "1/12/2017"
output: html_document
---

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:

```{r "setup", include=FALSE}
knitr::opts_knit$set(root.dir="~/Desktop/Hi-C/Bedtools_Analyze")
setwd("~/Desktop/Hi-C/Bedtools_Analyze")

##For plotly stuff
#Sys.setenv("plotly_username"="ittai")
#Sys.setenv("plotly_api_key"="FeF6FPAJDkpIb9pP4zq5")

## Useful
genome <- c("chr1", "chr2", "chr3", "chr4", "chr5", "chr6", "chr7", "chr8", "chr9", "chr10", "chr11", "chr12", "chr13", "chr14", "chr15", "chr16", "chr17", "chr18", "chr19", "chr20", "chr21", "chr22", "chrX", "chrY")

## Read in all DAT DATA!
coverage.full <- read.delim("coverage_5kb_1kb.bed", header=FALSE, stringsAsFactors = FALSE)
colnames(coverage.full) <- c("chr", "pos1", "pos2", "count")
```

```{r Bin Sum Optimization Function--preferred}
#Simple function that finds the sets of bins in each block of reads that will yield the greatest sum of reads, regardless of read inclusion.
optimize.bin.sums <- function(coverage_df){
  
   #define some necessary variables based on the coverage df; initiliaze some variables
  bin.size <- coverage_df$pos2[1]-coverage_df$pos1[1]
  window.slide <- coverage_df$pos1[2]-coverage_df$pos1[1]
  index.slide <- (bin.size/window.slide) #obtain the value for how many indices to slide forward for the sliding window!
  loc.index <- 1 #initialize an index for moving through the coverage_df
  coverage_df$bin <- 0 #intiailize another column in the coverage_df storing whether a given range ends up being a bin (1 for yes)
  
  while(loc.index<=nrow(coverage_df)){#iterate through all the positions in the coverage df
    
    if(coverage_df$count[loc.index]==0){ #move on to the next index if the given row has no reads in that bin--it's irrelevant.
      loc.index <- loc.index+1 #increment the index
      next}# move on to iterate this while loop again
    
    #If we see a switch from 0 to >0, the window has slid to include a new read in it. First, define the full range of the reads we can find here:
    end.index <- loc.index + match(0, coverage_df$count[loc.index:nrow(coverage_df)])-2 #pulls out just b/f the first instance of 0 in the count column--i.e. end.index is the last index at which a non-zero value is found
    tmp.indices <- seq(loc.index, end.index, index.slide) #grab first set of possible bins from this read region
    tmp.max <- sum(coverage_df$count[tmp.indices]) #store the sum of the read counts from this set as the temporary maximum
    tmp.max.indices <- tmp.indices #initialize the tmp.max.indices variable with these first indices, for proper bin assignation after the loop
    
    #Include something here to deal with the case when the initial tmp.indices is only a single element? I.e. where a region doesn't have options?
    #Check on the other regions where this is happening--the issue might not be that, but, rather, that there are ties!
    
    while(head(tmp.indices, 1)<=(loc.index+index.slide-1)){#keep going until we've considered all the possible bin combinations in this block of reads--check by stopping if the first value in tmp.indices is greater than index.slide above the loc.index
      tmp.indices <- tmp.indices + 1 #increment to look at the next set of bins
      tmp.indices <- tmp.indices[which(tmp.indices<=end.index)] #get rid of indices in tmp.indices that go past the end.index (i.e. move into other read blocks that are separated by a 0!)
      
      if(!length(tmp.indices)){break} #If tmp.indices turns into an empty vector (cases where there are not multiple options for bin combos), break this loop and move on (not gonna get higher tmp.max).
      
      if(sum(coverage_df$count[tmp.indices])>tmp.max){#If we have found a new way to maximize total read count in a bin set,
        tmp.max.indices <- tmp.indices #store these given indices in the tmp.max.indices variable, for bin assignation after the loop
        tmp.max <- sum(coverage_df$count[tmp.indices]) #make this the new tmp.max
      }
    }
    coverage_df$bin[tmp.max.indices] <- 1 #assign a status of being a chosen bin to the indices that gave us the max sum!
    loc.index <- tail(tmp.max.indices, 1)+index.slide #Move the loc index to the next possible bin after the chosen set (tmp.max.indices)
  }
  return(coverage_df[which(coverage_df$bin==1),-4:-5]) #return a df of all the bins and their counts. minus the "bin" column
}
```

```{r Contact Tabulations}
library(dplyr)
library(tidyr)
library("Hmisc")

#With all different types of bin creation methods in hand, I can now go about actually tabulating contact counts for individual loci!
basic_tabulation <- function(pos_df, distance1, distance2, bins.df){
  colnames(pos_df) <- c("pos1", "pos2") #ensure the column names of the pos_df work with the next lines
  
  pos1_new <- ifelse(pos_df$pos1<pos_df$pos2, pos_df$pos1, pos_df$pos2) #reorders columns so pos1 < pos2. important for not repeating bin pairs
  pos2_new <- ifelse(pos_df$pos1<pos_df$pos2, pos_df$pos2, pos_df$pos1) #If true, returns second arg; if false, returns third.
  pos_df$pos1 <- pos1_new #assignation of the new columns
  pos_df$pos2 <- pos2_new #assignation of the new columns
  
  pos_df$diff <- abs(pos_df$pos1-pos_df$pos2) #make a third column--the difference between the first two
  pos_df <- pos_df[pos_df$diff<(distance2)&pos_df$diff>=(distance1),1:2] #pull out pairs in the given distance range
  
  if(nrow(pos_df)>0){#If we've found pairs at the given distance, tabulate contact counts!
  bins.vec <- unique(sort(c(bins.df$pos1, bins.df$pos2))) #Pulls out the vector of bins for the given chromosome at given bin size.
  
  pos_df <- pos_df %>% mutate(pos1_bin = cut2(pos1, bins.vec, onlycuts=TRUE, oneval=FALSE, digits=9)) %>% mutate(pos2_bin = cut2(pos2, bins.vec, onlycuts=TRUE, oneval=FALSE, digits=9)) #make two new columns in the pos_df, with bin assignments for pos1 and pos2
  pos_df %>% group_by(pos1_bin, pos2_bin) %>% summarise(count=n()) #tabulate the bin contacts that are repeated to find total bin-bin contacts
  }
  else(return(0))#print("No contacts at that distance on this chromosome!"))
}
```

```{r Variable assignment for data creation}
#First, gotta get the coverage and optimize the bins for all chromosomes.
chr1.full <- coverage.full[which(coverage.full$chr=="chr1"),]
chr2.full <- coverage.full[which(coverage.full$chr=="chr2"),]
chr3.full <- coverage.full[which(coverage.full$chr=="chr3"),]
chr4.full <- coverage.full[which(coverage.full$chr=="chr4"),]
chr5.full <- coverage.full[which(coverage.full$chr=="chr5"),]
chr6.full <- coverage.full[which(coverage.full$chr=="chr6"),]
chr7.full <- coverage.full[which(coverage.full$chr=="chr7"),]
chr8.full <- coverage.full[which(coverage.full$chr=="chr8"),]
chr9.full <- coverage.full[which(coverage.full$chr=="chr9"),]
chr10.full <- coverage.full[which(coverage.full$chr=="chr10"),]
chr11.full <- coverage.full[which(coverage.full$chr=="chr11"),]
chr12.full <- coverage.full[which(coverage.full$chr=="chr12"),]
chr13.full <- coverage.full[which(coverage.full$chr=="chr13"),]
chr14.full <- coverage.full[which(coverage.full$chr=="chr14"),]
chr15.full <- coverage.full[which(coverage.full$chr=="chr15"),]
chr16.full <- coverage.full[which(coverage.full$chr=="chr16"),]
chr17.full <- coverage.full[which(coverage.full$chr=="chr17"),]
chr18.full <- coverage.full[which(coverage.full$chr=="chr18"),]
chr19.full <- coverage.full[which(coverage.full$chr=="chr19"),]
chr20.full <- coverage.full[which(coverage.full$chr=="chr20"),]
chr21.full <- coverage.full[which(coverage.full$chr=="chr21"),]
chr22.full <- coverage.full[which(coverage.full$chr=="chr22"),]
chrX.full <- coverage.full[which(coverage.full$chr=="chrX"),]
chrY.full <- coverage.full[which(coverage.full$chr=="chrY"),]

bins.1 <- optimize.bin.sums(chr1.full)
bins.2 <- optimize.bin.sums(chr2.full)
bins.3 <- optimize.bin.sums(chr3.full)
bins.4 <- optimize.bin.sums(chr4.full)
bins.5 <- optimize.bin.sums(chr5.full)
bins.6 <- optimize.bin.sums(chr6.full)
bins.7 <- optimize.bin.sums(chr7.full)
bins.8 <- optimize.bin.sums(chr8.full)
bins.9 <- optimize.bin.sums(chr9.full)
bins.10 <- optimize.bin.sums(chr10.full)
bins.11 <- optimize.bin.sums(chr11.full)
bins.12 <- optimize.bin.sums(chr12.full)
bins.13 <- optimize.bin.sums(chr13.full)
bins.14 <- optimize.bin.sums(chr14.full)
bins.15 <- optimize.bin.sums(chr15.full)
bins.16 <- optimize.bin.sums(chr16.full)
bins.17 <- optimize.bin.sums(chr17.full)
bins.18 <- optimize.bin.sums(chr18.full)
bins.19 <- optimize.bin.sums(chr19.full)
bins.20 <- optimize.bin.sums(chr20.full)
bins.21 <- optimize.bin.sums(chr21.full)
bins.22 <- optimize.bin.sums(chr22.full)
bins.X <- optimize.bin.sums(chrX.full)
bins.Y <- optimize.bin.sums(chrY.full)


#fread("just1", heder=FALSE, stringsAsFactors=FALSE) is faster for big files!

setwd("/Users/ittaieres/Desktop/Hi-C/Hi-C Basic Analyze")
pos_df.1 <- read.delim("just1", header=FALSE, stringsAsFactors = FALSE)
pos_df.2 <- read.delim("just2", header=FALSE, stringsAsFactors = FALSE)
pos_df.3 <- read.delim("just3", header=FALSE, stringsAsFactors = FALSE)
pos_df.4 <- read.delim("just4", header=FALSE, stringsAsFactors = FALSE)
pos_df.5 <- read.delim("just5", header=FALSE, stringsAsFactors = FALSE)
pos_df.6 <- read.delim("just6", header=FALSE, stringsAsFactors = FALSE)
pos_df.7 <- read.delim("just7", header=FALSE, stringsAsFactors = FALSE)
pos_df.8 <- read.delim("just8", header=FALSE, stringsAsFactors = FALSE)
pos_df.9 <- read.delim("just9", header=FALSE, stringsAsFactors = FALSE)
pos_df.10 <- read.delim("just10", header=FALSE, stringsAsFactors = FALSE)
pos_df.11 <- read.delim("just11", header=FALSE, stringsAsFactors = FALSE)
pos_df.12 <- read.delim("just12", header=FALSE, stringsAsFactors = FALSE)
pos_df.13 <- read.delim("just13", header=FALSE, stringsAsFactors = FALSE)
pos_df.14 <- read.delim("just14", header=FALSE, stringsAsFactors = FALSE)
pos_df.15 <- read.delim("just15", header=FALSE, stringsAsFactors = FALSE)
pos_df.16 <- read.delim("just16", header=FALSE, stringsAsFactors = FALSE)
pos_df.17 <- read.delim("just17", header=FALSE, stringsAsFactors = FALSE)
pos_df.18 <- read.delim("just18", header=FALSE, stringsAsFactors = FALSE)
pos_df.19 <- read.delim("just19", header=FALSE, stringsAsFactors = FALSE)
pos_df.20 <- read.delim("just20", header=FALSE, stringsAsFactors = FALSE)
pos_df.21 <- read.delim("just21", header=FALSE, stringsAsFactors = FALSE)
pos_df.22 <- read.delim("just22", header=FALSE, stringsAsFactors = FALSE)
pos_df.X <- read.delim("justX", header=FALSE, stringsAsFactors = FALSE)
pos_df.Y <- read.delim("justY", header=FALSE, stringsAsFactors = FALSE)

fread("just22", header=FALSE, stringsAsFactors = F)
```


```{r Plotting and examination of data Distribution of contact frequency for different distances}
#Now, actually use the basic tabulation function to populate matrices for each of the chromosomes with differing distances.
setwd("/Users/ittaieres/Desktop/Hi-C/Hi-C Basic Analyze")

indiv.chrom.tabulator <- function(pos_df, bins, start, end, by){
  distances <- seq(start, end, by) #Get the set of distances being looked at
  tabulation.list <- vector("list", length(distances))
  for(distance in 1:length(distances)){
    if(is.null(nrow(basic_tabulation(pos_df, distances[distance], distances[distance]+by, bins)))){next}
    tabulation.list[[distance]] <- basic_tabulation(pos_df, distances[distance], distances[distance]+by, bins)$count
    }
  names(tabulation.list) <- distances
  return(tabulation.list)
}

full.chrom.tabulator <- function(pos_df, bins, start, end, by, chr){
  distances <- seq(start, end, by) #Get the set of distances being looked at
  tabulation.list <- vector("list", length(distances))
  for(distance in 1:length(distances)){
    if(is.null(nrow(basic_tabulation(pos_df, distances[distance], distances[distance]+by, bins)))){next}
    tmp.counts <- basic_tabulation(pos_df, distances[distance], distances[distance]+by, bins)$count
    tmp.mean <- mean(tmp.counts)
    tabulation.list[[distance]] <- data.frame(count=tmp.counts, bin=distances[distance], chr=chr, mean=tmp.mean)
  }
  tabulation.df <- do.call(rbind, tabulation.list)
  return(tabulation.df)
}

tab_df.1 <- full.chrom.tabulator(pos_df.1, bins.1, 5000, 100000, 5000, "chr1")
tab_df.2 <- full.chrom.tabulator(pos_df.2, bins.2, 5000, 100000, 5000, "chr2")
tab_df.3 <- full.chrom.tabulator(pos_df.3, bins.3, 5000, 100000, 5000, "chr3")
tab_df.4 <- full.chrom.tabulator(pos_df.4, bins.4, 5000, 100000, 5000, "chr4")
tab_df.5 <- full.chrom.tabulator(pos_df.5, bins.5, 5000, 100000, 5000, "chr5")
tab_df.6 <- full.chrom.tabulator(pos_df.6, bins.6, 5000, 100000, 5000, "chr6")
tab_df.7 <- full.chrom.tabulator(pos_df.7, bins.7, 5000, 100000, 5000, "chr7")
tab_df.8 <- full.chrom.tabulator(pos_df.8, bins.8, 5000, 100000, 5000, "chr8")
tab_df.9 <- full.chrom.tabulator(pos_df.9, bins.9, 5000, 100000, 5000, "chr9")
tab_df.10 <- full.chrom.tabulator(pos_df.10, bins.10, 5000, 100000, 5000, "chr10")
tab_df.11 <- full.chrom.tabulator(pos_df.11, bins.11, 5000, 100000, 5000, "chr11")
tab_df.12 <- full.chrom.tabulator(pos_df.12, bins.12, 5000, 100000, 5000, "chr12")
tab_df.13 <- full.chrom.tabulator(pos_df.13, bins.13, 5000, 100000, 5000, "chr13")
tab_df.14 <- full.chrom.tabulator(pos_df.14, bins.14, 5000, 100000, 5000, "chr14")
tab_df.15 <- full.chrom.tabulator(pos_df.15, bins.15, 5000, 100000, 5000, "chr15")
tab_df.16 <- full.chrom.tabulator(pos_df.16, bins.16, 5000, 100000, 5000, "chr16")
tab_df.17 <- full.chrom.tabulator(pos_df.17, bins.17, 5000, 100000, 5000, "chr17")
tab_df.18 <- full.chrom.tabulator(pos_df.18, bins.18, 5000, 100000, 5000, "chr18")
tab_df.19 <- full.chrom.tabulator(pos_df.19, bins.19, 5000, 100000, 5000, "chr19")
tab_df.20 <- full.chrom.tabulator(pos_df.20, bins.20, 5000, 100000, 5000, "chr20")
tab_df.21 <- full.chrom.tabulator(pos_df.21, bins.21, 5000, 100000, 5000, "chr21")
tab_df.22 <- full.chrom.tabulator(pos_df.22, bins.22, 5000, 100000, 5000, "chr22")
tab_df.X <- full.chrom.tabulator(pos_df.X, bins.X, 5000, 100000, 5000, "chrX")
tab_df.Y <- full.chrom.tabulator(pos_df.Y, bins.Y, 5000, 100000, 5000, "chrY")

library(plyr)
library(dplyr)
tab_df.full <- rbind.fill(tab_df.1, tab_df.2, tab_df.3, tab_df.4, tab_df.5, tab_df.6, tab_df.7, tab_df.8, tab_df.9, tab_df.10, tab_df.11, tab_df.12, tab_df.13, tab_df.14, tab_df.15, tab_df.16, tab_df.17, tab_df.18, tab_df.19, tab_df.20, tab_df.21, tab_df.22, tab_df.X, tab_df.Y) #Merge all these data frames together into one big one for plotting.

#Produce boxplots of the distribution of means across the chromosomes at a given distance bin.
p <- plot_ly(tab_df.full) %>%
  add_trace(y=mean, type='box')

g <- ggplot(tab_df.full, aes(x=factor(bin), y=mean)) +
  geom_boxplot() +
  geom_jitter(aes(x=factor(bin), y=mean, text=chr))
h <- ggplotly(g)


i=
tab_df.full%>%filter(bin=="5000")%>%filter(chr=="chr10" | chr =="chr20")%>%
ggplot(., aes(x=count,fill=chr)) +
  geom_density()

+
  geom_jitter(aes(x=factor(bin), y=mean, text=chr))
h <- ggplotly(g)



this <- tab_df.full %>% filter(count, bin==5000)
hist(this$count)


tab_df.full%>%filter(bin=="5000")%>%filter(chr=="chr10" | chr =="chr20")%>%dim

write.table(tab_df.full, file='IEE_data', sep="\t", eol="\n", col.names=T, row.names=F, quote=F)

#Create an empty data frame for putting the means information into, for boxplotting afterwards.
means.df <- data.frame(matrix(ncol=40, nrow=24))
colnames(means.df) <- names(list.1)
rownames(means.df) <- lists

lists <- c("list.1", "list.2", "list.3", "list.4", "list.5", "list.6", "list.7", "list.8", "list.9", "list.10", "list.11", "list.12", "list.13", "list.14", "list.15", "list.16", "list.17", "list.18", "list.19", "list.20", "list.21", "list.22", "list.X", "list.Y")

for(chrom in lists){ #Iterate through all the lists and obtain the means for each of the different distance bins desired.
  means.df[chrom,] <- c(mean(get(chrom)[[1]]), mean(get(chrom)[[2]]), mean(get(chrom)[[3]]), mean(get(chrom)[[4]]), mean(get(chrom)[[5]]), mean(get(chrom)[[6]]), mean(get(chrom)[[7]]), mean(get(chrom)[[8]]), mean(get(chrom)[[9]]), mean(get(chrom)[[10]]), mean(get(chrom)[[11]]), mean(get(chrom)[[12]]), mean(get(chrom)[[13]]), mean(get(chrom)[[14]]), mean(get(chrom)[[15]]), mean(get(chrom)[[16]]), mean(get(chrom)[[17]]), mean(get(chrom)[[18]]), mean(get(chrom)[[19]]), mean(get(chrom)[[20]]), mean(get(chrom)[[21]]), mean(get(chrom)[[22]]), mean(get(chrom)[[23]]), mean(get(chrom)[[24]]), mean(get(chrom)[[25]]), mean(get(chrom)[[26]]), mean(get(chrom)[[27]]), mean(get(chrom)[[28]]), mean(get(chrom)[[29]]), mean(get(chrom)[[30]]), mean(get(chrom)[[31]]), mean(get(chrom)[[32]]), mean(get(chrom)[[33]]), mean(get(chrom)[[34]]), mean(get(chrom)[[35]]), mean(get(chrom)[[36]]), mean(get(chrom)[[37]]), mean(get(chrom)[[38]]), mean(get(chrom)[[39]]), mean(get(chrom)[[40]]))
}

rownames(means.df) <- genome

means.gg.df <- data.frame(matrix(ncol=3, nrow=960))

library(reshape2) #Load the reshape2 package for melting the data into the appropriate DF format
means.gg.df <- melt(means.df) #Melt! Doesn't add the chromosome column though, can do that manually
means.gg.df$chr <- rep(genome, 40)

suggested.bg <- lists[c(-1, -2, -23, -24)]


test <- ggplot(means.gg.df[1:480,], aes(x=variable, y=value)) + geom_boxplot()
test2 <- ggplotly(test, tooltip=means.gg.df$chr)

test3 <- plot_ly(means.gg.df, x=means.gg.df$variable, y=means.gg.df$value, text=means.gg.df$chr, hoverinfo="text", type=box)


for(list in lists){print(get(list)[[1]])}


mega.list <- list(list.3, list.4, list.5, list.6, list.7, list.8, list.9, list.10, list.11, list.12, list.13, list.14, list.15, list.16, list.17, list.18, list.19, list.20, list.21, list.22)
keys <- unique(unlist(lapply(mega.list, names)))
mega.list <- setNames(do.call(mapply, c(FUN=c, lapply(mega.list, "[", keys))), keys)

library(plyr)
test <- data.frame(matrix(ncol=2, nrow=sum(lengths(mega.list))))
colnames(test) <- c("bin", "ct")

#Make a data frame that will work with ggplot2 for plotting this ish out!!! Works, has been validated w/ sums of counts
for(element in 1:length(mega.list)){
  if(element==1){
  test[1:length(mega.list[[element]]),1] <- names(mega.list)[element]
  test[1:length(mega.list[[element]]),2] <- mega.list[[element]]
  start <- length(mega.list[[element]])+1
  }
  else{
    end <- start+length(mega.list[[element]])-1
    print(c(element, start, end))
    test[(start:end),1] <- names(mega.list)[element]
    test[(start:end),2] <- as.vector(mega.list[[element]])
    start <- end+1
  }
}

sum(mega.list[[3]])
```

```{r Plotting}
library(ggplot2)
?geom_boxplot()

g <- ggplot(test, aes(bin, ct)) + geom_boxplot()

library(dplyr)
test %>%
  filter(.,bin<=50000) %>%
    ggplot(., aes(bin, ct))+
      geom_boxplot() #Gets some good stuff but it's still not in order
```

```{r Old garbage}
setwd("~/Desktop/Hi-C/Hi-C Basic Analyze")
#Grab all the pos_dfs. Cut unix file of 2 paired end read columns, read pos1 and read pos2.
pos_df.1 <- fread("just1", header=FALSE, stringsAsFactors = FALSE)
pos_df.2 <- fread("just2", header=FALSE, stringsAsFactors = FALSE)
pos_df.3 <- fread("just3", header=FALSE, stringsAsFactors = FALSE)
pos_df.4 <- fread("just4", header=FALSE, stringsAsFactors = FALSE)
pos_df.5 <- fread("just5", header=FALSE, stringsAsFactors = FALSE)
pos_df.6 <- fread("just6", header=FALSE, stringsAsFactors = FALSE)
pos_df.7 <- fread("just7", header=FALSE, stringsAsFactors = FALSE)
pos_df.8 <- fread("just8", header=FALSE, stringsAsFactors = FALSE)
pos_df.9 <- fread("just9", header=FALSE, stringsAsFactors = FALSE)
pos_df.10 <- fread("just10", header=FALSE, stringsAsFactors = FALSE)
pos_df.11 <- fread("just11", header=FALSE, stringsAsFactors = FALSE)
pos_df.12 <- fread("just12", header=FALSE, stringsAsFactors = FALSE)
pos_df.13 <- fread("just13", header=FALSE, stringsAsFactors = FALSE)
pos_df.14 <- fread("just14", header=FALSE, stringsAsFactors = FALSE)
pos_df.15 <- fread("just15", header=FALSE, stringsAsFactors = FALSE)
pos_df.16 <- fread("just16", header=FALSE, stringsAsFactors = FALSE)
pos_df.17 <- fread("just17", header=FALSE, stringsAsFactors = FALSE)
pos_df.18 <- fread("just18", header=FALSE, stringsAsFactors = FALSE)
pos_df.19 <- fread("just19", header=FALSE, stringsAsFactors = FALSE)
pos_df.20 <- fread("just20", header=FALSE, stringsAsFactors = FALSE)
pos_df.21 <- fread("just21", header=FALSE, stringsAsFactors = FALSE)
pos_df.22 <- fread("just22", header=FALSE, stringsAsFactors = FALSE)
pos_df.X <- fread("justX", header=FALSE, stringsAsFactors = FALSE)
pos_df.Y <- fread("justY", header=FALSE, stringsAsFactors = FALSE)

#Get the bedtools-created coverage data frames in order to optimize bin sums.
setwd("~/Desktop/Hi-C/Hi-C Basic Analyze/coverage/current")
coverage.10kb <- read.delim("coverage_10kb_5kb.bed", header=FALSE, stringsAsFactors = FALSE)
coverage.20kb <- read.delim("coverage_20kb_5kb.bed", header=FALSE, stringsAsFactors = FALSE)
coverage.30kb <- read.delim("coverage_30kb_5kb.bed", header=FALSE, stringsAsFactors = FALSE)
coverage.40kb <- read.delim("coverage_40kb_5kb.bed", header=FALSE, stringsAsFactors = FALSE)
coverage.50kb <- read.delim("coverage_50kb_5kb.bed", header=FALSE, stringsAsFactors = FALSE)
coverage.60kb <- read.delim("coverage_60kb_5kb.bed", header=FALSE, stringsAsFactors = FALSE)
coverage.70kb <- read.delim("coverage_70kb_5kb.bed", header=FALSE, stringsAsFactors = FALSE)
coverage.80kb <- read.delim("coverage_80kb_5kb.bed", header=FALSE, stringsAsFactors = FALSE)
coverage.90kb <- read.delim("coverage_90kb_5kb.bed", header=FALSE, stringsAsFactors = FALSE)
coverage.100kb <- read.delim("coverage_100kb_5kb.bed", header=FALSE, stringsAsFactors = FALSE)
coverage.125kb <- read.delim("coverage_125kb_5kb.bed", header=FALSE, stringsAsFactors = FALSE)
coverage.150kb <- read.delim("coverage_150kb_5kb.bed", header=FALSE, stringsAsFactors = FALSE)
coverage.200kb <- read.delim("coverage_200kb_5kb.bed", header=FALSE, stringsAsFactors = FALSE)

#Assign their column names.
colnames(coverage.10kb) <- c("chr", "pos1", "pos2", "count")
colnames(coverage.20kb) <- c("chr", "pos1", "pos2", "count")
colnames(coverage.30kb) <- c("chr", "pos1", "pos2", "count")
colnames(coverage.40kb) <- c("chr", "pos1", "pos2", "count")
colnames(coverage.50kb) <- c("chr", "pos1", "pos2", "count")
colnames(coverage.60kb) <- c("chr", "pos1", "pos2", "count")
colnames(coverage.70kb) <- c("chr", "pos1", "pos2", "count")
colnames(coverage.80kb) <- c("chr", "pos1", "pos2", "count")
colnames(coverage.90kb) <- c("chr", "pos1", "pos2", "count")
colnames(coverage.100kb) <- c("chr", "pos1", "pos2", "count")
colnames(coverage.125kb) <- c("chr", "pos1", "pos2", "count")
colnames(coverage.150kb) <- c("chr", "pos1", "pos2", "count")
colnames(coverage.200kb) <- c("chr", "pos1", "pos2", "count")

#Get the coverage for .10kb for all the chromosomes
chr1.10kb <- coverage.10kb[which(coverage.10kb$chr=="chr1"),]
chr2.10kb <- coverage.10kb[which(coverage.10kb$chr=="chr2"),]
chr3.10kb <- coverage.10kb[which(coverage.10kb$chr=="chr3"),]
chr4.10kb <- coverage.10kb[which(coverage.10kb$chr=="chr4"),]
chr5.10kb <- coverage.10kb[which(coverage.10kb$chr=="chr5"),]
chr6.10kb <- coverage.10kb[which(coverage.10kb$chr=="chr6"),]
chr7.10kb <- coverage.10kb[which(coverage.10kb$chr=="chr7"),]
chr8.10kb <- coverage.10kb[which(coverage.10kb$chr=="chr8"),]
chr9.10kb <- coverage.10kb[which(coverage.10kb$chr=="chr9"),]
chr10.10kb <- coverage.10kb[which(coverage.10kb$chr=="chr10"),]
chr11.10kb <- coverage.10kb[which(coverage.10kb$chr=="chr11"),]
chr12.10kb <- coverage.10kb[which(coverage.10kb$chr=="chr12"),]
chr13.10kb <- coverage.10kb[which(coverage.10kb$chr=="chr13"),]
chr14.10kb <- coverage.10kb[which(coverage.10kb$chr=="chr14"),]
chr15.10kb <- coverage.10kb[which(coverage.10kb$chr=="chr15"),]
chr16.10kb <- coverage.10kb[which(coverage.10kb$chr=="chr16"),]
chr17.10kb <- coverage.10kb[which(coverage.10kb$chr=="chr17"),]
chr18.10kb <- coverage.10kb[which(coverage.10kb$chr=="chr18"),]
chr19.10kb <- coverage.10kb[which(coverage.10kb$chr=="chr19"),]
chr20.10kb <- coverage.10kb[which(coverage.10kb$chr=="chr20"),]
chr21.10kb <- coverage.10kb[which(coverage.10kb$chr=="chr21"),]
chr22.10kb <- coverage.10kb[which(coverage.10kb$chr=="chr22"),]
chrX.10kb <- coverage.10kb[which(coverage.10kb$chr=="chrX"),]
chrY.10kb <- coverage.10kb[which(coverage.10kb$chr=="chrY"),]

#Get the coverage for .20kb for all the chromosomes
chr1.20kb <- coverage.20kb[which(coverage.20kb$chr=="chr1"),]
chr2.20kb <- coverage.20kb[which(coverage.20kb$chr=="chr2"),]
chr3.20kb <- coverage.20kb[which(coverage.20kb$chr=="chr3"),]
chr4.20kb <- coverage.20kb[which(coverage.20kb$chr=="chr4"),]
chr5.20kb <- coverage.20kb[which(coverage.20kb$chr=="chr5"),]
chr6.20kb <- coverage.20kb[which(coverage.20kb$chr=="chr6"),]
chr7.20kb <- coverage.20kb[which(coverage.20kb$chr=="chr7"),]
chr8.20kb <- coverage.20kb[which(coverage.20kb$chr=="chr8"),]
chr9.20kb <- coverage.20kb[which(coverage.20kb$chr=="chr9"),]
chr10.20kb <- coverage.20kb[which(coverage.20kb$chr=="chr10"),]
chr11.20kb <- coverage.20kb[which(coverage.20kb$chr=="chr11"),]
chr12.20kb <- coverage.20kb[which(coverage.20kb$chr=="chr12"),]
chr13.20kb <- coverage.20kb[which(coverage.20kb$chr=="chr13"),]
chr14.20kb <- coverage.20kb[which(coverage.20kb$chr=="chr14"),]
chr15.20kb <- coverage.20kb[which(coverage.20kb$chr=="chr15"),]
chr16.20kb <- coverage.20kb[which(coverage.20kb$chr=="chr16"),]
chr17.20kb <- coverage.20kb[which(coverage.20kb$chr=="chr17"),]
chr18.20kb <- coverage.20kb[which(coverage.20kb$chr=="chr18"),]
chr19.20kb <- coverage.20kb[which(coverage.20kb$chr=="chr19"),]
chr20.20kb <- coverage.20kb[which(coverage.20kb$chr=="chr20"),]
chr21.20kb <- coverage.20kb[which(coverage.20kb$chr=="chr21"),]
chr22.20kb <- coverage.20kb[which(coverage.20kb$chr=="chr22"),]
chrX.20kb <- coverage.20kb[which(coverage.20kb$chr=="chrX"),]
chrY.20kb <- coverage.20kb[which(coverage.20kb$chr=="chrY"),]

#Get the coverage for .30kb for all the chromosomes
chr1.30kb <- coverage.30kb[which(coverage.30kb$chr=="chr1"),]
chr2.30kb <- coverage.30kb[which(coverage.30kb$chr=="chr2"),]
chr3.30kb <- coverage.30kb[which(coverage.30kb$chr=="chr3"),]
chr4.30kb <- coverage.30kb[which(coverage.30kb$chr=="chr4"),]
chr5.30kb <- coverage.30kb[which(coverage.30kb$chr=="chr5"),]
chr6.30kb <- coverage.30kb[which(coverage.30kb$chr=="chr6"),]
chr7.30kb <- coverage.30kb[which(coverage.30kb$chr=="chr7"),]
chr8.30kb <- coverage.30kb[which(coverage.30kb$chr=="chr8"),]
chr9.30kb <- coverage.30kb[which(coverage.30kb$chr=="chr9"),]
chr10.30kb <- coverage.30kb[which(coverage.30kb$chr=="chr10"),]
chr11.30kb <- coverage.30kb[which(coverage.30kb$chr=="chr11"),]
chr12.30kb <- coverage.30kb[which(coverage.30kb$chr=="chr12"),]
chr13.30kb <- coverage.30kb[which(coverage.30kb$chr=="chr13"),]
chr14.30kb <- coverage.30kb[which(coverage.30kb$chr=="chr14"),]
chr15.30kb <- coverage.30kb[which(coverage.30kb$chr=="chr15"),]
chr16.30kb <- coverage.30kb[which(coverage.30kb$chr=="chr16"),]
chr17.30kb <- coverage.30kb[which(coverage.30kb$chr=="chr17"),]
chr18.30kb <- coverage.30kb[which(coverage.30kb$chr=="chr18"),]
chr19.30kb <- coverage.30kb[which(coverage.30kb$chr=="chr19"),]
chr20.30kb <- coverage.30kb[which(coverage.30kb$chr=="chr20"),]
chr21.30kb <- coverage.30kb[which(coverage.30kb$chr=="chr21"),]
chr22.30kb <- coverage.30kb[which(coverage.30kb$chr=="chr22"),]
chrX.30kb <- coverage.30kb[which(coverage.30kb$chr=="chrX"),]
chrY.30kb <- coverage.30kb[which(coverage.30kb$chr=="chrY"),]

#Get the coverage for .40kb for all the chromosomes
chr1.40kb <- coverage.40kb[which(coverage.40kb$chr=="chr1"),]
chr2.40kb <- coverage.40kb[which(coverage.40kb$chr=="chr2"),]
chr3.40kb <- coverage.40kb[which(coverage.40kb$chr=="chr3"),]
chr4.40kb <- coverage.40kb[which(coverage.40kb$chr=="chr4"),]
chr5.40kb <- coverage.40kb[which(coverage.40kb$chr=="chr5"),]
chr6.40kb <- coverage.40kb[which(coverage.40kb$chr=="chr6"),]
chr7.40kb <- coverage.40kb[which(coverage.40kb$chr=="chr7"),]
chr8.40kb <- coverage.40kb[which(coverage.40kb$chr=="chr8"),]
chr9.40kb <- coverage.40kb[which(coverage.40kb$chr=="chr9"),]
chr10.40kb <- coverage.40kb[which(coverage.40kb$chr=="chr10"),]
chr11.40kb <- coverage.40kb[which(coverage.40kb$chr=="chr11"),]
chr12.40kb <- coverage.40kb[which(coverage.40kb$chr=="chr12"),]
chr13.40kb <- coverage.40kb[which(coverage.40kb$chr=="chr13"),]
chr14.40kb <- coverage.40kb[which(coverage.40kb$chr=="chr14"),]
chr15.40kb <- coverage.40kb[which(coverage.40kb$chr=="chr15"),]
chr16.40kb <- coverage.40kb[which(coverage.40kb$chr=="chr16"),]
chr17.40kb <- coverage.40kb[which(coverage.40kb$chr=="chr17"),]
chr18.40kb <- coverage.40kb[which(coverage.40kb$chr=="chr18"),]
chr19.40kb <- coverage.40kb[which(coverage.40kb$chr=="chr19"),]
chr20.40kb <- coverage.40kb[which(coverage.40kb$chr=="chr20"),]
chr21.40kb <- coverage.40kb[which(coverage.40kb$chr=="chr21"),]
chr22.40kb <- coverage.40kb[which(coverage.40kb$chr=="chr22"),]
chrX.40kb <- coverage.40kb[which(coverage.40kb$chr=="chrX"),]
chrY.40kb <- coverage.40kb[which(coverage.40kb$chr=="chrY"),]

#Get the coverage for .50kb for all the chromosomes
chr1.50kb <- coverage.50kb[which(coverage.50kb$chr=="chr1"),]
chr2.50kb <- coverage.50kb[which(coverage.50kb$chr=="chr2"),]
chr3.50kb <- coverage.50kb[which(coverage.50kb$chr=="chr3"),]
chr4.50kb <- coverage.50kb[which(coverage.50kb$chr=="chr4"),]
chr5.50kb <- coverage.50kb[which(coverage.50kb$chr=="chr5"),]
chr6.50kb <- coverage.50kb[which(coverage.50kb$chr=="chr6"),]
chr7.50kb <- coverage.50kb[which(coverage.50kb$chr=="chr7"),]
chr8.50kb <- coverage.50kb[which(coverage.50kb$chr=="chr8"),]
chr9.50kb <- coverage.50kb[which(coverage.50kb$chr=="chr9"),]
chr10.50kb <- coverage.50kb[which(coverage.50kb$chr=="chr10"),]
chr11.50kb <- coverage.50kb[which(coverage.50kb$chr=="chr11"),]
chr12.50kb <- coverage.50kb[which(coverage.50kb$chr=="chr12"),]
chr13.50kb <- coverage.50kb[which(coverage.50kb$chr=="chr13"),]
chr14.50kb <- coverage.50kb[which(coverage.50kb$chr=="chr14"),]
chr15.50kb <- coverage.50kb[which(coverage.50kb$chr=="chr15"),]
chr16.50kb <- coverage.50kb[which(coverage.50kb$chr=="chr16"),]
chr17.50kb <- coverage.50kb[which(coverage.50kb$chr=="chr17"),]
chr18.50kb <- coverage.50kb[which(coverage.50kb$chr=="chr18"),]
chr19.50kb <- coverage.50kb[which(coverage.50kb$chr=="chr19"),]
chr20.50kb <- coverage.50kb[which(coverage.50kb$chr=="chr20"),]
chr21.50kb <- coverage.50kb[which(coverage.50kb$chr=="chr21"),]
chr22.50kb <- coverage.50kb[which(coverage.50kb$chr=="chr22"),]
chrX.50kb <- coverage.50kb[which(coverage.50kb$chr=="chrX"),]
chrY.50kb <- coverage.50kb[which(coverage.50kb$chr=="chrY"),]

#Get the coverage for .60kb for all the chromosomes
chr1.60kb <- coverage.60kb[which(coverage.60kb$chr=="chr1"),]
chr2.60kb <- coverage.60kb[which(coverage.60kb$chr=="chr2"),]
chr3.60kb <- coverage.60kb[which(coverage.60kb$chr=="chr3"),]
chr4.60kb <- coverage.60kb[which(coverage.60kb$chr=="chr4"),]
chr5.60kb <- coverage.60kb[which(coverage.60kb$chr=="chr5"),]
chr6.60kb <- coverage.60kb[which(coverage.60kb$chr=="chr6"),]
chr7.60kb <- coverage.60kb[which(coverage.60kb$chr=="chr7"),]
chr8.60kb <- coverage.60kb[which(coverage.60kb$chr=="chr8"),]
chr9.60kb <- coverage.60kb[which(coverage.60kb$chr=="chr9"),]
chr10.60kb <- coverage.60kb[which(coverage.60kb$chr=="chr10"),]
chr11.60kb <- coverage.60kb[which(coverage.60kb$chr=="chr11"),]
chr12.60kb <- coverage.60kb[which(coverage.60kb$chr=="chr12"),]
chr13.60kb <- coverage.60kb[which(coverage.60kb$chr=="chr13"),]
chr14.60kb <- coverage.60kb[which(coverage.60kb$chr=="chr14"),]
chr15.60kb <- coverage.60kb[which(coverage.60kb$chr=="chr15"),]
chr16.60kb <- coverage.60kb[which(coverage.60kb$chr=="chr16"),]
chr17.60kb <- coverage.60kb[which(coverage.60kb$chr=="chr17"),]
chr18.60kb <- coverage.60kb[which(coverage.60kb$chr=="chr18"),]
chr19.60kb <- coverage.60kb[which(coverage.60kb$chr=="chr19"),]
chr20.60kb <- coverage.60kb[which(coverage.60kb$chr=="chr20"),]
chr21.60kb <- coverage.60kb[which(coverage.60kb$chr=="chr21"),]
chr22.60kb <- coverage.60kb[which(coverage.60kb$chr=="chr22"),]
chrX.60kb <- coverage.60kb[which(coverage.60kb$chr=="chrX"),]
chrY.60kb <- coverage.60kb[which(coverage.60kb$chr=="chrY"),]

#Get the coverage for .70kb for all the chromosomes
chr1.70kb <- coverage.70kb[which(coverage.70kb$chr=="chr1"),]
chr2.70kb <- coverage.70kb[which(coverage.70kb$chr=="chr2"),]
chr3.70kb <- coverage.70kb[which(coverage.70kb$chr=="chr3"),]
chr4.70kb <- coverage.70kb[which(coverage.70kb$chr=="chr4"),]
chr5.70kb <- coverage.70kb[which(coverage.70kb$chr=="chr5"),]
chr6.70kb <- coverage.70kb[which(coverage.70kb$chr=="chr6"),]
chr7.70kb <- coverage.70kb[which(coverage.70kb$chr=="chr7"),]
chr8.70kb <- coverage.70kb[which(coverage.70kb$chr=="chr8"),]
chr9.70kb <- coverage.70kb[which(coverage.70kb$chr=="chr9"),]
chr10.70kb <- coverage.70kb[which(coverage.70kb$chr=="chr10"),]
chr11.70kb <- coverage.70kb[which(coverage.70kb$chr=="chr11"),]
chr12.70kb <- coverage.70kb[which(coverage.70kb$chr=="chr12"),]
chr13.70kb <- coverage.70kb[which(coverage.70kb$chr=="chr13"),]
chr14.70kb <- coverage.70kb[which(coverage.70kb$chr=="chr14"),]
chr15.70kb <- coverage.70kb[which(coverage.70kb$chr=="chr15"),]
chr16.70kb <- coverage.70kb[which(coverage.70kb$chr=="chr16"),]
chr17.70kb <- coverage.70kb[which(coverage.70kb$chr=="chr17"),]
chr18.70kb <- coverage.70kb[which(coverage.70kb$chr=="chr18"),]
chr19.70kb <- coverage.70kb[which(coverage.70kb$chr=="chr19"),]
chr20.70kb <- coverage.70kb[which(coverage.70kb$chr=="chr20"),]
chr21.70kb <- coverage.70kb[which(coverage.70kb$chr=="chr21"),]
chr22.70kb <- coverage.70kb[which(coverage.70kb$chr=="chr22"),]
chrX.70kb <- coverage.70kb[which(coverage.70kb$chr=="chrX"),]
chrY.70kb <- coverage.70kb[which(coverage.70kb$chr=="chrY"),]

#Get the coverage for .80kb for all the chromosomes
chr1.80kb <- coverage.80kb[which(coverage.80kb$chr=="chr1"),]
chr2.80kb <- coverage.80kb[which(coverage.80kb$chr=="chr2"),]
chr3.80kb <- coverage.80kb[which(coverage.80kb$chr=="chr3"),]
chr4.80kb <- coverage.80kb[which(coverage.80kb$chr=="chr4"),]
chr5.80kb <- coverage.80kb[which(coverage.80kb$chr=="chr5"),]
chr6.80kb <- coverage.80kb[which(coverage.80kb$chr=="chr6"),]
chr7.80kb <- coverage.80kb[which(coverage.80kb$chr=="chr7"),]
chr8.80kb <- coverage.80kb[which(coverage.80kb$chr=="chr8"),]
chr9.80kb <- coverage.80kb[which(coverage.80kb$chr=="chr9"),]
chr10.80kb <- coverage.80kb[which(coverage.80kb$chr=="chr10"),]
chr11.80kb <- coverage.80kb[which(coverage.80kb$chr=="chr11"),]
chr12.80kb <- coverage.80kb[which(coverage.80kb$chr=="chr12"),]
chr13.80kb <- coverage.80kb[which(coverage.80kb$chr=="chr13"),]
chr14.80kb <- coverage.80kb[which(coverage.80kb$chr=="chr14"),]
chr15.80kb <- coverage.80kb[which(coverage.80kb$chr=="chr15"),]
chr16.80kb <- coverage.80kb[which(coverage.80kb$chr=="chr16"),]
chr17.80kb <- coverage.80kb[which(coverage.80kb$chr=="chr17"),]
chr18.80kb <- coverage.80kb[which(coverage.80kb$chr=="chr18"),]
chr19.80kb <- coverage.80kb[which(coverage.80kb$chr=="chr19"),]
chr20.80kb <- coverage.80kb[which(coverage.80kb$chr=="chr20"),]
chr21.80kb <- coverage.80kb[which(coverage.80kb$chr=="chr21"),]
chr22.80kb <- coverage.80kb[which(coverage.80kb$chr=="chr22"),]
chrX.80kb <- coverage.80kb[which(coverage.80kb$chr=="chrX"),]
chrY.80kb <- coverage.80kb[which(coverage.80kb$chr=="chrY"),]

#Get the coverage for .90kb for all the chromosomes
chr1.90kb <- coverage.90kb[which(coverage.90kb$chr=="chr1"),]
chr2.90kb <- coverage.90kb[which(coverage.90kb$chr=="chr2"),]
chr3.90kb <- coverage.90kb[which(coverage.90kb$chr=="chr3"),]
chr4.90kb <- coverage.90kb[which(coverage.90kb$chr=="chr4"),]
chr5.90kb <- coverage.90kb[which(coverage.90kb$chr=="chr5"),]
chr6.90kb <- coverage.90kb[which(coverage.90kb$chr=="chr6"),]
chr7.90kb <- coverage.90kb[which(coverage.90kb$chr=="chr7"),]
chr8.90kb <- coverage.90kb[which(coverage.90kb$chr=="chr8"),]
chr9.90kb <- coverage.90kb[which(coverage.90kb$chr=="chr9"),]
chr10.90kb <- coverage.90kb[which(coverage.90kb$chr=="chr10"),]
chr11.90kb <- coverage.90kb[which(coverage.90kb$chr=="chr11"),]
chr12.90kb <- coverage.90kb[which(coverage.90kb$chr=="chr12"),]
chr13.90kb <- coverage.90kb[which(coverage.90kb$chr=="chr13"),]
chr14.90kb <- coverage.90kb[which(coverage.90kb$chr=="chr14"),]
chr15.90kb <- coverage.90kb[which(coverage.90kb$chr=="chr15"),]
chr16.90kb <- coverage.90kb[which(coverage.90kb$chr=="chr16"),]
chr17.90kb <- coverage.90kb[which(coverage.90kb$chr=="chr17"),]
chr18.90kb <- coverage.90kb[which(coverage.90kb$chr=="chr18"),]
chr19.90kb <- coverage.90kb[which(coverage.90kb$chr=="chr19"),]
chr20.90kb <- coverage.90kb[which(coverage.90kb$chr=="chr20"),]
chr21.90kb <- coverage.90kb[which(coverage.90kb$chr=="chr21"),]
chr22.90kb <- coverage.90kb[which(coverage.90kb$chr=="chr22"),]
chrX.90kb <- coverage.90kb[which(coverage.90kb$chr=="chrX"),]
chrY.90kb <- coverage.90kb[which(coverage.90kb$chr=="chrY"),]

#Get the coverage for .100kb for all the chromosomes
chr1.100kb <- coverage.100kb[which(coverage.100kb$chr=="chr1"),]
chr2.100kb <- coverage.100kb[which(coverage.100kb$chr=="chr2"),]
chr3.100kb <- coverage.100kb[which(coverage.100kb$chr=="chr3"),]
chr4.100kb <- coverage.100kb[which(coverage.100kb$chr=="chr4"),]
chr5.100kb <- coverage.100kb[which(coverage.100kb$chr=="chr5"),]
chr6.100kb <- coverage.100kb[which(coverage.100kb$chr=="chr6"),]
chr7.100kb <- coverage.100kb[which(coverage.100kb$chr=="chr7"),]
chr8.100kb <- coverage.100kb[which(coverage.100kb$chr=="chr8"),]
chr9.100kb <- coverage.100kb[which(coverage.100kb$chr=="chr9"),]
chr10.100kb <- coverage.100kb[which(coverage.100kb$chr=="chr10"),]
chr11.100kb <- coverage.100kb[which(coverage.100kb$chr=="chr11"),]
chr12.100kb <- coverage.100kb[which(coverage.100kb$chr=="chr12"),]
chr13.100kb <- coverage.100kb[which(coverage.100kb$chr=="chr13"),]
chr14.100kb <- coverage.100kb[which(coverage.100kb$chr=="chr14"),]
chr15.100kb <- coverage.100kb[which(coverage.100kb$chr=="chr15"),]
chr16.100kb <- coverage.100kb[which(coverage.100kb$chr=="chr16"),]
chr17.100kb <- coverage.100kb[which(coverage.100kb$chr=="chr17"),]
chr18.100kb <- coverage.100kb[which(coverage.100kb$chr=="chr18"),]
chr19.100kb <- coverage.100kb[which(coverage.100kb$chr=="chr19"),]
chr20.100kb <- coverage.100kb[which(coverage.100kb$chr=="chr20"),]
chr21.100kb <- coverage.100kb[which(coverage.100kb$chr=="chr21"),]
chr22.100kb <- coverage.100kb[which(coverage.100kb$chr=="chr22"),]
chrX.100kb <- coverage.100kb[which(coverage.100kb$chr=="chrX"),]
chrY.100kb <- coverage.100kb[which(coverage.100kb$chr=="chrY"),]

#Get the coverage for .125kb for all the chromosomes
chr1.125kb <- coverage.125kb[which(coverage.125kb$chr=="chr1"),]
chr2.125kb <- coverage.125kb[which(coverage.125kb$chr=="chr2"),]
chr3.125kb <- coverage.125kb[which(coverage.125kb$chr=="chr3"),]
chr4.125kb <- coverage.125kb[which(coverage.125kb$chr=="chr4"),]
chr5.125kb <- coverage.125kb[which(coverage.125kb$chr=="chr5"),]
chr6.125kb <- coverage.125kb[which(coverage.125kb$chr=="chr6"),]
chr7.125kb <- coverage.125kb[which(coverage.125kb$chr=="chr7"),]
chr8.125kb <- coverage.125kb[which(coverage.125kb$chr=="chr8"),]
chr9.125kb <- coverage.125kb[which(coverage.125kb$chr=="chr9"),]
chr10.125kb <- coverage.125kb[which(coverage.125kb$chr=="chr10"),]
chr11.125kb <- coverage.125kb[which(coverage.125kb$chr=="chr11"),]
chr12.125kb <- coverage.125kb[which(coverage.125kb$chr=="chr12"),]
chr13.125kb <- coverage.125kb[which(coverage.125kb$chr=="chr13"),]
chr14.125kb <- coverage.125kb[which(coverage.125kb$chr=="chr14"),]
chr15.125kb <- coverage.125kb[which(coverage.125kb$chr=="chr15"),]
chr16.125kb <- coverage.125kb[which(coverage.125kb$chr=="chr16"),]
chr17.125kb <- coverage.125kb[which(coverage.125kb$chr=="chr17"),]
chr18.125kb <- coverage.125kb[which(coverage.125kb$chr=="chr18"),]
chr19.125kb <- coverage.125kb[which(coverage.125kb$chr=="chr19"),]
chr20.125kb <- coverage.125kb[which(coverage.125kb$chr=="chr20"),]
chr21.125kb <- coverage.125kb[which(coverage.125kb$chr=="chr21"),]
chr22.125kb <- coverage.125kb[which(coverage.125kb$chr=="chr22"),]
chrX.125kb <- coverage.125kb[which(coverage.125kb$chr=="chrX"),]
chrY.125kb <- coverage.125kb[which(coverage.125kb$chr=="chrY"),]

#Get the coverage for .150kb for all the chromosomes
chr1.150kb <- coverage.150kb[which(coverage.150kb$chr=="chr1"),]
chr2.150kb <- coverage.150kb[which(coverage.150kb$chr=="chr2"),]
chr3.150kb <- coverage.150kb[which(coverage.150kb$chr=="chr3"),]
chr4.150kb <- coverage.150kb[which(coverage.150kb$chr=="chr4"),]
chr5.150kb <- coverage.150kb[which(coverage.150kb$chr=="chr5"),]
chr6.150kb <- coverage.150kb[which(coverage.150kb$chr=="chr6"),]
chr7.150kb <- coverage.150kb[which(coverage.150kb$chr=="chr7"),]
chr8.150kb <- coverage.150kb[which(coverage.150kb$chr=="chr8"),]
chr9.150kb <- coverage.150kb[which(coverage.150kb$chr=="chr9"),]
chr10.150kb <- coverage.150kb[which(coverage.150kb$chr=="chr10"),]
chr11.150kb <- coverage.150kb[which(coverage.150kb$chr=="chr11"),]
chr12.150kb <- coverage.150kb[which(coverage.150kb$chr=="chr12"),]
chr13.150kb <- coverage.150kb[which(coverage.150kb$chr=="chr13"),]
chr14.150kb <- coverage.150kb[which(coverage.150kb$chr=="chr14"),]
chr15.150kb <- coverage.150kb[which(coverage.150kb$chr=="chr15"),]
chr16.150kb <- coverage.150kb[which(coverage.150kb$chr=="chr16"),]
chr17.150kb <- coverage.150kb[which(coverage.150kb$chr=="chr17"),]
chr18.150kb <- coverage.150kb[which(coverage.150kb$chr=="chr18"),]
chr19.150kb <- coverage.150kb[which(coverage.150kb$chr=="chr19"),]
chr20.150kb <- coverage.150kb[which(coverage.150kb$chr=="chr20"),]
chr21.150kb <- coverage.150kb[which(coverage.150kb$chr=="chr21"),]
chr22.150kb <- coverage.150kb[which(coverage.150kb$chr=="chr22"),]
chrX.150kb <- coverage.150kb[which(coverage.150kb$chr=="chrX"),]
chrY.150kb <- coverage.150kb[which(coverage.150kb$chr=="chrY"),]

#Get the coverage for .200kb for all the chromosomes
chr1.200kb <- coverage.200kb[which(coverage.200kb$chr=="chr1"),]
chr2.200kb <- coverage.200kb[which(coverage.200kb$chr=="chr2"),]
chr3.200kb <- coverage.200kb[which(coverage.200kb$chr=="chr3"),]
chr4.200kb <- coverage.200kb[which(coverage.200kb$chr=="chr4"),]
chr5.200kb <- coverage.200kb[which(coverage.200kb$chr=="chr5"),]
chr6.200kb <- coverage.200kb[which(coverage.200kb$chr=="chr6"),]
chr7.200kb <- coverage.200kb[which(coverage.200kb$chr=="chr7"),]
chr8.200kb <- coverage.200kb[which(coverage.200kb$chr=="chr8"),]
chr9.200kb <- coverage.200kb[which(coverage.200kb$chr=="chr9"),]
chr10.200kb <- coverage.200kb[which(coverage.200kb$chr=="chr10"),]
chr11.200kb <- coverage.200kb[which(coverage.200kb$chr=="chr11"),]
chr12.200kb <- coverage.200kb[which(coverage.200kb$chr=="chr12"),]
chr13.200kb <- coverage.200kb[which(coverage.200kb$chr=="chr13"),]
chr14.200kb <- coverage.200kb[which(coverage.200kb$chr=="chr14"),]
chr15.200kb <- coverage.200kb[which(coverage.200kb$chr=="chr15"),]
chr16.200kb <- coverage.200kb[which(coverage.200kb$chr=="chr16"),]
chr17.200kb <- coverage.200kb[which(coverage.200kb$chr=="chr17"),]
chr18.200kb <- coverage.200kb[which(coverage.200kb$chr=="chr18"),]
chr19.200kb <- coverage.200kb[which(coverage.200kb$chr=="chr19"),]
chr20.200kb <- coverage.200kb[which(coverage.200kb$chr=="chr20"),]
chr21.200kb <- coverage.200kb[which(coverage.200kb$chr=="chr21"),]
chr22.200kb <- coverage.200kb[which(coverage.200kb$chr=="chr22"),]
chrX.200kb <- coverage.200kb[which(coverage.200kb$chr=="chrX"),]
chrY.200kb <- coverage.200kb[which(coverage.200kb$chr=="chrY"),]



#Get the bins for .50kb for all the chromosomes.
bins.1.50kb <- optimize.bin.sums(chr1.50kb)
bins.2.50kb <- optimize.bin.sums(chr2.50kb)
bins.3.50kb <- optimize.bin.sums(chr3.50kb)
bins.4.50kb <- optimize.bin.sums(chr4.50kb)
bins.5.50kb <- optimize.bin.sums(chr5.50kb)
bins.6.50kb <- optimize.bin.sums(chr6.50kb)
bins.7.50kb <- optimize.bin.sums(chr7.50kb)
bins.8.50kb <- optimize.bin.sums(chr8.50kb)
bins.9.50kb <- optimize.bin.sums(chr9.50kb)
bins.10.50kb <- optimize.bin.sums(chr10.50kb)
bins.11.50kb <- optimize.bin.sums(chr11.50kb)
bins.12.50kb <- optimize.bin.sums(chr12.50kb)
bins.13.50kb <- optimize.bin.sums(chr13.50kb)
bins.14.50kb <- optimize.bin.sums(chr14.50kb)
bins.15.50kb <- optimize.bin.sums(chr15.50kb)
bins.16.50kb <- optimize.bin.sums(chr16.50kb)
bins.17.50kb <- optimize.bin.sums(chr17.50kb)
bins.18.50kb <- optimize.bin.sums(chr18.50kb)
bins.19.50kb <- optimize.bin.sums(chr19.50kb)
bins.20.50kb <- optimize.bin.sums(chr20.50kb)
bins.21.50kb <- optimize.bin.sums(chr21.50kb)
bins.22.50kb <- optimize.bin.sums(chr22.50kb)
bins.X.50kb <- optimize.bin.sums(chrX.50kb)
bins.Y.50kb <- optimize.bin.sums(chrY.50kb)

#Get the bins for .60kb for all the chromosomes.
bins.1.60kb <- optimize.bin.sums(chr1.60kb)
bins.2.60kb <- optimize.bin.sums(chr2.60kb)
bins.3.60kb <- optimize.bin.sums(chr3.60kb)
bins.4.60kb <- optimize.bin.sums(chr4.60kb)
bins.5.60kb <- optimize.bin.sums(chr5.60kb)
bins.6.60kb <- optimize.bin.sums(chr6.60kb)
bins.7.60kb <- optimize.bin.sums(chr7.60kb)
bins.8.60kb <- optimize.bin.sums(chr8.60kb)
bins.9.60kb <- optimize.bin.sums(chr9.60kb)
bins.10.60kb <- optimize.bin.sums(chr10.60kb)
bins.11.60kb <- optimize.bin.sums(chr11.60kb)
bins.12.60kb <- optimize.bin.sums(chr12.60kb)
bins.13.60kb <- optimize.bin.sums(chr13.60kb)
bins.14.60kb <- optimize.bin.sums(chr14.60kb)
bins.15.60kb <- optimize.bin.sums(chr15.60kb)
bins.16.60kb <- optimize.bin.sums(chr16.60kb)
bins.17.60kb <- optimize.bin.sums(chr17.60kb)
bins.18.60kb <- optimize.bin.sums(chr18.60kb)
bins.19.60kb <- optimize.bin.sums(chr19.60kb)
bins.20.60kb <- optimize.bin.sums(chr20.60kb)
bins.21.60kb <- optimize.bin.sums(chr21.60kb)
bins.22.60kb <- optimize.bin.sums(chr22.60kb)
bins.X.60kb <- optimize.bin.sums(chrX.60kb)
bins.Y.60kb <- optimize.bin.sums(chrY.60kb)

#Get the bins for .70kb for all the chromosomes.
bins.1.70kb <- optimize.bin.sums(chr1.70kb)
bins.2.70kb <- optimize.bin.sums(chr2.70kb)
bins.3.70kb <- optimize.bin.sums(chr3.70kb)
bins.4.70kb <- optimize.bin.sums(chr4.70kb)
bins.5.70kb <- optimize.bin.sums(chr5.70kb)
bins.6.70kb <- optimize.bin.sums(chr6.70kb)
bins.7.70kb <- optimize.bin.sums(chr7.70kb)
bins.8.70kb <- optimize.bin.sums(chr8.70kb)
bins.9.70kb <- optimize.bin.sums(chr9.70kb)
bins.10.70kb <- optimize.bin.sums(chr10.70kb)
bins.11.70kb <- optimize.bin.sums(chr11.70kb)
bins.12.70kb <- optimize.bin.sums(chr12.70kb)
bins.13.70kb <- optimize.bin.sums(chr13.70kb)
bins.14.70kb <- optimize.bin.sums(chr14.70kb)
bins.15.70kb <- optimize.bin.sums(chr15.70kb)
bins.16.70kb <- optimize.bin.sums(chr16.70kb)
bins.17.70kb <- optimize.bin.sums(chr17.70kb)
bins.18.70kb <- optimize.bin.sums(chr18.70kb)
bins.19.70kb <- optimize.bin.sums(chr19.70kb)
bins.20.70kb <- optimize.bin.sums(chr20.70kb)
bins.21.70kb <- optimize.bin.sums(chr21.70kb)
bins.22.70kb <- optimize.bin.sums(chr22.70kb)
bins.X.70kb <- optimize.bin.sums(chrX.70kb)
bins.Y.70kb <- optimize.bin.sums(chrY.70kb)

#Get the bins for .80kb for all the chromosomes.
bins.1.80kb <- optimize.bin.sums(chr1.80kb)
bins.2.80kb <- optimize.bin.sums(chr2.80kb)
bins.3.80kb <- optimize.bin.sums(chr3.80kb)
bins.4.80kb <- optimize.bin.sums(chr4.80kb)
bins.5.80kb <- optimize.bin.sums(chr5.80kb)
bins.6.80kb <- optimize.bin.sums(chr6.80kb)
bins.7.80kb <- optimize.bin.sums(chr7.80kb)
bins.8.80kb <- optimize.bin.sums(chr8.80kb)
bins.9.80kb <- optimize.bin.sums(chr9.80kb)
bins.10.80kb <- optimize.bin.sums(chr10.80kb)
bins.11.80kb <- optimize.bin.sums(chr11.80kb)
bins.12.80kb <- optimize.bin.sums(chr12.80kb)
bins.13.80kb <- optimize.bin.sums(chr13.80kb)
bins.14.80kb <- optimize.bin.sums(chr14.80kb)
bins.15.80kb <- optimize.bin.sums(chr15.80kb)
bins.16.80kb <- optimize.bin.sums(chr16.80kb)
bins.17.80kb <- optimize.bin.sums(chr17.80kb)
bins.18.80kb <- optimize.bin.sums(chr18.80kb)
bins.19.80kb <- optimize.bin.sums(chr19.80kb)
bins.20.80kb <- optimize.bin.sums(chr20.80kb)
bins.21.80kb <- optimize.bin.sums(chr21.80kb)
bins.22.80kb <- optimize.bin.sums(chr22.80kb)
bins.X.80kb <- optimize.bin.sums(chrX.80kb)
bins.Y.80kb <- optimize.bin.sums(chrY.80kb)

#Get the bins for .90kb for all the chromosomes.
bins.1.90kb <- optimize.bin.sums(chr1.90kb)
bins.2.90kb <- optimize.bin.sums(chr2.90kb)
bins.3.90kb <- optimize.bin.sums(chr3.90kb)
bins.4.90kb <- optimize.bin.sums(chr4.90kb)
bins.5.90kb <- optimize.bin.sums(chr5.90kb)
bins.6.90kb <- optimize.bin.sums(chr6.90kb)
bins.7.90kb <- optimize.bin.sums(chr7.90kb)
bins.8.90kb <- optimize.bin.sums(chr8.90kb)
bins.9.90kb <- optimize.bin.sums(chr9.90kb)
bins.10.90kb <- optimize.bin.sums(chr10.90kb)
bins.11.90kb <- optimize.bin.sums(chr11.90kb)
bins.12.90kb <- optimize.bin.sums(chr12.90kb)
bins.13.90kb <- optimize.bin.sums(chr13.90kb)
bins.14.90kb <- optimize.bin.sums(chr14.90kb)
bins.15.90kb <- optimize.bin.sums(chr15.90kb)
bins.16.90kb <- optimize.bin.sums(chr16.90kb)
bins.17.90kb <- optimize.bin.sums(chr17.90kb)
bins.18.90kb <- optimize.bin.sums(chr18.90kb)
bins.19.90kb <- optimize.bin.sums(chr19.90kb)
bins.20.90kb <- optimize.bin.sums(chr20.90kb)
bins.21.90kb <- optimize.bin.sums(chr21.90kb)
bins.22.90kb <- optimize.bin.sums(chr22.90kb)
bins.X.90kb <- optimize.bin.sums(chrX.90kb)
bins.Y.90kb <- optimize.bin.sums(chrY.90kb)

#Get the bins for .100kb for all the chromosomes.
bins.1.100kb <- optimize.bin.sums(chr1.100kb)
bins.2.100kb <- optimize.bin.sums(chr2.100kb)
bins.3.100kb <- optimize.bin.sums(chr3.100kb)
bins.4.100kb <- optimize.bin.sums(chr4.100kb)
bins.5.100kb <- optimize.bin.sums(chr5.100kb)
bins.6.100kb <- optimize.bin.sums(chr6.100kb)
bins.7.100kb <- optimize.bin.sums(chr7.100kb)
bins.8.100kb <- optimize.bin.sums(chr8.100kb)
bins.9.100kb <- optimize.bin.sums(chr9.100kb)
bins.10.100kb <- optimize.bin.sums(chr10.100kb)
bins.11.100kb <- optimize.bin.sums(chr11.100kb)
bins.12.100kb <- optimize.bin.sums(chr12.100kb)
bins.13.100kb <- optimize.bin.sums(chr13.100kb)
bins.14.100kb <- optimize.bin.sums(chr14.100kb)
bins.15.100kb <- optimize.bin.sums(chr15.100kb)
bins.16.100kb <- optimize.bin.sums(chr16.100kb)
bins.17.100kb <- optimize.bin.sums(chr17.100kb)
bins.18.100kb <- optimize.bin.sums(chr18.100kb)
bins.19.100kb <- optimize.bin.sums(chr19.100kb)
bins.20.100kb <- optimize.bin.sums(chr20.100kb)
bins.21.100kb <- optimize.bin.sums(chr21.100kb)
bins.22.100kb <- optimize.bin.sums(chr22.100kb)
bins.X.100kb <- optimize.bin.sums(chrX.100kb)
bins.Y.100kb <- optimize.bin.sums(chrY.100kb)

#Get the bins for .125kb for all the chromosomes.
bins.1.125kb <- optimize.bin.sums(chr1.125kb)
bins.2.125kb <- optimize.bin.sums(chr2.125kb)
bins.3.125kb <- optimize.bin.sums(chr3.125kb)
bins.4.125kb <- optimize.bin.sums(chr4.125kb)
bins.5.125kb <- optimize.bin.sums(chr5.125kb)
bins.6.125kb <- optimize.bin.sums(chr6.125kb)
bins.7.125kb <- optimize.bin.sums(chr7.125kb)
bins.8.125kb <- optimize.bin.sums(chr8.125kb)
bins.9.125kb <- optimize.bin.sums(chr9.125kb)
bins.10.125kb <- optimize.bin.sums(chr10.125kb)
bins.11.125kb <- optimize.bin.sums(chr11.125kb)
bins.12.125kb <- optimize.bin.sums(chr12.125kb)
bins.13.125kb <- optimize.bin.sums(chr13.125kb)
bins.14.125kb <- optimize.bin.sums(chr14.125kb)
bins.15.125kb <- optimize.bin.sums(chr15.125kb)
bins.16.125kb <- optimize.bin.sums(chr16.125kb)
bins.17.125kb <- optimize.bin.sums(chr17.125kb)
bins.18.125kb <- optimize.bin.sums(chr18.125kb)
bins.19.125kb <- optimize.bin.sums(chr19.125kb)
bins.20.125kb <- optimize.bin.sums(chr20.125kb)
bins.21.125kb <- optimize.bin.sums(chr21.125kb)
bins.22.125kb <- optimize.bin.sums(chr22.125kb)
bins.X.125kb <- optimize.bin.sums(chrX.125kb)
bins.Y.125kb <- optimize.bin.sums(chrY.125kb)

#Get the bins for .150kb for all the chromosomes.
bins.1.150kb <- optimize.bin.sums(chr1.150kb)
bins.2.150kb <- optimize.bin.sums(chr2.150kb)
bins.3.150kb <- optimize.bin.sums(chr3.150kb)
bins.4.150kb <- optimize.bin.sums(chr4.150kb)
bins.5.150kb <- optimize.bin.sums(chr5.150kb)
bins.6.150kb <- optimize.bin.sums(chr6.150kb)
bins.7.150kb <- optimize.bin.sums(chr7.150kb)
bins.8.150kb <- optimize.bin.sums(chr8.150kb)
bins.9.150kb <- optimize.bin.sums(chr9.150kb)
bins.10.150kb <- optimize.bin.sums(chr10.150kb)
bins.11.150kb <- optimize.bin.sums(chr11.150kb)
bins.12.150kb <- optimize.bin.sums(chr12.150kb)
bins.13.150kb <- optimize.bin.sums(chr13.150kb)
bins.14.150kb <- optimize.bin.sums(chr14.150kb)
bins.15.150kb <- optimize.bin.sums(chr15.150kb)
bins.16.150kb <- optimize.bin.sums(chr16.150kb)
bins.17.150kb <- optimize.bin.sums(chr17.150kb)
bins.18.150kb <- optimize.bin.sums(chr18.150kb)
bins.19.150kb <- optimize.bin.sums(chr19.150kb)
bins.20.150kb <- optimize.bin.sums(chr20.150kb)
bins.21.150kb <- optimize.bin.sums(chr21.150kb)
bins.22.150kb <- optimize.bin.sums(chr22.150kb)
bins.X.150kb <- optimize.bin.sums(chrX.150kb)
bins.Y.150kb <- optimize.bin.sums(chrY.150kb)

#Get the bins for .200kb for all the chromosomes.
bins.1.200kb <- optimize.bin.sums(chr1.200kb)
bins.2.200kb <- optimize.bin.sums(chr2.200kb)
bins.3.200kb <- optimize.bin.sums(chr3.200kb)
bins.4.200kb <- optimize.bin.sums(chr4.200kb)
bins.5.200kb <- optimize.bin.sums(chr5.200kb)
bins.6.200kb <- optimize.bin.sums(chr6.200kb)
bins.7.200kb <- optimize.bin.sums(chr7.200kb)
bins.8.200kb <- optimize.bin.sums(chr8.200kb)
bins.9.200kb <- optimize.bin.sums(chr9.200kb)
bins.10.200kb <- optimize.bin.sums(chr10.200kb)
bins.11.200kb <- optimize.bin.sums(chr11.200kb)
bins.12.200kb <- optimize.bin.sums(chr12.200kb)
bins.13.200kb <- optimize.bin.sums(chr13.200kb)
bins.14.200kb <- optimize.bin.sums(chr14.200kb)
bins.15.200kb <- optimize.bin.sums(chr15.200kb)
bins.16.200kb <- optimize.bin.sums(chr16.200kb)
bins.17.200kb <- optimize.bin.sums(chr17.200kb)
bins.18.200kb <- optimize.bin.sums(chr18.200kb)
bins.19.200kb <- optimize.bin.sums(chr19.200kb)
bins.20.200kb <- optimize.bin.sums(chr20.200kb)
bins.21.200kb <- optimize.bin.sums(chr21.200kb)
bins.22.200kb <- optimize.bin.sums(chr22.200kb)
bins.X.200kb <- optimize.bin.sums(chrX.200kb)
bins.Y.200kb <- optimize.bin.sums(chrY.200kb)
```

```{r New Data assignation, legit AF. Saved in one big dataframe at the end!}

#Get pos dfs!
setwd("/Users/ittaieres/Desktop/Hi-C/All_good_analysis/cis_only/paired_read_midpoint_positions")
pos_df.1 <- fread("midpoint.pairs.1", header=FALSE, stringsAsFactors = FALSE)
pos_df.2 <- fread("midpoint.pairs.2", header=FALSE, stringsAsFactors = FALSE)
pos_df.3 <- fread("midpoint.pairs.3", header=FALSE, stringsAsFactors = FALSE)
pos_df.4 <- fread("midpoint.pairs.4", header=FALSE, stringsAsFactors = FALSE)
pos_df.5 <- fread("midpoint.pairs.5", header=FALSE, stringsAsFactors = FALSE)
pos_df.6 <- fread("midpoint.pairs.6", header=FALSE, stringsAsFactors = FALSE)
pos_df.7 <- fread("midpoint.pairs.7", header=FALSE, stringsAsFactors = FALSE)
pos_df.8 <- fread("midpoint.pairs.8", header=FALSE, stringsAsFactors = FALSE)
pos_df.9 <- fread("midpoint.pairs.9", header=FALSE, stringsAsFactors = FALSE)
pos_df.10 <- fread("midpoint.pairs.10", header=FALSE, stringsAsFactors = FALSE)
pos_df.11 <- fread("midpoint.pairs.11", header=FALSE, stringsAsFactors = FALSE)
pos_df.12 <- fread("midpoint.pairs.12", header=FALSE, stringsAsFactors = FALSE)
pos_df.13 <- fread("midpoint.pairs.13", header=FALSE, stringsAsFactors = FALSE)
pos_df.14 <- fread("midpoint.pairs.14", header=FALSE, stringsAsFactors = FALSE)
pos_df.15 <- fread("midpoint.pairs.15", header=FALSE, stringsAsFactors = FALSE)
pos_df.16 <- fread("midpoint.pairs.16", header=FALSE, stringsAsFactors = FALSE)
pos_df.17 <- fread("midpoint.pairs.17", header=FALSE, stringsAsFactors = FALSE)
pos_df.18 <- fread("midpoint.pairs.18", header=FALSE, stringsAsFactors = FALSE)
pos_df.19 <- fread("midpoint.pairs.19", header=FALSE, stringsAsFactors = FALSE)
pos_df.20 <- fread("midpoint.pairs.20", header=FALSE, stringsAsFactors = FALSE)
pos_df.21 <- fread("midpoint.pairs.21", header=FALSE, stringsAsFactors = FALSE)
pos_df.22 <- fread("midpoint.pairs.22", header=FALSE, stringsAsFactors = FALSE)
pos_df.X <- fread("midpoint.pairs.X", header=FALSE, stringsAsFactors = FALSE)
pos_df.Y <- fread("midpoint.pairs.Y", header=FALSE, stringsAsFactors = FALSE)

#Get coverage dfs!
setwd("/Users/ittaieres/Desktop/Hi-C/All_good_analysis/cis_only/cis_only_coverage")
coverage.10kb <- fread("coverage_10kb_5kb.bed", sep="\t", header=FALSE, stringsAsFactors = FALSE)
coverage.20kb <- fread("coverage_20kb_5kb.bed", sep="\t", header=FALSE, stringsAsFactors = FALSE)
coverage.30kb <- fread("coverage_30kb_5kb.bed", sep="\t", header=FALSE, stringsAsFactors = FALSE)
coverage.40kb <- fread("coverage_40kb_5kb.bed", sep="\t", header=FALSE, stringsAsFactors = FALSE)
coverage.50kb <- fread("coverage_50kb_5kb.bed", sep="\t", header=FALSE, stringsAsFactors = FALSE)
coverage.60kb <- fread("coverage_60kb_5kb.bed", sep="\t", header=FALSE, stringsAsFactors = FALSE)
coverage.70kb <- fread("coverage_70kb_5kb.bed", sep="\t", header=FALSE, stringsAsFactors = FALSE)
coverage.80kb <- fread("coverage_80kb_5kb.bed", sep="\t", header=FALSE, stringsAsFactors = FALSE)
coverage.90kb <- fread("coverage_90kb_5kb.bed", sep="\t", header=FALSE, stringsAsFactors = FALSE)
coverage.100kb <- fread("coverage_100kb_5kb.bed", sep="\t", header=FALSE, stringsAsFactors = FALSE)
coverage.125kb <- fread("coverage_125kb_5kb.bed", sep="\t", header=FALSE, stringsAsFactors = FALSE)
coverage.150kb <- fread("coverage_150kb_5kb.bed", sep="\t", header=FALSE, stringsAsFactors = FALSE)
coverage.200kb <- fread("coverage_200kb_5kb.bed", sep="\t", header=FALSE, stringsAsFactors = FALSE)
#Assign their column names.
colnames(coverage.10kb) <- c("chr", "start", "end", "count")
colnames(coverage.20kb) <- c("chr", "start", "end", "count")
colnames(coverage.30kb) <- c("chr", "start", "end", "count")
colnames(coverage.40kb) <- c("chr", "start", "end", "count")
colnames(coverage.50kb) <- c("chr", "start", "end", "count")
colnames(coverage.60kb) <- c("chr", "start", "end", "count")
colnames(coverage.70kb) <- c("chr", "start", "end", "count")
colnames(coverage.80kb) <- c("chr", "start", "end", "count")
colnames(coverage.90kb) <- c("chr", "start", "end", "count")
colnames(coverage.100kb) <- c("chr", "start", "end", "count")
colnames(coverage.125kb) <- c("chr", "start", "end", "count")
colnames(coverage.150kb) <- c("chr", "start", "end", "count")
colnames(coverage.200kb) <- c("chr", "start", "end", "count")

##Get coverage dfs individually by chromosome, across all locus bin sizes.
#Get the coverage for .10kb for all the chromosomes
chr1.10kb <- coverage.10kb[which(coverage.10kb$chr=="chr1"),]
chr2.10kb <- coverage.10kb[which(coverage.10kb$chr=="chr2"),]
chr3.10kb <- coverage.10kb[which(coverage.10kb$chr=="chr3"),]
chr4.10kb <- coverage.10kb[which(coverage.10kb$chr=="chr4"),]
chr5.10kb <- coverage.10kb[which(coverage.10kb$chr=="chr5"),]
chr6.10kb <- coverage.10kb[which(coverage.10kb$chr=="chr6"),]
chr7.10kb <- coverage.10kb[which(coverage.10kb$chr=="chr7"),]
chr8.10kb <- coverage.10kb[which(coverage.10kb$chr=="chr8"),]
chr9.10kb <- coverage.10kb[which(coverage.10kb$chr=="chr9"),]
chr10.10kb <- coverage.10kb[which(coverage.10kb$chr=="chr10"),]
chr11.10kb <- coverage.10kb[which(coverage.10kb$chr=="chr11"),]
chr12.10kb <- coverage.10kb[which(coverage.10kb$chr=="chr12"),]
chr13.10kb <- coverage.10kb[which(coverage.10kb$chr=="chr13"),]
chr14.10kb <- coverage.10kb[which(coverage.10kb$chr=="chr14"),]
chr15.10kb <- coverage.10kb[which(coverage.10kb$chr=="chr15"),]
chr16.10kb <- coverage.10kb[which(coverage.10kb$chr=="chr16"),]
chr17.10kb <- coverage.10kb[which(coverage.10kb$chr=="chr17"),]
chr18.10kb <- coverage.10kb[which(coverage.10kb$chr=="chr18"),]
chr19.10kb <- coverage.10kb[which(coverage.10kb$chr=="chr19"),]
chr20.10kb <- coverage.10kb[which(coverage.10kb$chr=="chr20"),]
chr21.10kb <- coverage.10kb[which(coverage.10kb$chr=="chr21"),]
chr22.10kb <- coverage.10kb[which(coverage.10kb$chr=="chr22"),]
chrX.10kb <- coverage.10kb[which(coverage.10kb$chr=="chrX"),]
chrY.10kb <- coverage.10kb[which(coverage.10kb$chr=="chrY"),]

#Get the coverage for .20kb for all the chromosomes
chr1.20kb <- coverage.20kb[which(coverage.20kb$chr=="chr1"),]
chr2.20kb <- coverage.20kb[which(coverage.20kb$chr=="chr2"),]
chr3.20kb <- coverage.20kb[which(coverage.20kb$chr=="chr3"),]
chr4.20kb <- coverage.20kb[which(coverage.20kb$chr=="chr4"),]
chr5.20kb <- coverage.20kb[which(coverage.20kb$chr=="chr5"),]
chr6.20kb <- coverage.20kb[which(coverage.20kb$chr=="chr6"),]
chr7.20kb <- coverage.20kb[which(coverage.20kb$chr=="chr7"),]
chr8.20kb <- coverage.20kb[which(coverage.20kb$chr=="chr8"),]
chr9.20kb <- coverage.20kb[which(coverage.20kb$chr=="chr9"),]
chr10.20kb <- coverage.20kb[which(coverage.20kb$chr=="chr10"),]
chr11.20kb <- coverage.20kb[which(coverage.20kb$chr=="chr11"),]
chr12.20kb <- coverage.20kb[which(coverage.20kb$chr=="chr12"),]
chr13.20kb <- coverage.20kb[which(coverage.20kb$chr=="chr13"),]
chr14.20kb <- coverage.20kb[which(coverage.20kb$chr=="chr14"),]
chr15.20kb <- coverage.20kb[which(coverage.20kb$chr=="chr15"),]
chr16.20kb <- coverage.20kb[which(coverage.20kb$chr=="chr16"),]
chr17.20kb <- coverage.20kb[which(coverage.20kb$chr=="chr17"),]
chr18.20kb <- coverage.20kb[which(coverage.20kb$chr=="chr18"),]
chr19.20kb <- coverage.20kb[which(coverage.20kb$chr=="chr19"),]
chr20.20kb <- coverage.20kb[which(coverage.20kb$chr=="chr20"),]
chr21.20kb <- coverage.20kb[which(coverage.20kb$chr=="chr21"),]
chr22.20kb <- coverage.20kb[which(coverage.20kb$chr=="chr22"),]
chrX.20kb <- coverage.20kb[which(coverage.20kb$chr=="chrX"),]
chrY.20kb <- coverage.20kb[which(coverage.20kb$chr=="chrY"),]

#Get the coverage for .30kb for all the chromosomes
chr1.30kb <- coverage.30kb[which(coverage.30kb$chr=="chr1"),]
chr2.30kb <- coverage.30kb[which(coverage.30kb$chr=="chr2"),]
chr3.30kb <- coverage.30kb[which(coverage.30kb$chr=="chr3"),]
chr4.30kb <- coverage.30kb[which(coverage.30kb$chr=="chr4"),]
chr5.30kb <- coverage.30kb[which(coverage.30kb$chr=="chr5"),]
chr6.30kb <- coverage.30kb[which(coverage.30kb$chr=="chr6"),]
chr7.30kb <- coverage.30kb[which(coverage.30kb$chr=="chr7"),]
chr8.30kb <- coverage.30kb[which(coverage.30kb$chr=="chr8"),]
chr9.30kb <- coverage.30kb[which(coverage.30kb$chr=="chr9"),]
chr10.30kb <- coverage.30kb[which(coverage.30kb$chr=="chr10"),]
chr11.30kb <- coverage.30kb[which(coverage.30kb$chr=="chr11"),]
chr12.30kb <- coverage.30kb[which(coverage.30kb$chr=="chr12"),]
chr13.30kb <- coverage.30kb[which(coverage.30kb$chr=="chr13"),]
chr14.30kb <- coverage.30kb[which(coverage.30kb$chr=="chr14"),]
chr15.30kb <- coverage.30kb[which(coverage.30kb$chr=="chr15"),]
chr16.30kb <- coverage.30kb[which(coverage.30kb$chr=="chr16"),]
chr17.30kb <- coverage.30kb[which(coverage.30kb$chr=="chr17"),]
chr18.30kb <- coverage.30kb[which(coverage.30kb$chr=="chr18"),]
chr19.30kb <- coverage.30kb[which(coverage.30kb$chr=="chr19"),]
chr20.30kb <- coverage.30kb[which(coverage.30kb$chr=="chr20"),]
chr21.30kb <- coverage.30kb[which(coverage.30kb$chr=="chr21"),]
chr22.30kb <- coverage.30kb[which(coverage.30kb$chr=="chr22"),]
chrX.30kb <- coverage.30kb[which(coverage.30kb$chr=="chrX"),]
chrY.30kb <- coverage.30kb[which(coverage.30kb$chr=="chrY"),]

#Get the coverage for .40kb for all the chromosomes
chr1.40kb <- coverage.40kb[which(coverage.40kb$chr=="chr1"),]
chr2.40kb <- coverage.40kb[which(coverage.40kb$chr=="chr2"),]
chr3.40kb <- coverage.40kb[which(coverage.40kb$chr=="chr3"),]
chr4.40kb <- coverage.40kb[which(coverage.40kb$chr=="chr4"),]
chr5.40kb <- coverage.40kb[which(coverage.40kb$chr=="chr5"),]
chr6.40kb <- coverage.40kb[which(coverage.40kb$chr=="chr6"),]
chr7.40kb <- coverage.40kb[which(coverage.40kb$chr=="chr7"),]
chr8.40kb <- coverage.40kb[which(coverage.40kb$chr=="chr8"),]
chr9.40kb <- coverage.40kb[which(coverage.40kb$chr=="chr9"),]
chr10.40kb <- coverage.40kb[which(coverage.40kb$chr=="chr10"),]
chr11.40kb <- coverage.40kb[which(coverage.40kb$chr=="chr11"),]
chr12.40kb <- coverage.40kb[which(coverage.40kb$chr=="chr12"),]
chr13.40kb <- coverage.40kb[which(coverage.40kb$chr=="chr13"),]
chr14.40kb <- coverage.40kb[which(coverage.40kb$chr=="chr14"),]
chr15.40kb <- coverage.40kb[which(coverage.40kb$chr=="chr15"),]
chr16.40kb <- coverage.40kb[which(coverage.40kb$chr=="chr16"),]
chr17.40kb <- coverage.40kb[which(coverage.40kb$chr=="chr17"),]
chr18.40kb <- coverage.40kb[which(coverage.40kb$chr=="chr18"),]
chr19.40kb <- coverage.40kb[which(coverage.40kb$chr=="chr19"),]
chr20.40kb <- coverage.40kb[which(coverage.40kb$chr=="chr20"),]
chr21.40kb <- coverage.40kb[which(coverage.40kb$chr=="chr21"),]
chr22.40kb <- coverage.40kb[which(coverage.40kb$chr=="chr22"),]
chrX.40kb <- coverage.40kb[which(coverage.40kb$chr=="chrX"),]
chrY.40kb <- coverage.40kb[which(coverage.40kb$chr=="chrY"),]

#Get the coverage for .50kb for all the chromosomes
chr1.50kb <- coverage.50kb[which(coverage.50kb$chr=="chr1"),]
chr2.50kb <- coverage.50kb[which(coverage.50kb$chr=="chr2"),]
chr3.50kb <- coverage.50kb[which(coverage.50kb$chr=="chr3"),]
chr4.50kb <- coverage.50kb[which(coverage.50kb$chr=="chr4"),]
chr5.50kb <- coverage.50kb[which(coverage.50kb$chr=="chr5"),]
chr6.50kb <- coverage.50kb[which(coverage.50kb$chr=="chr6"),]
chr7.50kb <- coverage.50kb[which(coverage.50kb$chr=="chr7"),]
chr8.50kb <- coverage.50kb[which(coverage.50kb$chr=="chr8"),]
chr9.50kb <- coverage.50kb[which(coverage.50kb$chr=="chr9"),]
chr10.50kb <- coverage.50kb[which(coverage.50kb$chr=="chr10"),]
chr11.50kb <- coverage.50kb[which(coverage.50kb$chr=="chr11"),]
chr12.50kb <- coverage.50kb[which(coverage.50kb$chr=="chr12"),]
chr13.50kb <- coverage.50kb[which(coverage.50kb$chr=="chr13"),]
chr14.50kb <- coverage.50kb[which(coverage.50kb$chr=="chr14"),]
chr15.50kb <- coverage.50kb[which(coverage.50kb$chr=="chr15"),]
chr16.50kb <- coverage.50kb[which(coverage.50kb$chr=="chr16"),]
chr17.50kb <- coverage.50kb[which(coverage.50kb$chr=="chr17"),]
chr18.50kb <- coverage.50kb[which(coverage.50kb$chr=="chr18"),]
chr19.50kb <- coverage.50kb[which(coverage.50kb$chr=="chr19"),]
chr20.50kb <- coverage.50kb[which(coverage.50kb$chr=="chr20"),]
chr21.50kb <- coverage.50kb[which(coverage.50kb$chr=="chr21"),]
chr22.50kb <- coverage.50kb[which(coverage.50kb$chr=="chr22"),]
chrX.50kb <- coverage.50kb[which(coverage.50kb$chr=="chrX"),]
chrY.50kb <- coverage.50kb[which(coverage.50kb$chr=="chrY"),]

#Get the coverage for .60kb for all the chromosomes
chr1.60kb <- coverage.60kb[which(coverage.60kb$chr=="chr1"),]
chr2.60kb <- coverage.60kb[which(coverage.60kb$chr=="chr2"),]
chr3.60kb <- coverage.60kb[which(coverage.60kb$chr=="chr3"),]
chr4.60kb <- coverage.60kb[which(coverage.60kb$chr=="chr4"),]
chr5.60kb <- coverage.60kb[which(coverage.60kb$chr=="chr5"),]
chr6.60kb <- coverage.60kb[which(coverage.60kb$chr=="chr6"),]
chr7.60kb <- coverage.60kb[which(coverage.60kb$chr=="chr7"),]
chr8.60kb <- coverage.60kb[which(coverage.60kb$chr=="chr8"),]
chr9.60kb <- coverage.60kb[which(coverage.60kb$chr=="chr9"),]
chr10.60kb <- coverage.60kb[which(coverage.60kb$chr=="chr10"),]
chr11.60kb <- coverage.60kb[which(coverage.60kb$chr=="chr11"),]
chr12.60kb <- coverage.60kb[which(coverage.60kb$chr=="chr12"),]
chr13.60kb <- coverage.60kb[which(coverage.60kb$chr=="chr13"),]
chr14.60kb <- coverage.60kb[which(coverage.60kb$chr=="chr14"),]
chr15.60kb <- coverage.60kb[which(coverage.60kb$chr=="chr15"),]
chr16.60kb <- coverage.60kb[which(coverage.60kb$chr=="chr16"),]
chr17.60kb <- coverage.60kb[which(coverage.60kb$chr=="chr17"),]
chr18.60kb <- coverage.60kb[which(coverage.60kb$chr=="chr18"),]
chr19.60kb <- coverage.60kb[which(coverage.60kb$chr=="chr19"),]
chr20.60kb <- coverage.60kb[which(coverage.60kb$chr=="chr20"),]
chr21.60kb <- coverage.60kb[which(coverage.60kb$chr=="chr21"),]
chr22.60kb <- coverage.60kb[which(coverage.60kb$chr=="chr22"),]
chrX.60kb <- coverage.60kb[which(coverage.60kb$chr=="chrX"),]
chrY.60kb <- coverage.60kb[which(coverage.60kb$chr=="chrY"),]

#Get the coverage for .70kb for all the chromosomes
chr1.70kb <- coverage.70kb[which(coverage.70kb$chr=="chr1"),]
chr2.70kb <- coverage.70kb[which(coverage.70kb$chr=="chr2"),]
chr3.70kb <- coverage.70kb[which(coverage.70kb$chr=="chr3"),]
chr4.70kb <- coverage.70kb[which(coverage.70kb$chr=="chr4"),]
chr5.70kb <- coverage.70kb[which(coverage.70kb$chr=="chr5"),]
chr6.70kb <- coverage.70kb[which(coverage.70kb$chr=="chr6"),]
chr7.70kb <- coverage.70kb[which(coverage.70kb$chr=="chr7"),]
chr8.70kb <- coverage.70kb[which(coverage.70kb$chr=="chr8"),]
chr9.70kb <- coverage.70kb[which(coverage.70kb$chr=="chr9"),]
chr10.70kb <- coverage.70kb[which(coverage.70kb$chr=="chr10"),]
chr11.70kb <- coverage.70kb[which(coverage.70kb$chr=="chr11"),]
chr12.70kb <- coverage.70kb[which(coverage.70kb$chr=="chr12"),]
chr13.70kb <- coverage.70kb[which(coverage.70kb$chr=="chr13"),]
chr14.70kb <- coverage.70kb[which(coverage.70kb$chr=="chr14"),]
chr15.70kb <- coverage.70kb[which(coverage.70kb$chr=="chr15"),]
chr16.70kb <- coverage.70kb[which(coverage.70kb$chr=="chr16"),]
chr17.70kb <- coverage.70kb[which(coverage.70kb$chr=="chr17"),]
chr18.70kb <- coverage.70kb[which(coverage.70kb$chr=="chr18"),]
chr19.70kb <- coverage.70kb[which(coverage.70kb$chr=="chr19"),]
chr20.70kb <- coverage.70kb[which(coverage.70kb$chr=="chr20"),]
chr21.70kb <- coverage.70kb[which(coverage.70kb$chr=="chr21"),]
chr22.70kb <- coverage.70kb[which(coverage.70kb$chr=="chr22"),]
chrX.70kb <- coverage.70kb[which(coverage.70kb$chr=="chrX"),]
chrY.70kb <- coverage.70kb[which(coverage.70kb$chr=="chrY"),]

#Get the coverage for .80kb for all the chromosomes
chr1.80kb <- coverage.80kb[which(coverage.80kb$chr=="chr1"),]
chr2.80kb <- coverage.80kb[which(coverage.80kb$chr=="chr2"),]
chr3.80kb <- coverage.80kb[which(coverage.80kb$chr=="chr3"),]
chr4.80kb <- coverage.80kb[which(coverage.80kb$chr=="chr4"),]
chr5.80kb <- coverage.80kb[which(coverage.80kb$chr=="chr5"),]
chr6.80kb <- coverage.80kb[which(coverage.80kb$chr=="chr6"),]
chr7.80kb <- coverage.80kb[which(coverage.80kb$chr=="chr7"),]
chr8.80kb <- coverage.80kb[which(coverage.80kb$chr=="chr8"),]
chr9.80kb <- coverage.80kb[which(coverage.80kb$chr=="chr9"),]
chr10.80kb <- coverage.80kb[which(coverage.80kb$chr=="chr10"),]
chr11.80kb <- coverage.80kb[which(coverage.80kb$chr=="chr11"),]
chr12.80kb <- coverage.80kb[which(coverage.80kb$chr=="chr12"),]
chr13.80kb <- coverage.80kb[which(coverage.80kb$chr=="chr13"),]
chr14.80kb <- coverage.80kb[which(coverage.80kb$chr=="chr14"),]
chr15.80kb <- coverage.80kb[which(coverage.80kb$chr=="chr15"),]
chr16.80kb <- coverage.80kb[which(coverage.80kb$chr=="chr16"),]
chr17.80kb <- coverage.80kb[which(coverage.80kb$chr=="chr17"),]
chr18.80kb <- coverage.80kb[which(coverage.80kb$chr=="chr18"),]
chr19.80kb <- coverage.80kb[which(coverage.80kb$chr=="chr19"),]
chr20.80kb <- coverage.80kb[which(coverage.80kb$chr=="chr20"),]
chr21.80kb <- coverage.80kb[which(coverage.80kb$chr=="chr21"),]
chr22.80kb <- coverage.80kb[which(coverage.80kb$chr=="chr22"),]
chrX.80kb <- coverage.80kb[which(coverage.80kb$chr=="chrX"),]
chrY.80kb <- coverage.80kb[which(coverage.80kb$chr=="chrY"),]

#Get the coverage for .90kb for all the chromosomes
chr1.90kb <- coverage.90kb[which(coverage.90kb$chr=="chr1"),]
chr2.90kb <- coverage.90kb[which(coverage.90kb$chr=="chr2"),]
chr3.90kb <- coverage.90kb[which(coverage.90kb$chr=="chr3"),]
chr4.90kb <- coverage.90kb[which(coverage.90kb$chr=="chr4"),]
chr5.90kb <- coverage.90kb[which(coverage.90kb$chr=="chr5"),]
chr6.90kb <- coverage.90kb[which(coverage.90kb$chr=="chr6"),]
chr7.90kb <- coverage.90kb[which(coverage.90kb$chr=="chr7"),]
chr8.90kb <- coverage.90kb[which(coverage.90kb$chr=="chr8"),]
chr9.90kb <- coverage.90kb[which(coverage.90kb$chr=="chr9"),]
chr10.90kb <- coverage.90kb[which(coverage.90kb$chr=="chr10"),]
chr11.90kb <- coverage.90kb[which(coverage.90kb$chr=="chr11"),]
chr12.90kb <- coverage.90kb[which(coverage.90kb$chr=="chr12"),]
chr13.90kb <- coverage.90kb[which(coverage.90kb$chr=="chr13"),]
chr14.90kb <- coverage.90kb[which(coverage.90kb$chr=="chr14"),]
chr15.90kb <- coverage.90kb[which(coverage.90kb$chr=="chr15"),]
chr16.90kb <- coverage.90kb[which(coverage.90kb$chr=="chr16"),]
chr17.90kb <- coverage.90kb[which(coverage.90kb$chr=="chr17"),]
chr18.90kb <- coverage.90kb[which(coverage.90kb$chr=="chr18"),]
chr19.90kb <- coverage.90kb[which(coverage.90kb$chr=="chr19"),]
chr20.90kb <- coverage.90kb[which(coverage.90kb$chr=="chr20"),]
chr21.90kb <- coverage.90kb[which(coverage.90kb$chr=="chr21"),]
chr22.90kb <- coverage.90kb[which(coverage.90kb$chr=="chr22"),]
chrX.90kb <- coverage.90kb[which(coverage.90kb$chr=="chrX"),]
chrY.90kb <- coverage.90kb[which(coverage.90kb$chr=="chrY"),]

#Get the coverage for .100kb for all the chromosomes
chr1.100kb <- coverage.100kb[which(coverage.100kb$chr=="chr1"),]
chr2.100kb <- coverage.100kb[which(coverage.100kb$chr=="chr2"),]
chr3.100kb <- coverage.100kb[which(coverage.100kb$chr=="chr3"),]
chr4.100kb <- coverage.100kb[which(coverage.100kb$chr=="chr4"),]
chr5.100kb <- coverage.100kb[which(coverage.100kb$chr=="chr5"),]
chr6.100kb <- coverage.100kb[which(coverage.100kb$chr=="chr6"),]
chr7.100kb <- coverage.100kb[which(coverage.100kb$chr=="chr7"),]
chr8.100kb <- coverage.100kb[which(coverage.100kb$chr=="chr8"),]
chr9.100kb <- coverage.100kb[which(coverage.100kb$chr=="chr9"),]
chr10.100kb <- coverage.100kb[which(coverage.100kb$chr=="chr10"),]
chr11.100kb <- coverage.100kb[which(coverage.100kb$chr=="chr11"),]
chr12.100kb <- coverage.100kb[which(coverage.100kb$chr=="chr12"),]
chr13.100kb <- coverage.100kb[which(coverage.100kb$chr=="chr13"),]
chr14.100kb <- coverage.100kb[which(coverage.100kb$chr=="chr14"),]
chr15.100kb <- coverage.100kb[which(coverage.100kb$chr=="chr15"),]
chr16.100kb <- coverage.100kb[which(coverage.100kb$chr=="chr16"),]
chr17.100kb <- coverage.100kb[which(coverage.100kb$chr=="chr17"),]
chr18.100kb <- coverage.100kb[which(coverage.100kb$chr=="chr18"),]
chr19.100kb <- coverage.100kb[which(coverage.100kb$chr=="chr19"),]
chr20.100kb <- coverage.100kb[which(coverage.100kb$chr=="chr20"),]
chr21.100kb <- coverage.100kb[which(coverage.100kb$chr=="chr21"),]
chr22.100kb <- coverage.100kb[which(coverage.100kb$chr=="chr22"),]
chrX.100kb <- coverage.100kb[which(coverage.100kb$chr=="chrX"),]
chrY.100kb <- coverage.100kb[which(coverage.100kb$chr=="chrY"),]

#Get the coverage for .125kb for all the chromosomes
chr1.125kb <- coverage.125kb[which(coverage.125kb$chr=="chr1"),]
chr2.125kb <- coverage.125kb[which(coverage.125kb$chr=="chr2"),]
chr3.125kb <- coverage.125kb[which(coverage.125kb$chr=="chr3"),]
chr4.125kb <- coverage.125kb[which(coverage.125kb$chr=="chr4"),]
chr5.125kb <- coverage.125kb[which(coverage.125kb$chr=="chr5"),]
chr6.125kb <- coverage.125kb[which(coverage.125kb$chr=="chr6"),]
chr7.125kb <- coverage.125kb[which(coverage.125kb$chr=="chr7"),]
chr8.125kb <- coverage.125kb[which(coverage.125kb$chr=="chr8"),]
chr9.125kb <- coverage.125kb[which(coverage.125kb$chr=="chr9"),]
chr10.125kb <- coverage.125kb[which(coverage.125kb$chr=="chr10"),]
chr11.125kb <- coverage.125kb[which(coverage.125kb$chr=="chr11"),]
chr12.125kb <- coverage.125kb[which(coverage.125kb$chr=="chr12"),]
chr13.125kb <- coverage.125kb[which(coverage.125kb$chr=="chr13"),]
chr14.125kb <- coverage.125kb[which(coverage.125kb$chr=="chr14"),]
chr15.125kb <- coverage.125kb[which(coverage.125kb$chr=="chr15"),]
chr16.125kb <- coverage.125kb[which(coverage.125kb$chr=="chr16"),]
chr17.125kb <- coverage.125kb[which(coverage.125kb$chr=="chr17"),]
chr18.125kb <- coverage.125kb[which(coverage.125kb$chr=="chr18"),]
chr19.125kb <- coverage.125kb[which(coverage.125kb$chr=="chr19"),]
chr20.125kb <- coverage.125kb[which(coverage.125kb$chr=="chr20"),]
chr21.125kb <- coverage.125kb[which(coverage.125kb$chr=="chr21"),]
chr22.125kb <- coverage.125kb[which(coverage.125kb$chr=="chr22"),]
chrX.125kb <- coverage.125kb[which(coverage.125kb$chr=="chrX"),]
chrY.125kb <- coverage.125kb[which(coverage.125kb$chr=="chrY"),]

#Get the coverage for .150kb for all the chromosomes
chr1.150kb <- coverage.150kb[which(coverage.150kb$chr=="chr1"),]
chr2.150kb <- coverage.150kb[which(coverage.150kb$chr=="chr2"),]
chr3.150kb <- coverage.150kb[which(coverage.150kb$chr=="chr3"),]
chr4.150kb <- coverage.150kb[which(coverage.150kb$chr=="chr4"),]
chr5.150kb <- coverage.150kb[which(coverage.150kb$chr=="chr5"),]
chr6.150kb <- coverage.150kb[which(coverage.150kb$chr=="chr6"),]
chr7.150kb <- coverage.150kb[which(coverage.150kb$chr=="chr7"),]
chr8.150kb <- coverage.150kb[which(coverage.150kb$chr=="chr8"),]
chr9.150kb <- coverage.150kb[which(coverage.150kb$chr=="chr9"),]
chr10.150kb <- coverage.150kb[which(coverage.150kb$chr=="chr10"),]
chr11.150kb <- coverage.150kb[which(coverage.150kb$chr=="chr11"),]
chr12.150kb <- coverage.150kb[which(coverage.150kb$chr=="chr12"),]
chr13.150kb <- coverage.150kb[which(coverage.150kb$chr=="chr13"),]
chr14.150kb <- coverage.150kb[which(coverage.150kb$chr=="chr14"),]
chr15.150kb <- coverage.150kb[which(coverage.150kb$chr=="chr15"),]
chr16.150kb <- coverage.150kb[which(coverage.150kb$chr=="chr16"),]
chr17.150kb <- coverage.150kb[which(coverage.150kb$chr=="chr17"),]
chr18.150kb <- coverage.150kb[which(coverage.150kb$chr=="chr18"),]
chr19.150kb <- coverage.150kb[which(coverage.150kb$chr=="chr19"),]
chr20.150kb <- coverage.150kb[which(coverage.150kb$chr=="chr20"),]
chr21.150kb <- coverage.150kb[which(coverage.150kb$chr=="chr21"),]
chr22.150kb <- coverage.150kb[which(coverage.150kb$chr=="chr22"),]
chrX.150kb <- coverage.150kb[which(coverage.150kb$chr=="chrX"),]
chrY.150kb <- coverage.150kb[which(coverage.150kb$chr=="chrY"),]

#Get the coverage for .200kb for all the chromosomes
chr1.200kb <- coverage.200kb[which(coverage.200kb$chr=="chr1"),]
chr2.200kb <- coverage.200kb[which(coverage.200kb$chr=="chr2"),]
chr3.200kb <- coverage.200kb[which(coverage.200kb$chr=="chr3"),]
chr4.200kb <- coverage.200kb[which(coverage.200kb$chr=="chr4"),]
chr5.200kb <- coverage.200kb[which(coverage.200kb$chr=="chr5"),]
chr6.200kb <- coverage.200kb[which(coverage.200kb$chr=="chr6"),]
chr7.200kb <- coverage.200kb[which(coverage.200kb$chr=="chr7"),]
chr8.200kb <- coverage.200kb[which(coverage.200kb$chr=="chr8"),]
chr9.200kb <- coverage.200kb[which(coverage.200kb$chr=="chr9"),]
chr10.200kb <- coverage.200kb[which(coverage.200kb$chr=="chr10"),]
chr11.200kb <- coverage.200kb[which(coverage.200kb$chr=="chr11"),]
chr12.200kb <- coverage.200kb[which(coverage.200kb$chr=="chr12"),]
chr13.200kb <- coverage.200kb[which(coverage.200kb$chr=="chr13"),]
chr14.200kb <- coverage.200kb[which(coverage.200kb$chr=="chr14"),]
chr15.200kb <- coverage.200kb[which(coverage.200kb$chr=="chr15"),]
chr16.200kb <- coverage.200kb[which(coverage.200kb$chr=="chr16"),]
chr17.200kb <- coverage.200kb[which(coverage.200kb$chr=="chr17"),]
chr18.200kb <- coverage.200kb[which(coverage.200kb$chr=="chr18"),]
chr19.200kb <- coverage.200kb[which(coverage.200kb$chr=="chr19"),]
chr20.200kb <- coverage.200kb[which(coverage.200kb$chr=="chr20"),]
chr21.200kb <- coverage.200kb[which(coverage.200kb$chr=="chr21"),]
chr22.200kb <- coverage.200kb[which(coverage.200kb$chr=="chr22"),]
chrX.200kb <- coverage.200kb[which(coverage.200kb$chr=="chrX"),]
chrY.200kb <- coverage.200kb[which(coverage.200kb$chr=="chrY"),]

#Get the tabulation DFs with basic count and bin factors for all the different kb "locus" sizes
tab_df.1.10kb <- full.chrom.tabulator(chr1.10kb, pos_df.1, 0, 400000, 10000, "chr1")
tab_df.2.10kb <- full.chrom.tabulator(chr2.10kb, pos_df.2, 0, 400000, 10000, "chr2")
tab_df.3.10kb <- full.chrom.tabulator(chr3.10kb, pos_df.3, 0, 400000, 10000, "chr3")
tab_df.4.10kb <- full.chrom.tabulator(chr4.10kb, pos_df.4, 0, 400000, 10000, "chr4")
tab_df.5.10kb <- full.chrom.tabulator(chr5.10kb, pos_df.5, 0, 400000, 10000, "chr5")
tab_df.6.10kb <- full.chrom.tabulator(chr6.10kb, pos_df.6, 0, 400000, 10000, "chr6")
tab_df.7.10kb <- full.chrom.tabulator(chr7.10kb, pos_df.7, 0, 400000, 10000, "chr7")
tab_df.8.10kb <- full.chrom.tabulator(chr8.10kb, pos_df.8, 0, 400000, 10000, "chr8")
tab_df.9.10kb <- full.chrom.tabulator(chr9.10kb, pos_df.9, 0, 400000, 10000, "chr9")
tab_df.10.10kb <- full.chrom.tabulator(chr10.10kb, pos_df.10, 0, 400000, 10000, "chr10")
tab_df.11.10kb <- full.chrom.tabulator(chr11.10kb, pos_df.11, 0, 400000, 10000, "chr11")
tab_df.12.10kb <- full.chrom.tabulator(chr12.10kb, pos_df.12, 0, 400000, 10000, "chr12")
tab_df.13.10kb <- full.chrom.tabulator(chr13.10kb, pos_df.13, 0, 400000, 10000, "chr13")
tab_df.14.10kb <- full.chrom.tabulator(chr14.10kb, pos_df.14, 0, 400000, 10000, "chr14")
tab_df.15.10kb <- full.chrom.tabulator(chr15.10kb, pos_df.15, 0, 400000, 10000, "chr15")
tab_df.16.10kb <- full.chrom.tabulator(chr16.10kb, pos_df.16, 0, 400000, 10000, "chr16")
tab_df.17.10kb <- full.chrom.tabulator(chr17.10kb, pos_df.17, 0, 400000, 10000, "chr17")
tab_df.18.10kb <- full.chrom.tabulator(chr18.10kb, pos_df.18, 0, 400000, 10000, "chr18")
tab_df.19.10kb <- full.chrom.tabulator(chr19.10kb, pos_df.19, 0, 400000, 10000, "chr19")
tab_df.20.10kb <- full.chrom.tabulator(chr20.10kb, pos_df.20, 0, 400000, 10000, "chr20")
tab_df.21.10kb <- full.chrom.tabulator(chr21.10kb, pos_df.21, 0, 400000, 10000, "chr21")
tab_df.22.10kb <- full.chrom.tabulator(chr22.10kb, pos_df.22, 0, 400000, 10000, "chr22")
tab_df.X.10kb <- full.chrom.tabulator(chrX.10kb, pos_df.X, 0, 400000, 10000, "chrX")
tab_df.Y.10kb <- full.chrom.tabulator(chrY.10kb, pos_df.Y, 0, 400000, 10000, "chrY")

tab_df.10kb.full <- rbind.fill(tab_df.1.10kb, tab_df.2.10kb, tab_df.3.10kb, tab_df.4.10kb, tab_df.5.10kb, tab_df.6.10kb, tab_df.7.10kb, tab_df.8.10kb, tab_df.9.10kb, tab_df.10.10kb, tab_df.11.10kb, tab_df.12.10kb, tab_df.13.10kb, tab_df.14.10kb, tab_df.15.10kb, tab_df.16.10kb, tab_df.17.10kb, tab_df.18.10kb, tab_df.19.10kb, tab_df.20.10kb, tab_df.21.10kb, tab_df.22.10kb, tab_df.X.10kb, tab_df.Y.10kb)
write.table(tab_df.10kb.full, file="/Users/ittaieres/Desktop/Hi-C/All_good_analysis/full_data_frames/full_data.10kb", sep="\t", col.names=T, row.names=F, quote=F)

#Get the tabulation DFs with basic count and bin factors for all the different kb "locus" sizes
tab_df.1.20kb <- full.chrom.tabulator(chr1.20kb, pos_df.1, 0, 400000, 10000, "chr1")
tab_df.2.20kb <- full.chrom.tabulator(chr2.20kb, pos_df.2, 0, 400000, 10000, "chr2")
tab_df.3.20kb <- full.chrom.tabulator(chr3.20kb, pos_df.3, 0, 400000, 10000, "chr3")
tab_df.4.20kb <- full.chrom.tabulator(chr4.20kb, pos_df.4, 0, 400000, 10000, "chr4")
tab_df.5.20kb <- full.chrom.tabulator(chr5.20kb, pos_df.5, 0, 400000, 10000, "chr5")
tab_df.6.20kb <- full.chrom.tabulator(chr6.20kb, pos_df.6, 0, 400000, 10000, "chr6")
tab_df.7.20kb <- full.chrom.tabulator(chr7.20kb, pos_df.7, 0, 400000, 10000, "chr7")
tab_df.8.20kb <- full.chrom.tabulator(chr8.20kb, pos_df.8, 0, 400000, 10000, "chr8")
tab_df.9.20kb <- full.chrom.tabulator(chr9.20kb, pos_df.9, 0, 400000, 10000, "chr9")
tab_df.10.20kb <- full.chrom.tabulator(chr10.20kb, pos_df.10, 0, 400000, 10000, "chr10")
tab_df.11.20kb <- full.chrom.tabulator(chr11.20kb, pos_df.11, 0, 400000, 10000, "chr11")
tab_df.12.20kb <- full.chrom.tabulator(chr12.20kb, pos_df.12, 0, 400000, 10000, "chr12")
tab_df.13.20kb <- full.chrom.tabulator(chr13.20kb, pos_df.13, 0, 400000, 10000, "chr13")
tab_df.14.20kb <- full.chrom.tabulator(chr14.20kb, pos_df.14, 0, 400000, 10000, "chr14")
tab_df.15.20kb <- full.chrom.tabulator(chr15.20kb, pos_df.15, 0, 400000, 10000, "chr15")
tab_df.16.20kb <- full.chrom.tabulator(chr16.20kb, pos_df.16, 0, 400000, 10000, "chr16")
tab_df.17.20kb <- full.chrom.tabulator(chr17.20kb, pos_df.17, 0, 400000, 10000, "chr17")
tab_df.18.20kb <- full.chrom.tabulator(chr18.20kb, pos_df.18, 0, 400000, 10000, "chr18")
tab_df.19.20kb <- full.chrom.tabulator(chr19.20kb, pos_df.19, 0, 400000, 10000, "chr19")
tab_df.20.20kb <- full.chrom.tabulator(chr20.20kb, pos_df.20, 0, 400000, 10000, "chr20")
tab_df.21.20kb <- full.chrom.tabulator(chr21.20kb, pos_df.21, 0, 400000, 10000, "chr21")
tab_df.22.20kb <- full.chrom.tabulator(chr22.20kb, pos_df.22, 0, 400000, 10000, "chr22")
tab_df.X.20kb <- full.chrom.tabulator(chrX.20kb, pos_df.X, 0, 400000, 10000, "chrX")
tab_df.Y.20kb <- full.chrom.tabulator(chrY.20kb, pos_df.Y, 0, 400000, 10000, "chrY")

tab_df.20kb.full <- rbind.fill(tab_df.1.20kb, tab_df.2.20kb, tab_df.3.20kb, tab_df.4.20kb, tab_df.5.20kb, tab_df.6.20kb, tab_df.7.20kb, tab_df.8.20kb, tab_df.9.20kb, tab_df.10.20kb, tab_df.11.20kb, tab_df.12.20kb, tab_df.13.20kb, tab_df.14.20kb, tab_df.15.20kb, tab_df.16.20kb, tab_df.17.20kb, tab_df.18.20kb, tab_df.19.20kb, tab_df.20.20kb, tab_df.21.20kb, tab_df.22.20kb, tab_df.X.20kb, tab_df.Y.20kb)
write.table(tab_df.20kb.full, file="/Users/ittaieres/Desktop/Hi-C/All_good_analysis/full_data_frames/full_data.20kb", sep="\t", col.names=T, row.names=F, quote=F)

#Get the tabulation DFs with basic count and bin factors for all the different kb "locus" sizes
tab_df.1.30kb <- full.chrom.tabulator(chr1.30kb, pos_df.1, 0, 400000, 10000, "chr1")
tab_df.2.30kb <- full.chrom.tabulator(chr2.30kb, pos_df.2, 0, 400000, 10000, "chr2")
tab_df.3.30kb <- full.chrom.tabulator(chr3.30kb, pos_df.3, 0, 400000, 10000, "chr3")
tab_df.4.30kb <- full.chrom.tabulator(chr4.30kb, pos_df.4, 0, 400000, 10000, "chr4")
tab_df.5.30kb <- full.chrom.tabulator(chr5.30kb, pos_df.5, 0, 400000, 10000, "chr5")
tab_df.6.30kb <- full.chrom.tabulator(chr6.30kb, pos_df.6, 0, 400000, 10000, "chr6")
tab_df.7.30kb <- full.chrom.tabulator(chr7.30kb, pos_df.7, 0, 400000, 10000, "chr7")
tab_df.8.30kb <- full.chrom.tabulator(chr8.30kb, pos_df.8, 0, 400000, 10000, "chr8")
tab_df.9.30kb <- full.chrom.tabulator(chr9.30kb, pos_df.9, 0, 400000, 10000, "chr9")
tab_df.10.30kb <- full.chrom.tabulator(chr10.30kb, pos_df.10, 0, 400000, 10000, "chr10")
tab_df.11.30kb <- full.chrom.tabulator(chr11.30kb, pos_df.11, 0, 400000, 10000, "chr11")
tab_df.12.30kb <- full.chrom.tabulator(chr12.30kb, pos_df.12, 0, 400000, 10000, "chr12")
tab_df.13.30kb <- full.chrom.tabulator(chr13.30kb, pos_df.13, 0, 400000, 10000, "chr13")
tab_df.14.30kb <- full.chrom.tabulator(chr14.30kb, pos_df.14, 0, 400000, 10000, "chr14")
tab_df.15.30kb <- full.chrom.tabulator(chr15.30kb, pos_df.15, 0, 400000, 10000, "chr15")
tab_df.16.30kb <- full.chrom.tabulator(chr16.30kb, pos_df.16, 0, 400000, 10000, "chr16")
tab_df.17.30kb <- full.chrom.tabulator(chr17.30kb, pos_df.17, 0, 400000, 10000, "chr17")
tab_df.18.30kb <- full.chrom.tabulator(chr18.30kb, pos_df.18, 0, 400000, 10000, "chr18")
tab_df.19.30kb <- full.chrom.tabulator(chr19.30kb, pos_df.19, 0, 400000, 10000, "chr19")
tab_df.20.30kb <- full.chrom.tabulator(chr20.30kb, pos_df.20, 0, 400000, 10000, "chr20")
tab_df.21.30kb <- full.chrom.tabulator(chr21.30kb, pos_df.21, 0, 400000, 10000, "chr21")
tab_df.22.30kb <- full.chrom.tabulator(chr22.30kb, pos_df.22, 0, 400000, 10000, "chr22")
tab_df.X.30kb <- full.chrom.tabulator(chrX.30kb, pos_df.X, 0, 400000, 10000, "chrX")
tab_df.Y.30kb <- full.chrom.tabulator(chrY.30kb, pos_df.Y, 0, 400000, 10000, "chrY")

tab_df.30kb.full <- rbind.fill(tab_df.1.30kb, tab_df.2.30kb, tab_df.3.30kb, tab_df.4.30kb, tab_df.5.30kb, tab_df.6.30kb, tab_df.7.30kb, tab_df.8.30kb, tab_df.9.30kb, tab_df.10.30kb, tab_df.11.30kb, tab_df.12.30kb, tab_df.13.30kb, tab_df.14.30kb, tab_df.15.30kb, tab_df.16.30kb, tab_df.17.30kb, tab_df.18.30kb, tab_df.19.30kb, tab_df.20.30kb, tab_df.21.30kb, tab_df.22.30kb, tab_df.X.30kb, tab_df.Y.30kb)
write.table(tab_df.30kb.full, file="/Users/ittaieres/Desktop/Hi-C/All_good_analysis/full_data_frames/full_data.30kb", sep="\t", col.names=T, row.names=F, quote=F)


##############
#Get all the comparison dfs for all chromosomes at locus bins of .10kb

overlap.chr1.10kb <- signif.comparison("chr1_10kb_siglocs.txt", "chr1", tab_df.10kb.full, 10)
overlap.chr2.10kb <- signif.comparison("chr2_10kb_siglocs.txt", "chr2", tab_df.10kb.full, 10)
overlap.chr3.10kb <- signif.comparison("chr3_10kb_siglocs.txt", "chr3", tab_df.10kb.full, 10)
overlap.chr4.10kb <- signif.comparison("chr4_10kb_siglocs.txt", "chr4", tab_df.10kb.full, 10)
overlap.chr5.10kb <- signif.comparison("chr5_10kb_siglocs.txt", "chr5", tab_df.10kb.full, 10)
overlap.chr6.10kb <- signif.comparison("chr6_10kb_siglocs.txt", "chr6", tab_df.10kb.full, 10)
overlap.chr7.10kb <- signif.comparison("chr7_10kb_siglocs.txt", "chr7", tab_df.10kb.full, 10)
overlap.chr8.10kb <- signif.comparison("chr8_10kb_siglocs.txt", "chr8", tab_df.10kb.full, 10)
overlap.chr9.10kb <- signif.comparison("chr9_10kb_siglocs.txt", "chr9", tab_df.10kb.full, 10)
overlap.chr10.10kb <- signif.comparison("chr10_10kb_siglocs.txt", "chr10", tab_df.10kb.full, 10)
overlap.chr11.10kb <- signif.comparison("chr11_10kb_siglocs.txt", "chr11", tab_df.10kb.full, 10)
overlap.chr12.10kb <- signif.comparison("chr12_10kb_siglocs.txt", "chr12", tab_df.10kb.full, 10)
overlap.chr13.10kb <- signif.comparison("chr13_10kb_siglocs.txt", "chr13", tab_df.10kb.full, 10)
overlap.chr14.10kb <- signif.comparison("chr14_10kb_siglocs.txt", "chr14", tab_df.10kb.full, 10)
overlap.chr15.10kb <- signif.comparison("chr15_10kb_siglocs.txt", "chr15", tab_df.10kb.full, 10)
overlap.chr16.10kb <- signif.comparison("chr16_10kb_siglocs.txt", "chr16", tab_df.10kb.full, 10)
overlap.chr17.10kb <- signif.comparison("chr17_10kb_siglocs.txt", "chr17", tab_df.10kb.full, 10)
overlap.chr18.10kb <- signif.comparison("chr18_10kb_siglocs.txt", "chr18", tab_df.10kb.full, 10)
overlap.chr19.10kb <- signif.comparison("chr19_10kb_siglocs.txt", "chr19", tab_df.10kb.full, 10)
overlap.chr20.10kb <- signif.comparison("chr20_10kb_siglocs.txt", "chr20", tab_df.10kb.full, 10)
overlap.chr21.10kb <- signif.comparison("chr21_10kb_siglocs.txt", "chr21", tab_df.10kb.full, 10)
overlap.chr22.10kb <- signif.comparison("chr22_10kb_siglocs.txt", "chr22", tab_df.10kb.full, 10)
overlap.chrX.10kb <- signif.comparison("chrX_10kb_siglocs.txt", "chrX", tab_df.10kb.full, 10)
overlap.chrY.10kb <- signif.comparison("chrY_10kb_siglocs.txt", "chrY", tab_df.10kb.full, 10)

overlap.10kb.GW <- rbind.fill(overlap.chr1.10kb, overlap.chr2.10kb, overlap.chr3.10kb, overlap.chr4.10kb, overlap.chr5.10kb, overlap.chr6.10kb, overlap.chr7.10kb, overlap.chr8.10kb, overlap.chr9.10kb, overlap.chr10.10kb, overlap.chr11.10kb, overlap.chr12.10kb, overlap.chr13.10kb, overlap.chr14.10kb, overlap.chr15.10kb, overlap.chr16.10kb, overlap.chr17.10kb, overlap.chr18.10kb, overlap.chr19.10kb, overlap.chr20.10kb, overlap.chr21.10kb, overlap.chr22.10kb, overlap.chrX.10kb, overlap.chrY.10kb)
write.table(overlap.10kb.GW, file="/Users/ittaieres/Desktop/Hi-C/All_good_analysis/signif_comparisons/full_signif.10kb", sep="\t", col.names=T, row.names=F, quote=F)

#Get all the comparison dfs for all chromosomes at locus bins of .20kb

overlap.chr1.20kb <- signif.comparison("chr1_20kb_siglocs.txt", "chr1", tab_df.20kb.full, 20)
overlap.chr2.20kb <- signif.comparison("chr2_20kb_siglocs.txt", "chr2", tab_df.20kb.full, 20)
overlap.chr3.20kb <- signif.comparison("chr3_20kb_siglocs.txt", "chr3", tab_df.20kb.full, 20)
overlap.chr4.20kb <- signif.comparison("chr4_20kb_siglocs.txt", "chr4", tab_df.20kb.full, 20)
overlap.chr5.20kb <- signif.comparison("chr5_20kb_siglocs.txt", "chr5", tab_df.20kb.full, 20)
overlap.chr6.20kb <- signif.comparison("chr6_20kb_siglocs.txt", "chr6", tab_df.20kb.full, 20)
overlap.chr7.20kb <- signif.comparison("chr7_20kb_siglocs.txt", "chr7", tab_df.20kb.full, 20)
overlap.chr8.20kb <- signif.comparison("chr8_20kb_siglocs.txt", "chr8", tab_df.20kb.full, 20)
overlap.chr9.20kb <- signif.comparison("chr9_20kb_siglocs.txt", "chr9", tab_df.20kb.full, 20)
overlap.chr10.20kb <- signif.comparison("chr10_20kb_siglocs.txt", "chr10", tab_df.20kb.full, 20)
overlap.chr11.20kb <- signif.comparison("chr11_20kb_siglocs.txt", "chr11", tab_df.20kb.full, 20)
overlap.chr12.20kb <- signif.comparison("chr12_20kb_siglocs.txt", "chr12", tab_df.20kb.full, 20)
overlap.chr13.20kb <- signif.comparison("chr13_20kb_siglocs.txt", "chr13", tab_df.20kb.full, 20)
overlap.chr14.20kb <- signif.comparison("chr14_20kb_siglocs.txt", "chr14", tab_df.20kb.full, 20)
overlap.chr15.20kb <- signif.comparison("chr15_20kb_siglocs.txt", "chr15", tab_df.20kb.full, 20)
overlap.chr16.20kb <- signif.comparison("chr16_20kb_siglocs.txt", "chr16", tab_df.20kb.full, 20)
overlap.chr17.20kb <- signif.comparison("chr17_20kb_siglocs.txt", "chr17", tab_df.20kb.full, 20)
overlap.chr18.20kb <- signif.comparison("chr18_20kb_siglocs.txt", "chr18", tab_df.20kb.full, 20)
overlap.chr19.20kb <- signif.comparison("chr19_20kb_siglocs.txt", "chr19", tab_df.20kb.full, 20)
overlap.chr20.20kb <- signif.comparison("chr20_20kb_siglocs.txt", "chr20", tab_df.20kb.full, 20)
overlap.chr21.20kb <- signif.comparison("chr21_20kb_siglocs.txt", "chr21", tab_df.20kb.full, 20)
overlap.chr22.20kb <- signif.comparison("chr22_20kb_siglocs.txt", "chr22", tab_df.20kb.full, 20)
overlap.chrX.20kb <- signif.comparison("chrX_20kb_siglocs.txt", "chrX", tab_df.20kb.full, 20)
overlap.chrY.20kb <- signif.comparison("chrY_20kb_siglocs.txt", "chrY", tab_df.20kb.full, 20)

overlap.20kb.GW <- rbind.fill(overlap.chr1.20kb, overlap.chr2.20kb, overlap.chr3.20kb, overlap.chr4.20kb, overlap.chr5.20kb, overlap.chr6.20kb, overlap.chr7.20kb, overlap.chr8.20kb, overlap.chr9.20kb, overlap.chr10.20kb, overlap.chr11.20kb, overlap.chr12.20kb, overlap.chr13.20kb, overlap.chr14.20kb, overlap.chr15.20kb, overlap.chr16.20kb, overlap.chr17.20kb, overlap.chr18.20kb, overlap.chr19.20kb, overlap.chr20.20kb, overlap.chr21.20kb, overlap.chr22.20kb, overlap.chrX.20kb, overlap.chrY.20kb)
write.table(overlap.20kb.GW, file="/Users/ittaieres/Desktop/Hi-C/All_good_analysis/signif_comparisons/full_signif.20kb", sep="\t", col.names=T, row.names=F, quote=F)

#Get all the comparison dfs for all chromosomes at locus bins of .30kb

overlap.chr1.30kb <- signif.comparison("chr1_30kb_siglocs.txt", "chr1", tab_df.30kb.full, 30)
overlap.chr2.30kb <- signif.comparison("chr2_30kb_siglocs.txt", "chr2", tab_df.30kb.full, 30)
overlap.chr3.30kb <- signif.comparison("chr3_30kb_siglocs.txt", "chr3", tab_df.30kb.full, 30)
overlap.chr4.30kb <- signif.comparison("chr4_30kb_siglocs.txt", "chr4", tab_df.30kb.full, 30)
overlap.chr5.30kb <- signif.comparison("chr5_30kb_siglocs.txt", "chr5", tab_df.30kb.full, 30)
overlap.chr6.30kb <- signif.comparison("chr6_30kb_siglocs.txt", "chr6", tab_df.30kb.full, 30)
overlap.chr7.30kb <- signif.comparison("chr7_30kb_siglocs.txt", "chr7", tab_df.30kb.full, 30)
overlap.chr8.30kb <- signif.comparison("chr8_30kb_siglocs.txt", "chr8", tab_df.30kb.full, 30)
overlap.chr9.30kb <- signif.comparison("chr9_30kb_siglocs.txt", "chr9", tab_df.30kb.full, 30)
overlap.chr10.30kb <- signif.comparison("chr10_30kb_siglocs.txt", "chr10", tab_df.30kb.full, 30)
overlap.chr11.30kb <- signif.comparison("chr11_30kb_siglocs.txt", "chr11", tab_df.30kb.full, 30)
overlap.chr12.30kb <- signif.comparison("chr12_30kb_siglocs.txt", "chr12", tab_df.30kb.full, 30)
overlap.chr13.30kb <- signif.comparison("chr13_30kb_siglocs.txt", "chr13", tab_df.30kb.full, 30)
overlap.chr14.30kb <- signif.comparison("chr14_30kb_siglocs.txt", "chr14", tab_df.30kb.full, 30)
overlap.chr15.30kb <- signif.comparison("chr15_30kb_siglocs.txt", "chr15", tab_df.30kb.full, 30)
overlap.chr16.30kb <- signif.comparison("chr16_30kb_siglocs.txt", "chr16", tab_df.30kb.full, 30)
overlap.chr17.30kb <- signif.comparison("chr17_30kb_siglocs.txt", "chr17", tab_df.30kb.full, 30)
overlap.chr18.30kb <- signif.comparison("chr18_30kb_siglocs.txt", "chr18", tab_df.30kb.full, 30)
overlap.chr19.30kb <- signif.comparison("chr19_30kb_siglocs.txt", "chr19", tab_df.30kb.full, 30)
overlap.chr20.30kb <- signif.comparison("chr20_30kb_siglocs.txt", "chr20", tab_df.30kb.full, 30)
overlap.chr21.30kb <- signif.comparison("chr21_30kb_siglocs.txt", "chr21", tab_df.30kb.full, 30)
overlap.chr22.30kb <- signif.comparison("chr22_30kb_siglocs.txt", "chr22", tab_df.30kb.full, 30)
overlap.chrX.30kb <- signif.comparison("chrX_30kb_siglocs.txt", "chrX", tab_df.30kb.full, 30)
overlap.chrY.30kb <- signif.comparison("chrY_30kb_siglocs.txt", "chrY", tab_df.30kb.full, 30)

overlap.30kb.GW <- rbind.fill(overlap.chr1.30kb, overlap.chr2.30kb, overlap.chr3.30kb, overlap.chr4.30kb, overlap.chr5.30kb, overlap.chr6.30kb, overlap.chr7.30kb, overlap.chr8.30kb, overlap.chr9.30kb, overlap.chr10.30kb, overlap.chr11.30kb, overlap.chr12.30kb, overlap.chr13.30kb, overlap.chr14.30kb, overlap.chr15.30kb, overlap.chr16.30kb, overlap.chr17.30kb, overlap.chr18.30kb, overlap.chr19.30kb, overlap.chr20.30kb, overlap.chr21.30kb, overlap.chr22.30kb, overlap.chrX.30kb, overlap.chrY.30kb)
write.table(overlap.30kb.GW, file="/Users/ittaieres/Desktop/Hi-C/All_good_analysis/signif_comparisons/full_signif.30kb", sep="\t", col.names=T, row.names=F, quote=F)

######################


#Get the tabulation DFs with basic count and bin factors for all the different kb "locus" sizes
tab_df.1.40kb <- full.chrom.tabulator(chr1.40kb, pos_df.1, 0, 400000, 10000, "chr1")
tab_df.2.40kb <- full.chrom.tabulator(chr2.40kb, pos_df.2, 0, 400000, 10000, "chr2")
tab_df.3.40kb <- full.chrom.tabulator(chr3.40kb, pos_df.3, 0, 400000, 10000, "chr3")
tab_df.4.40kb <- full.chrom.tabulator(chr4.40kb, pos_df.4, 0, 400000, 10000, "chr4")
tab_df.5.40kb <- full.chrom.tabulator(chr5.40kb, pos_df.5, 0, 400000, 10000, "chr5")
tab_df.6.40kb <- full.chrom.tabulator(chr6.40kb, pos_df.6, 0, 400000, 10000, "chr6")
tab_df.7.40kb <- full.chrom.tabulator(chr7.40kb, pos_df.7, 0, 400000, 10000, "chr7")
tab_df.8.40kb <- full.chrom.tabulator(chr8.40kb, pos_df.8, 0, 400000, 10000, "chr8")
tab_df.9.40kb <- full.chrom.tabulator(chr9.40kb, pos_df.9, 0, 400000, 10000, "chr9")
tab_df.10.40kb <- full.chrom.tabulator(chr10.40kb, pos_df.10, 0, 400000, 10000, "chr10")
tab_df.11.40kb <- full.chrom.tabulator(chr11.40kb, pos_df.11, 0, 400000, 10000, "chr11")
tab_df.12.40kb <- full.chrom.tabulator(chr12.40kb, pos_df.12, 0, 400000, 10000, "chr12")
tab_df.13.40kb <- full.chrom.tabulator(chr13.40kb, pos_df.13, 0, 400000, 10000, "chr13")
tab_df.14.40kb <- full.chrom.tabulator(chr14.40kb, pos_df.14, 0, 400000, 10000, "chr14")
tab_df.15.40kb <- full.chrom.tabulator(chr15.40kb, pos_df.15, 0, 400000, 10000, "chr15")
tab_df.16.40kb <- full.chrom.tabulator(chr16.40kb, pos_df.16, 0, 400000, 10000, "chr16")
tab_df.17.40kb <- full.chrom.tabulator(chr17.40kb, pos_df.17, 0, 400000, 10000, "chr17")
tab_df.18.40kb <- full.chrom.tabulator(chr18.40kb, pos_df.18, 0, 400000, 10000, "chr18")
tab_df.19.40kb <- full.chrom.tabulator(chr19.40kb, pos_df.19, 0, 400000, 10000, "chr19")
tab_df.20.40kb <- full.chrom.tabulator(chr20.40kb, pos_df.20, 0, 400000, 10000, "chr20")
tab_df.21.40kb <- full.chrom.tabulator(chr21.40kb, pos_df.21, 0, 400000, 10000, "chr21")
tab_df.22.40kb <- full.chrom.tabulator(chr22.40kb, pos_df.22, 0, 400000, 10000, "chr22")
tab_df.X.40kb <- full.chrom.tabulator(chrX.40kb, pos_df.X, 0, 400000, 10000, "chrX")
tab_df.Y.40kb <- full.chrom.tabulator(chrY.40kb, pos_df.Y, 0, 400000, 10000, "chrY")

tab_df.40kb.full <- rbind.fill(tab_df.1.40kb, tab_df.2.40kb, tab_df.3.40kb, tab_df.4.40kb, tab_df.5.40kb, tab_df.6.40kb, tab_df.7.40kb, tab_df.8.40kb, tab_df.9.40kb, tab_df.10.40kb, tab_df.11.40kb, tab_df.12.40kb, tab_df.13.40kb, tab_df.14.40kb, tab_df.15.40kb, tab_df.16.40kb, tab_df.17.40kb, tab_df.18.40kb, tab_df.19.40kb, tab_df.20.40kb, tab_df.21.40kb, tab_df.22.40kb, tab_df.X.40kb, tab_df.Y.40kb)
write.table(tab_df.40kb.full, file="/Users/ittaieres/Desktop/Hi-C/All_good_analysis/full_data_frames/full_data.40kb", sep="\t", col.names=T, row.names=F, quote=F)

#Get all the comparison dfs for all chromosomes at locus bins of .40kb

overlap.chr1.40kb <- signif.comparison("chr1_40kb_siglocs.txt", "chr1", tab_df.40kb.full, 40)
overlap.chr2.40kb <- signif.comparison("chr2_40kb_siglocs.txt", "chr2", tab_df.40kb.full, 40)
overlap.chr3.40kb <- signif.comparison("chr3_40kb_siglocs.txt", "chr3", tab_df.40kb.full, 40)
overlap.chr4.40kb <- signif.comparison("chr4_40kb_siglocs.txt", "chr4", tab_df.40kb.full, 40)
overlap.chr5.40kb <- signif.comparison("chr5_40kb_siglocs.txt", "chr5", tab_df.40kb.full, 40)
overlap.chr6.40kb <- signif.comparison("chr6_40kb_siglocs.txt", "chr6", tab_df.40kb.full, 40)
overlap.chr7.40kb <- signif.comparison("chr7_40kb_siglocs.txt", "chr7", tab_df.40kb.full, 40)
overlap.chr8.40kb <- signif.comparison("chr8_40kb_siglocs.txt", "chr8", tab_df.40kb.full, 40)
overlap.chr9.40kb <- signif.comparison("chr9_40kb_siglocs.txt", "chr9", tab_df.40kb.full, 40)
overlap.chr10.40kb <- signif.comparison("chr10_40kb_siglocs.txt", "chr10", tab_df.40kb.full, 40)
overlap.chr11.40kb <- signif.comparison("chr11_40kb_siglocs.txt", "chr11", tab_df.40kb.full, 40)
overlap.chr12.40kb <- signif.comparison("chr12_40kb_siglocs.txt", "chr12", tab_df.40kb.full, 40)
overlap.chr13.40kb <- signif.comparison("chr13_40kb_siglocs.txt", "chr13", tab_df.40kb.full, 40)
overlap.chr14.40kb <- signif.comparison("chr14_40kb_siglocs.txt", "chr14", tab_df.40kb.full, 40)
overlap.chr15.40kb <- signif.comparison("chr15_40kb_siglocs.txt", "chr15", tab_df.40kb.full, 40)
overlap.chr16.40kb <- signif.comparison("chr16_40kb_siglocs.txt", "chr16", tab_df.40kb.full, 40)
overlap.chr17.40kb <- signif.comparison("chr17_40kb_siglocs.txt", "chr17", tab_df.40kb.full, 40)
overlap.chr18.40kb <- signif.comparison("chr18_40kb_siglocs.txt", "chr18", tab_df.40kb.full, 40)
overlap.chr19.40kb <- signif.comparison("chr19_40kb_siglocs.txt", "chr19", tab_df.40kb.full, 40)
overlap.chr20.40kb <- signif.comparison("chr20_40kb_siglocs.txt", "chr20", tab_df.40kb.full, 40)
overlap.chr21.40kb <- signif.comparison("chr21_40kb_siglocs.txt", "chr21", tab_df.40kb.full, 40)
overlap.chr22.40kb <- signif.comparison("chr22_40kb_siglocs.txt", "chr22", tab_df.40kb.full, 40)
overlap.chrX.40kb <- signif.comparison("chrX_40kb_siglocs.txt", "chrX", tab_df.40kb.full, 40)
overlap.chrY.40kb <- signif.comparison("chrY_40kb_siglocs.txt", "chrY", tab_df.40kb.full, 40)

overlap.40kb.GW <- rbind.fill(overlap.chr1.40kb, overlap.chr2.40kb, overlap.chr3.40kb, overlap.chr4.40kb, overlap.chr5.40kb, overlap.chr6.40kb, overlap.chr7.40kb, overlap.chr8.40kb, overlap.chr9.40kb, overlap.chr10.40kb, overlap.chr11.40kb, overlap.chr12.40kb, overlap.chr13.40kb, overlap.chr14.40kb, overlap.chr15.40kb, overlap.chr16.40kb, overlap.chr17.40kb, overlap.chr18.40kb, overlap.chr19.40kb, overlap.chr20.40kb, overlap.chr21.40kb, overlap.chr22.40kb, overlap.chrX.40kb, overlap.chrY.40kb)
write.table(overlap.40kb.GW, file="/Users/ittaieres/Desktop/Hi-C/All_good_analysis/signif_comparisons/full_signif.40kb", sep="\t", col.names=T, row.names=F, quote=F)

#Get the tabulation DFs with basic count and bin factors for all the different kb "locus" sizes
tab_df.1.50kb <- full.chrom.tabulator(chr1.50kb, pos_df.1, 0, 400000, 10000, "chr1")
tab_df.2.50kb <- full.chrom.tabulator(chr2.50kb, pos_df.2, 0, 400000, 10000, "chr2")
tab_df.3.50kb <- full.chrom.tabulator(chr3.50kb, pos_df.3, 0, 400000, 10000, "chr3")
tab_df.4.50kb <- full.chrom.tabulator(chr4.50kb, pos_df.4, 0, 400000, 10000, "chr4")
tab_df.5.50kb <- full.chrom.tabulator(chr5.50kb, pos_df.5, 0, 400000, 10000, "chr5")
tab_df.6.50kb <- full.chrom.tabulator(chr6.50kb, pos_df.6, 0, 400000, 10000, "chr6")
tab_df.7.50kb <- full.chrom.tabulator(chr7.50kb, pos_df.7, 0, 400000, 10000, "chr7")
tab_df.8.50kb <- full.chrom.tabulator(chr8.50kb, pos_df.8, 0, 400000, 10000, "chr8")
tab_df.9.50kb <- full.chrom.tabulator(chr9.50kb, pos_df.9, 0, 400000, 10000, "chr9")
tab_df.10.50kb <- full.chrom.tabulator(chr10.50kb, pos_df.10, 0, 400000, 10000, "chr10")
tab_df.11.50kb <- full.chrom.tabulator(chr11.50kb, pos_df.11, 0, 400000, 10000, "chr11")
tab_df.12.50kb <- full.chrom.tabulator(chr12.50kb, pos_df.12, 0, 400000, 10000, "chr12")
tab_df.13.50kb <- full.chrom.tabulator(chr13.50kb, pos_df.13, 0, 400000, 10000, "chr13")
tab_df.14.50kb <- full.chrom.tabulator(chr14.50kb, pos_df.14, 0, 400000, 10000, "chr14")
tab_df.15.50kb <- full.chrom.tabulator(chr15.50kb, pos_df.15, 0, 400000, 10000, "chr15")
tab_df.16.50kb <- full.chrom.tabulator(chr16.50kb, pos_df.16, 0, 400000, 10000, "chr16")
tab_df.17.50kb <- full.chrom.tabulator(chr17.50kb, pos_df.17, 0, 400000, 10000, "chr17")
tab_df.18.50kb <- full.chrom.tabulator(chr18.50kb, pos_df.18, 0, 400000, 10000, "chr18")
tab_df.19.50kb <- full.chrom.tabulator(chr19.50kb, pos_df.19, 0, 400000, 10000, "chr19")
tab_df.20.50kb <- full.chrom.tabulator(chr20.50kb, pos_df.20, 0, 400000, 10000, "chr20")
tab_df.21.50kb <- full.chrom.tabulator(chr21.50kb, pos_df.21, 0, 400000, 10000, "chr21")
tab_df.22.50kb <- full.chrom.tabulator(chr22.50kb, pos_df.22, 0, 400000, 10000, "chr22")
tab_df.X.50kb <- full.chrom.tabulator(chrX.50kb, pos_df.X, 0, 400000, 10000, "chrX")
tab_df.Y.50kb <- full.chrom.tabulator(chrY.50kb, pos_df.Y, 0, 400000, 10000, "chrY")

tab_df.50kb.full <- rbind.fill(tab_df.1.50kb, tab_df.2.50kb, tab_df.3.50kb, tab_df.4.50kb, tab_df.5.50kb, tab_df.6.50kb, tab_df.7.50kb, tab_df.8.50kb, tab_df.9.50kb, tab_df.10.50kb, tab_df.11.50kb, tab_df.12.50kb, tab_df.13.50kb, tab_df.14.50kb, tab_df.15.50kb, tab_df.16.50kb, tab_df.17.50kb, tab_df.18.50kb, tab_df.19.50kb, tab_df.20.50kb, tab_df.21.50kb, tab_df.22.50kb, tab_df.X.50kb, tab_df.Y.50kb)
write.table(tab_df.50kb.full, file="/Users/ittaieres/Desktop/Hi-C/All_good_analysis/full_data_frames/full_data.50kb", sep="\t", col.names=T, row.names=F, quote=F)

#Get all the comparison dfs for all chromosomes at locus bins of .50kb

overlap.chr1.50kb <- signif.comparison("chr1_50kb_siglocs.txt", "chr1", tab_df.50kb.full, 50)
overlap.chr2.50kb <- signif.comparison("chr2_50kb_siglocs.txt", "chr2", tab_df.50kb.full, 50)
overlap.chr3.50kb <- signif.comparison("chr3_50kb_siglocs.txt", "chr3", tab_df.50kb.full, 50)
overlap.chr4.50kb <- signif.comparison("chr4_50kb_siglocs.txt", "chr4", tab_df.50kb.full, 50)
overlap.chr5.50kb <- signif.comparison("chr5_50kb_siglocs.txt", "chr5", tab_df.50kb.full, 50)
overlap.chr6.50kb <- signif.comparison("chr6_50kb_siglocs.txt", "chr6", tab_df.50kb.full, 50)
overlap.chr7.50kb <- signif.comparison("chr7_50kb_siglocs.txt", "chr7", tab_df.50kb.full, 50)
overlap.chr8.50kb <- signif.comparison("chr8_50kb_siglocs.txt", "chr8", tab_df.50kb.full, 50)
overlap.chr9.50kb <- signif.comparison("chr9_50kb_siglocs.txt", "chr9", tab_df.50kb.full, 50)
overlap.chr10.50kb <- signif.comparison("chr10_50kb_siglocs.txt", "chr10", tab_df.50kb.full, 50)
overlap.chr11.50kb <- signif.comparison("chr11_50kb_siglocs.txt", "chr11", tab_df.50kb.full, 50)
overlap.chr12.50kb <- signif.comparison("chr12_50kb_siglocs.txt", "chr12", tab_df.50kb.full, 50)
overlap.chr13.50kb <- signif.comparison("chr13_50kb_siglocs.txt", "chr13", tab_df.50kb.full, 50)
overlap.chr14.50kb <- signif.comparison("chr14_50kb_siglocs.txt", "chr14", tab_df.50kb.full, 50)
overlap.chr15.50kb <- signif.comparison("chr15_50kb_siglocs.txt", "chr15", tab_df.50kb.full, 50)
overlap.chr16.50kb <- signif.comparison("chr16_50kb_siglocs.txt", "chr16", tab_df.50kb.full, 50)
overlap.chr17.50kb <- signif.comparison("chr17_50kb_siglocs.txt", "chr17", tab_df.50kb.full, 50)
overlap.chr18.50kb <- signif.comparison("chr18_50kb_siglocs.txt", "chr18", tab_df.50kb.full, 50)
overlap.chr19.50kb <- signif.comparison("chr19_50kb_siglocs.txt", "chr19", tab_df.50kb.full, 50)
overlap.chr20.50kb <- signif.comparison("chr20_50kb_siglocs.txt", "chr20", tab_df.50kb.full, 50)
overlap.chr21.50kb <- signif.comparison("chr21_50kb_siglocs.txt", "chr21", tab_df.50kb.full, 50)
overlap.chr22.50kb <- signif.comparison("chr22_50kb_siglocs.txt", "chr22", tab_df.50kb.full, 50)
overlap.chrX.50kb <- signif.comparison("chrX_50kb_siglocs.txt", "chrX", tab_df.50kb.full, 50)
overlap.chrY.50kb <- signif.comparison("chrY_50kb_siglocs.txt", "chrY", tab_df.50kb.full, 50)

overlap.50kb.GW <- rbind.fill(overlap.chr1.50kb, overlap.chr2.50kb, overlap.chr3.50kb, overlap.chr4.50kb, overlap.chr5.50kb, overlap.chr6.50kb, overlap.chr7.50kb, overlap.chr8.50kb, overlap.chr9.50kb, overlap.chr10.50kb, overlap.chr11.50kb, overlap.chr12.50kb, overlap.chr13.50kb, overlap.chr14.50kb, overlap.chr15.50kb, overlap.chr16.50kb, overlap.chr17.50kb, overlap.chr18.50kb, overlap.chr19.50kb, overlap.chr20.50kb, overlap.chr21.50kb, overlap.chr22.50kb, overlap.chrX.50kb, overlap.chrY.50kb)
write.table(overlap.50kb.GW, file="/Users/ittaieres/Desktop/Hi-C/All_good_analysis/signif_comparisons/full_signif.50kb", sep="\t", col.names=T, row.names=T, quote=F)

#Get the tabulation DFs with basic count and bin factors for all the different kb "locus" sizes
tab_df.1.60kb <- full.chrom.tabulator(chr1.60kb, pos_df.1, 0, 400000, 10000, "chr1")
tab_df.2.60kb <- full.chrom.tabulator(chr2.60kb, pos_df.2, 0, 400000, 10000, "chr2")
tab_df.3.60kb <- full.chrom.tabulator(chr3.60kb, pos_df.3, 0, 400000, 10000, "chr3")
tab_df.4.60kb <- full.chrom.tabulator(chr4.60kb, pos_df.4, 0, 400000, 10000, "chr4")
tab_df.5.60kb <- full.chrom.tabulator(chr5.60kb, pos_df.5, 0, 400000, 10000, "chr5")
tab_df.6.60kb <- full.chrom.tabulator(chr6.60kb, pos_df.6, 0, 400000, 10000, "chr6")
tab_df.7.60kb <- full.chrom.tabulator(chr7.60kb, pos_df.7, 0, 400000, 10000, "chr7")
tab_df.8.60kb <- full.chrom.tabulator(chr8.60kb, pos_df.8, 0, 400000, 10000, "chr8")
tab_df.9.60kb <- full.chrom.tabulator(chr9.60kb, pos_df.9, 0, 400000, 10000, "chr9")
tab_df.10.60kb <- full.chrom.tabulator(chr10.60kb, pos_df.10, 0, 400000, 10000, "chr10")
tab_df.11.60kb <- full.chrom.tabulator(chr11.60kb, pos_df.11, 0, 400000, 10000, "chr11")
tab_df.12.60kb <- full.chrom.tabulator(chr12.60kb, pos_df.12, 0, 400000, 10000, "chr12")
tab_df.13.60kb <- full.chrom.tabulator(chr13.60kb, pos_df.13, 0, 400000, 10000, "chr13")
tab_df.14.60kb <- full.chrom.tabulator(chr14.60kb, pos_df.14, 0, 400000, 10000, "chr14")
tab_df.15.60kb <- full.chrom.tabulator(chr15.60kb, pos_df.15, 0, 400000, 10000, "chr15")
tab_df.16.60kb <- full.chrom.tabulator(chr16.60kb, pos_df.16, 0, 400000, 10000, "chr16")
tab_df.17.60kb <- full.chrom.tabulator(chr17.60kb, pos_df.17, 0, 400000, 10000, "chr17")
tab_df.18.60kb <- full.chrom.tabulator(chr18.60kb, pos_df.18, 0, 400000, 10000, "chr18")
tab_df.19.60kb <- full.chrom.tabulator(chr19.60kb, pos_df.19, 0, 400000, 10000, "chr19")
tab_df.20.60kb <- full.chrom.tabulator(chr20.60kb, pos_df.20, 0, 400000, 10000, "chr20")
tab_df.21.60kb <- full.chrom.tabulator(chr21.60kb, pos_df.21, 0, 400000, 10000, "chr21")
tab_df.22.60kb <- full.chrom.tabulator(chr22.60kb, pos_df.22, 0, 400000, 10000, "chr22")
tab_df.X.60kb <- full.chrom.tabulator(chrX.60kb, pos_df.X, 0, 400000, 10000, "chrX")
tab_df.Y.60kb <- full.chrom.tabulator(chrY.60kb, pos_df.Y, 0, 400000, 10000, "chrY")

tab_df.60kb.full <- rbind.fill(tab_df.1.60kb, tab_df.2.60kb, tab_df.3.60kb, tab_df.4.60kb, tab_df.5.60kb, tab_df.6.60kb, tab_df.7.60kb, tab_df.8.60kb, tab_df.9.60kb, tab_df.10.60kb, tab_df.11.60kb, tab_df.12.60kb, tab_df.13.60kb, tab_df.14.60kb, tab_df.15.60kb, tab_df.16.60kb, tab_df.17.60kb, tab_df.18.60kb, tab_df.19.60kb, tab_df.20.60kb, tab_df.21.60kb, tab_df.22.60kb, tab_df.X.60kb, tab_df.Y.60kb)
write.table(tab_df.60kb.full, file="/Users/ittaieres/Desktop/Hi-C/All_good_analysis/full_data_frames/full_data.60kb", sep="\t", col.names=T, row.names=F, quote=F)

#Get all the comparison dfs for all chromosomes at locus bins of .60kb

overlap.chr1.60kb <- signif.comparison("chr1_60kb_siglocs.txt", "chr1", tab_df.60kb.full, 60)
overlap.chr2.60kb <- signif.comparison("chr2_60kb_siglocs.txt", "chr2", tab_df.60kb.full, 60)
overlap.chr3.60kb <- signif.comparison("chr3_60kb_siglocs.txt", "chr3", tab_df.60kb.full, 60)
overlap.chr4.60kb <- signif.comparison("chr4_60kb_siglocs.txt", "chr4", tab_df.60kb.full, 60)
overlap.chr5.60kb <- signif.comparison("chr5_60kb_siglocs.txt", "chr5", tab_df.60kb.full, 60)
overlap.chr6.60kb <- signif.comparison("chr6_60kb_siglocs.txt", "chr6", tab_df.60kb.full, 60)
overlap.chr7.60kb <- signif.comparison("chr7_60kb_siglocs.txt", "chr7", tab_df.60kb.full, 60)
overlap.chr8.60kb <- signif.comparison("chr8_60kb_siglocs.txt", "chr8", tab_df.60kb.full, 60)
overlap.chr9.60kb <- signif.comparison("chr9_60kb_siglocs.txt", "chr9", tab_df.60kb.full, 60)
overlap.chr10.60kb <- signif.comparison("chr10_60kb_siglocs.txt", "chr10", tab_df.60kb.full, 60)
overlap.chr11.60kb <- signif.comparison("chr11_60kb_siglocs.txt", "chr11", tab_df.60kb.full, 60)
overlap.chr12.60kb <- signif.comparison("chr12_60kb_siglocs.txt", "chr12", tab_df.60kb.full, 60)
overlap.chr13.60kb <- signif.comparison("chr13_60kb_siglocs.txt", "chr13", tab_df.60kb.full, 60)
overlap.chr14.60kb <- signif.comparison("chr14_60kb_siglocs.txt", "chr14", tab_df.60kb.full, 60)
overlap.chr15.60kb <- signif.comparison("chr15_60kb_siglocs.txt", "chr15", tab_df.60kb.full, 60)
overlap.chr16.60kb <- signif.comparison("chr16_60kb_siglocs.txt", "chr16", tab_df.60kb.full, 60)
overlap.chr17.60kb <- signif.comparison("chr17_60kb_siglocs.txt", "chr17", tab_df.60kb.full, 60)
overlap.chr18.60kb <- signif.comparison("chr18_60kb_siglocs.txt", "chr18", tab_df.60kb.full, 60)
overlap.chr19.60kb <- signif.comparison("chr19_60kb_siglocs.txt", "chr19", tab_df.60kb.full, 60)
overlap.chr20.60kb <- signif.comparison("chr20_60kb_siglocs.txt", "chr20", tab_df.60kb.full, 60)
overlap.chr21.60kb <- signif.comparison("chr21_60kb_siglocs.txt", "chr21", tab_df.60kb.full, 60)
overlap.chr22.60kb <- signif.comparison("chr22_60kb_siglocs.txt", "chr22", tab_df.60kb.full, 60)
overlap.chrX.60kb <- signif.comparison("chrX_60kb_siglocs.txt", "chrX", tab_df.60kb.full, 60)
overlap.chrY.60kb <- signif.comparison("chrY_60kb_siglocs.txt", "chrY", tab_df.60kb.full, 60)

overlap.60kb.GW <- rbind.fill(overlap.chr1.60kb, overlap.chr2.60kb, overlap.chr3.60kb, overlap.chr4.60kb, overlap.chr5.60kb, overlap.chr6.60kb, overlap.chr7.60kb, overlap.chr8.60kb, overlap.chr9.60kb, overlap.chr10.60kb, overlap.chr11.60kb, overlap.chr12.60kb, overlap.chr13.60kb, overlap.chr14.60kb, overlap.chr15.60kb, overlap.chr16.60kb, overlap.chr17.60kb, overlap.chr18.60kb, overlap.chr19.60kb, overlap.chr20.60kb, overlap.chr21.60kb, overlap.chr22.60kb, overlap.chrX.60kb, overlap.chrY.60kb)
write.table(overlap.60kb.GW, file="/Users/ittaieres/Desktop/Hi-C/All_good_analysis/signif_comparisons/full_signif.60kb", sep="\t", col.names=T, row.names=T, quote=F)



#Get the tabulation DFs with basic count and bin factors for all the different kb "locus" sizes
tab_df.1.70kb <- full.chrom.tabulator(chr1.70kb, pos_df.1, 0, 400000, 10000, "chr1")
tab_df.2.70kb <- full.chrom.tabulator(chr2.70kb, pos_df.2, 0, 400000, 10000, "chr2")
tab_df.3.70kb <- full.chrom.tabulator(chr3.70kb, pos_df.3, 0, 400000, 10000, "chr3")
tab_df.4.70kb <- full.chrom.tabulator(chr4.70kb, pos_df.4, 0, 400000, 10000, "chr4")
tab_df.5.70kb <- full.chrom.tabulator(chr5.70kb, pos_df.5, 0, 400000, 10000, "chr5")
tab_df.6.70kb <- full.chrom.tabulator(chr6.70kb, pos_df.6, 0, 400000, 10000, "chr6")
tab_df.7.70kb <- full.chrom.tabulator(chr7.70kb, pos_df.7, 0, 400000, 10000, "chr7")
tab_df.8.70kb <- full.chrom.tabulator(chr8.70kb, pos_df.8, 0, 400000, 10000, "chr8")
tab_df.9.70kb <- full.chrom.tabulator(chr9.70kb, pos_df.9, 0, 400000, 10000, "chr9")
tab_df.10.70kb <- full.chrom.tabulator(chr10.70kb, pos_df.10, 0, 400000, 10000, "chr10")
tab_df.11.70kb <- full.chrom.tabulator(chr11.70kb, pos_df.11, 0, 400000, 10000, "chr11")
tab_df.12.70kb <- full.chrom.tabulator(chr12.70kb, pos_df.12, 0, 400000, 10000, "chr12")
tab_df.13.70kb <- full.chrom.tabulator(chr13.70kb, pos_df.13, 0, 400000, 10000, "chr13")
tab_df.14.70kb <- full.chrom.tabulator(chr14.70kb, pos_df.14, 0, 400000, 10000, "chr14")
tab_df.15.70kb <- full.chrom.tabulator(chr15.70kb, pos_df.15, 0, 400000, 10000, "chr15")
tab_df.16.70kb <- full.chrom.tabulator(chr16.70kb, pos_df.16, 0, 400000, 10000, "chr16")
tab_df.17.70kb <- full.chrom.tabulator(chr17.70kb, pos_df.17, 0, 400000, 10000, "chr17")
tab_df.18.70kb <- full.chrom.tabulator(chr18.70kb, pos_df.18, 0, 400000, 10000, "chr18")
tab_df.19.70kb <- full.chrom.tabulator(chr19.70kb, pos_df.19, 0, 400000, 10000, "chr19")
tab_df.20.70kb <- full.chrom.tabulator(chr20.70kb, pos_df.20, 0, 400000, 10000, "chr20")
tab_df.21.70kb <- full.chrom.tabulator(chr21.70kb, pos_df.21, 0, 400000, 10000, "chr21")
tab_df.22.70kb <- full.chrom.tabulator(chr22.70kb, pos_df.22, 0, 400000, 10000, "chr22")
tab_df.X.70kb <- full.chrom.tabulator(chrX.70kb, pos_df.X, 0, 400000, 10000, "chrX")
tab_df.Y.70kb <- full.chrom.tabulator(chrY.70kb, pos_df.Y, 0, 400000, 10000, "chrY")

tab_df.70kb.full <- rbind.fill(tab_df.1.70kb, tab_df.2.70kb, tab_df.3.70kb, tab_df.4.70kb, tab_df.5.70kb, tab_df.6.70kb, tab_df.7.70kb, tab_df.8.70kb, tab_df.9.70kb, tab_df.10.70kb, tab_df.11.70kb, tab_df.12.70kb, tab_df.13.70kb, tab_df.14.70kb, tab_df.15.70kb, tab_df.16.70kb, tab_df.17.70kb, tab_df.18.70kb, tab_df.19.70kb, tab_df.20.70kb, tab_df.21.70kb, tab_df.22.70kb, tab_df.X.70kb, tab_df.Y.70kb)
write.table(tab_df.70kb.full, file="/Users/ittaieres/Desktop/Hi-C/All_good_analysis/full_data_frames/full_data.70kb", sep="\t", col.names=T, row.names=F, quote=F)

#Get all the comparison dfs for all chromosomes at locus bins of .70kb

overlap.chr1.70kb <- signif.comparison("chr1_70kb_siglocs.txt", "chr1", tab_df.70kb.full, 70)
overlap.chr2.70kb <- signif.comparison("chr2_70kb_siglocs.txt", "chr2", tab_df.70kb.full, 70)
overlap.chr3.70kb <- signif.comparison("chr3_70kb_siglocs.txt", "chr3", tab_df.70kb.full, 70)
overlap.chr4.70kb <- signif.comparison("chr4_70kb_siglocs.txt", "chr4", tab_df.70kb.full, 70)
overlap.chr5.70kb <- signif.comparison("chr5_70kb_siglocs.txt", "chr5", tab_df.70kb.full, 70)
overlap.chr6.70kb <- signif.comparison("chr6_70kb_siglocs.txt", "chr6", tab_df.70kb.full, 70)
overlap.chr7.70kb <- signif.comparison("chr7_70kb_siglocs.txt", "chr7", tab_df.70kb.full, 70)
overlap.chr8.70kb <- signif.comparison("chr8_70kb_siglocs.txt", "chr8", tab_df.70kb.full, 70)
overlap.chr9.70kb <- signif.comparison("chr9_70kb_siglocs.txt", "chr9", tab_df.70kb.full, 70)
overlap.chr10.70kb <- signif.comparison("chr10_70kb_siglocs.txt", "chr10", tab_df.70kb.full, 70)
overlap.chr11.70kb <- signif.comparison("chr11_70kb_siglocs.txt", "chr11", tab_df.70kb.full, 70)
overlap.chr12.70kb <- signif.comparison("chr12_70kb_siglocs.txt", "chr12", tab_df.70kb.full, 70)
overlap.chr13.70kb <- signif.comparison("chr13_70kb_siglocs.txt", "chr13", tab_df.70kb.full, 70)
overlap.chr14.70kb <- signif.comparison("chr14_70kb_siglocs.txt", "chr14", tab_df.70kb.full, 70)
overlap.chr15.70kb <- signif.comparison("chr15_70kb_siglocs.txt", "chr15", tab_df.70kb.full, 70)
overlap.chr16.70kb <- signif.comparison("chr16_70kb_siglocs.txt", "chr16", tab_df.70kb.full, 70)
overlap.chr17.70kb <- signif.comparison("chr17_70kb_siglocs.txt", "chr17", tab_df.70kb.full, 70)
overlap.chr18.70kb <- signif.comparison("chr18_70kb_siglocs.txt", "chr18", tab_df.70kb.full, 70)
overlap.chr19.70kb <- signif.comparison("chr19_70kb_siglocs.txt", "chr19", tab_df.70kb.full, 70)
overlap.chr20.70kb <- signif.comparison("chr20_70kb_siglocs.txt", "chr20", tab_df.70kb.full, 70)
overlap.chr21.70kb <- signif.comparison("chr21_70kb_siglocs.txt", "chr21", tab_df.70kb.full, 70)
overlap.chr22.70kb <- signif.comparison("chr22_70kb_siglocs.txt", "chr22", tab_df.70kb.full, 70)
overlap.chrX.70kb <- signif.comparison("chrX_70kb_siglocs.txt", "chrX", tab_df.70kb.full, 70)
overlap.chrY.70kb <- signif.comparison("chrY_70kb_siglocs.txt", "chrY", tab_df.70kb.full, 70)

overlap.70kb.GW <- rbind.fill(overlap.chr1.70kb, overlap.chr2.70kb, overlap.chr3.70kb, overlap.chr4.70kb, overlap.chr5.70kb, overlap.chr6.70kb, overlap.chr7.70kb, overlap.chr8.70kb, overlap.chr9.70kb, overlap.chr10.70kb, overlap.chr11.70kb, overlap.chr12.70kb, overlap.chr13.70kb, overlap.chr14.70kb, overlap.chr15.70kb, overlap.chr16.70kb, overlap.chr17.70kb, overlap.chr18.70kb, overlap.chr19.70kb, overlap.chr20.70kb, overlap.chr21.70kb, overlap.chr22.70kb, overlap.chrX.70kb, overlap.chrY.70kb)
write.table(overlap.70kb.GW, file="/Users/ittaieres/Desktop/Hi-C/All_good_analysis/signif_comparisons/full_signif.70kb", sep="\t", col.names=T, row.names=T, quote=F)



#Get the tabulation DFs with basic count and bin factors for all the different kb "locus" sizes
tab_df.1.80kb <- full.chrom.tabulator(chr1.80kb, pos_df.1, 0, 400000, 10000, "chr1")
tab_df.2.80kb <- full.chrom.tabulator(chr2.80kb, pos_df.2, 0, 400000, 10000, "chr2")
tab_df.3.80kb <- full.chrom.tabulator(chr3.80kb, pos_df.3, 0, 400000, 10000, "chr3")
tab_df.4.80kb <- full.chrom.tabulator(chr4.80kb, pos_df.4, 0, 400000, 10000, "chr4")
tab_df.5.80kb <- full.chrom.tabulator(chr5.80kb, pos_df.5, 0, 400000, 10000, "chr5")
tab_df.6.80kb <- full.chrom.tabulator(chr6.80kb, pos_df.6, 0, 400000, 10000, "chr6")
tab_df.7.80kb <- full.chrom.tabulator(chr7.80kb, pos_df.7, 0, 400000, 10000, "chr7")
tab_df.8.80kb <- full.chrom.tabulator(chr8.80kb, pos_df.8, 0, 400000, 10000, "chr8")
tab_df.9.80kb <- full.chrom.tabulator(chr9.80kb, pos_df.9, 0, 400000, 10000, "chr9")
tab_df.10.80kb <- full.chrom.tabulator(chr10.80kb, pos_df.10, 0, 400000, 10000, "chr10")
tab_df.11.80kb <- full.chrom.tabulator(chr11.80kb, pos_df.11, 0, 400000, 10000, "chr11")
tab_df.12.80kb <- full.chrom.tabulator(chr12.80kb, pos_df.12, 0, 400000, 10000, "chr12")
tab_df.13.80kb <- full.chrom.tabulator(chr13.80kb, pos_df.13, 0, 400000, 10000, "chr13")
tab_df.14.80kb <- full.chrom.tabulator(chr14.80kb, pos_df.14, 0, 400000, 10000, "chr14")
tab_df.15.80kb <- full.chrom.tabulator(chr15.80kb, pos_df.15, 0, 400000, 10000, "chr15")
tab_df.16.80kb <- full.chrom.tabulator(chr16.80kb, pos_df.16, 0, 400000, 10000, "chr16")
tab_df.17.80kb <- full.chrom.tabulator(chr17.80kb, pos_df.17, 0, 400000, 10000, "chr17")
tab_df.18.80kb <- full.chrom.tabulator(chr18.80kb, pos_df.18, 0, 400000, 10000, "chr18")
tab_df.19.80kb <- full.chrom.tabulator(chr19.80kb, pos_df.19, 0, 400000, 10000, "chr19")
tab_df.20.80kb <- full.chrom.tabulator(chr20.80kb, pos_df.20, 0, 400000, 10000, "chr20")
tab_df.21.80kb <- full.chrom.tabulator(chr21.80kb, pos_df.21, 0, 400000, 10000, "chr21")
tab_df.22.80kb <- full.chrom.tabulator(chr22.80kb, pos_df.22, 0, 400000, 10000, "chr22")
tab_df.X.80kb <- full.chrom.tabulator(chrX.80kb, pos_df.X, 0, 400000, 10000, "chrX")
tab_df.Y.80kb <- full.chrom.tabulator(chrY.80kb, pos_df.Y, 0, 400000, 10000, "chrY")

tab_df.80kb.full <- rbind.fill(tab_df.1.80kb, tab_df.2.80kb, tab_df.3.80kb, tab_df.4.80kb, tab_df.5.80kb, tab_df.6.80kb, tab_df.7.80kb, tab_df.8.80kb, tab_df.9.80kb, tab_df.10.80kb, tab_df.11.80kb, tab_df.12.80kb, tab_df.13.80kb, tab_df.14.80kb, tab_df.15.80kb, tab_df.16.80kb, tab_df.17.80kb, tab_df.18.80kb, tab_df.19.80kb, tab_df.20.80kb, tab_df.21.80kb, tab_df.22.80kb, tab_df.X.80kb, tab_df.Y.80kb)
write.table(tab_df.80kb.full, file="/Users/ittaieres/Desktop/Hi-C/All_good_analysis/full_data_frames/full_data.80kb", sep="\t", col.names=T, row.names=F, quote=F)

#Get all the comparison dfs for all chromosomes at locus bins of .80kb

overlap.chr1.80kb <- signif.comparison("chr1_80kb_siglocs.txt", "chr1", tab_df.80kb.full, 80)
overlap.chr2.80kb <- signif.comparison("chr2_80kb_siglocs.txt", "chr2", tab_df.80kb.full, 80)
overlap.chr3.80kb <- signif.comparison("chr3_80kb_siglocs.txt", "chr3", tab_df.80kb.full, 80)
overlap.chr4.80kb <- signif.comparison("chr4_80kb_siglocs.txt", "chr4", tab_df.80kb.full, 80)
overlap.chr5.80kb <- signif.comparison("chr5_80kb_siglocs.txt", "chr5", tab_df.80kb.full, 80)
overlap.chr6.80kb <- signif.comparison("chr6_80kb_siglocs.txt", "chr6", tab_df.80kb.full, 80)
overlap.chr7.80kb <- signif.comparison("chr7_80kb_siglocs.txt", "chr7", tab_df.80kb.full, 80)
overlap.chr8.80kb <- signif.comparison("chr8_80kb_siglocs.txt", "chr8", tab_df.80kb.full, 80)
overlap.chr9.80kb <- signif.comparison("chr9_80kb_siglocs.txt", "chr9", tab_df.80kb.full, 80)
overlap.chr10.80kb <- signif.comparison("chr10_80kb_siglocs.txt", "chr10", tab_df.80kb.full, 80)
overlap.chr11.80kb <- signif.comparison("chr11_80kb_siglocs.txt", "chr11", tab_df.80kb.full, 80)
overlap.chr12.80kb <- signif.comparison("chr12_80kb_siglocs.txt", "chr12", tab_df.80kb.full, 80)
overlap.chr13.80kb <- signif.comparison("chr13_80kb_siglocs.txt", "chr13", tab_df.80kb.full, 80)
overlap.chr14.80kb <- signif.comparison("chr14_80kb_siglocs.txt", "chr14", tab_df.80kb.full, 80)
overlap.chr15.80kb <- signif.comparison("chr15_80kb_siglocs.txt", "chr15", tab_df.80kb.full, 80)
overlap.chr16.80kb <- signif.comparison("chr16_80kb_siglocs.txt", "chr16", tab_df.80kb.full, 80)
overlap.chr17.80kb <- signif.comparison("chr17_80kb_siglocs.txt", "chr17", tab_df.80kb.full, 80)
overlap.chr18.80kb <- signif.comparison("chr18_80kb_siglocs.txt", "chr18", tab_df.80kb.full, 80)
overlap.chr19.80kb <- signif.comparison("chr19_80kb_siglocs.txt", "chr19", tab_df.80kb.full, 80)
overlap.chr20.80kb <- signif.comparison("chr20_80kb_siglocs.txt", "chr20", tab_df.80kb.full, 80)
overlap.chr21.80kb <- signif.comparison("chr21_80kb_siglocs.txt", "chr21", tab_df.80kb.full, 80)
overlap.chr22.80kb <- signif.comparison("chr22_80kb_siglocs.txt", "chr22", tab_df.80kb.full, 80)
overlap.chrX.80kb <- signif.comparison("chrX_80kb_siglocs.txt", "chrX", tab_df.80kb.full, 80)
overlap.chrY.80kb <- signif.comparison("chrY_80kb_siglocs.txt", "chrY", tab_df.80kb.full, 80)

overlap.80kb.GW <- rbind.fill(overlap.chr1.80kb, overlap.chr2.80kb, overlap.chr3.80kb, overlap.chr4.80kb, overlap.chr5.80kb, overlap.chr6.80kb, overlap.chr7.80kb, overlap.chr8.80kb, overlap.chr9.80kb, overlap.chr10.80kb, overlap.chr11.80kb, overlap.chr12.80kb, overlap.chr13.80kb, overlap.chr14.80kb, overlap.chr15.80kb, overlap.chr16.80kb, overlap.chr17.80kb, overlap.chr18.80kb, overlap.chr19.80kb, overlap.chr20.80kb, overlap.chr21.80kb, overlap.chr22.80kb, overlap.chrX.80kb, overlap.chrY.80kb)
write.table(overlap.80kb.GW, file="/Users/ittaieres/Desktop/Hi-C/All_good_analysis/signif_comparisons/full_signif.80kb", sep="\t", col.names=T, row.names=T, quote=F)



#Get the tabulation DFs with basic count and bin factors for all the different kb "locus" sizes
tab_df.1.90kb <- full.chrom.tabulator(chr1.90kb, pos_df.1, 0, 400000, 10000, "chr1")
tab_df.2.90kb <- full.chrom.tabulator(chr2.90kb, pos_df.2, 0, 400000, 10000, "chr2")
tab_df.3.90kb <- full.chrom.tabulator(chr3.90kb, pos_df.3, 0, 400000, 10000, "chr3")
tab_df.4.90kb <- full.chrom.tabulator(chr4.90kb, pos_df.4, 0, 400000, 10000, "chr4")
tab_df.5.90kb <- full.chrom.tabulator(chr5.90kb, pos_df.5, 0, 400000, 10000, "chr5")
tab_df.6.90kb <- full.chrom.tabulator(chr6.90kb, pos_df.6, 0, 400000, 10000, "chr6")
tab_df.7.90kb <- full.chrom.tabulator(chr7.90kb, pos_df.7, 0, 400000, 10000, "chr7")
tab_df.8.90kb <- full.chrom.tabulator(chr8.90kb, pos_df.8, 0, 400000, 10000, "chr8")
tab_df.9.90kb <- full.chrom.tabulator(chr9.90kb, pos_df.9, 0, 400000, 10000, "chr9")
tab_df.10.90kb <- full.chrom.tabulator(chr10.90kb, pos_df.10, 0, 400000, 10000, "chr10")
tab_df.11.90kb <- full.chrom.tabulator(chr11.90kb, pos_df.11, 0, 400000, 10000, "chr11")
tab_df.12.90kb <- full.chrom.tabulator(chr12.90kb, pos_df.12, 0, 400000, 10000, "chr12")
tab_df.13.90kb <- full.chrom.tabulator(chr13.90kb, pos_df.13, 0, 400000, 10000, "chr13")
tab_df.14.90kb <- full.chrom.tabulator(chr14.90kb, pos_df.14, 0, 400000, 10000, "chr14")
tab_df.15.90kb <- full.chrom.tabulator(chr15.90kb, pos_df.15, 0, 400000, 10000, "chr15")
tab_df.16.90kb <- full.chrom.tabulator(chr16.90kb, pos_df.16, 0, 400000, 10000, "chr16")
tab_df.17.90kb <- full.chrom.tabulator(chr17.90kb, pos_df.17, 0, 400000, 10000, "chr17")
tab_df.18.90kb <- full.chrom.tabulator(chr18.90kb, pos_df.18, 0, 400000, 10000, "chr18")
tab_df.19.90kb <- full.chrom.tabulator(chr19.90kb, pos_df.19, 0, 400000, 10000, "chr19")
tab_df.20.90kb <- full.chrom.tabulator(chr20.90kb, pos_df.20, 0, 400000, 10000, "chr20")
tab_df.21.90kb <- full.chrom.tabulator(chr21.90kb, pos_df.21, 0, 400000, 10000, "chr21")
tab_df.22.90kb <- full.chrom.tabulator(chr22.90kb, pos_df.22, 0, 400000, 10000, "chr22")
tab_df.X.90kb <- full.chrom.tabulator(chrX.90kb, pos_df.X, 0, 400000, 10000, "chrX")
tab_df.Y.90kb <- full.chrom.tabulator(chrY.90kb, pos_df.Y, 0, 400000, 10000, "chrY")

tab_df.90kb.full <- rbind.fill(tab_df.1.90kb, tab_df.2.90kb, tab_df.3.90kb, tab_df.4.90kb, tab_df.5.90kb, tab_df.6.90kb, tab_df.7.90kb, tab_df.8.90kb, tab_df.9.90kb, tab_df.10.90kb, tab_df.11.90kb, tab_df.12.90kb, tab_df.13.90kb, tab_df.14.90kb, tab_df.15.90kb, tab_df.16.90kb, tab_df.17.90kb, tab_df.18.90kb, tab_df.19.90kb, tab_df.20.90kb, tab_df.21.90kb, tab_df.22.90kb, tab_df.X.90kb, tab_df.Y.90kb)
write.table(tab_df.90kb.full, file="/Users/ittaieres/Desktop/Hi-C/All_good_analysis/full_data_frames/full_data.90kb", sep="\t", col.names=T, row.names=F, quote=F)

#Get all the comparison dfs for all chromosomes at locus bins of .90kb

overlap.chr1.90kb <- signif.comparison("chr1_90kb_siglocs.txt", "chr1", tab_df.90kb.full, 90)
overlap.chr2.90kb <- signif.comparison("chr2_90kb_siglocs.txt", "chr2", tab_df.90kb.full, 90)
overlap.chr3.90kb <- signif.comparison("chr3_90kb_siglocs.txt", "chr3", tab_df.90kb.full, 90)
overlap.chr4.90kb <- signif.comparison("chr4_90kb_siglocs.txt", "chr4", tab_df.90kb.full, 90)
overlap.chr5.90kb <- signif.comparison("chr5_90kb_siglocs.txt", "chr5", tab_df.90kb.full, 90)
overlap.chr6.90kb <- signif.comparison("chr6_90kb_siglocs.txt", "chr6", tab_df.90kb.full, 90)
overlap.chr7.90kb <- signif.comparison("chr7_90kb_siglocs.txt", "chr7", tab_df.90kb.full, 90)
overlap.chr8.90kb <- signif.comparison("chr8_90kb_siglocs.txt", "chr8", tab_df.90kb.full, 90)
overlap.chr9.90kb <- signif.comparison("chr9_90kb_siglocs.txt", "chr9", tab_df.90kb.full, 90)
overlap.chr10.90kb <- signif.comparison("chr10_90kb_siglocs.txt", "chr10", tab_df.90kb.full, 90)
overlap.chr11.90kb <- signif.comparison("chr11_90kb_siglocs.txt", "chr11", tab_df.90kb.full, 90)
overlap.chr12.90kb <- signif.comparison("chr12_90kb_siglocs.txt", "chr12", tab_df.90kb.full, 90)
overlap.chr13.90kb <- signif.comparison("chr13_90kb_siglocs.txt", "chr13", tab_df.90kb.full, 90)
overlap.chr14.90kb <- signif.comparison("chr14_90kb_siglocs.txt", "chr14", tab_df.90kb.full, 90)
overlap.chr15.90kb <- signif.comparison("chr15_90kb_siglocs.txt", "chr15", tab_df.90kb.full, 90)
overlap.chr16.90kb <- signif.comparison("chr16_90kb_siglocs.txt", "chr16", tab_df.90kb.full, 90)
overlap.chr17.90kb <- signif.comparison("chr17_90kb_siglocs.txt", "chr17", tab_df.90kb.full, 90)
overlap.chr18.90kb <- signif.comparison("chr18_90kb_siglocs.txt", "chr18", tab_df.90kb.full, 90)
overlap.chr19.90kb <- signif.comparison("chr19_90kb_siglocs.txt", "chr19", tab_df.90kb.full, 90)
overlap.chr20.90kb <- signif.comparison("chr20_90kb_siglocs.txt", "chr20", tab_df.90kb.full, 90)
overlap.chr21.90kb <- signif.comparison("chr21_90kb_siglocs.txt", "chr21", tab_df.90kb.full, 90)
overlap.chr22.90kb <- signif.comparison("chr22_90kb_siglocs.txt", "chr22", tab_df.90kb.full, 90)
overlap.chrX.90kb <- signif.comparison("chrX_90kb_siglocs.txt", "chrX", tab_df.90kb.full, 90)
overlap.chrY.90kb <- signif.comparison("chrY_90kb_siglocs.txt", "chrY", tab_df.90kb.full, 90)

overlap.90kb.GW <- rbind.fill(overlap.chr1.90kb, overlap.chr2.90kb, overlap.chr3.90kb, overlap.chr4.90kb, overlap.chr5.90kb, overlap.chr6.90kb, overlap.chr7.90kb, overlap.chr8.90kb, overlap.chr9.90kb, overlap.chr10.90kb, overlap.chr11.90kb, overlap.chr12.90kb, overlap.chr13.90kb, overlap.chr14.90kb, overlap.chr15.90kb, overlap.chr16.90kb, overlap.chr17.90kb, overlap.chr18.90kb, overlap.chr19.90kb, overlap.chr20.90kb, overlap.chr21.90kb, overlap.chr22.90kb, overlap.chrX.90kb, overlap.chrY.90kb)
write.table(overlap.90kb.GW, file="/Users/ittaieres/Desktop/Hi-C/All_good_analysis/signif_comparisons/full_signif.90kb", sep="\t", col.names=T, row.names=T, quote=F)



#Get the tabulation DFs with basic count and bin factors for all the different kb "locus" sizes
tab_df.1.100kb <- full.chrom.tabulator(chr1.100kb, pos_df.1, 0, 400000, 10000, "chr1")
tab_df.2.100kb <- full.chrom.tabulator(chr2.100kb, pos_df.2, 0, 400000, 10000, "chr2")
tab_df.3.100kb <- full.chrom.tabulator(chr3.100kb, pos_df.3, 0, 400000, 10000, "chr3")
tab_df.4.100kb <- full.chrom.tabulator(chr4.100kb, pos_df.4, 0, 400000, 10000, "chr4")
tab_df.5.100kb <- full.chrom.tabulator(chr5.100kb, pos_df.5, 0, 400000, 10000, "chr5")
tab_df.6.100kb <- full.chrom.tabulator(chr6.100kb, pos_df.6, 0, 400000, 10000, "chr6")
tab_df.7.100kb <- full.chrom.tabulator(chr7.100kb, pos_df.7, 0, 400000, 10000, "chr7")
tab_df.8.100kb <- full.chrom.tabulator(chr8.100kb, pos_df.8, 0, 400000, 10000, "chr8")
tab_df.9.100kb <- full.chrom.tabulator(chr9.100kb, pos_df.9, 0, 400000, 10000, "chr9")
tab_df.10.100kb <- full.chrom.tabulator(chr10.100kb, pos_df.10, 0, 400000, 10000, "chr10")
tab_df.11.100kb <- full.chrom.tabulator(chr11.100kb, pos_df.11, 0, 400000, 10000, "chr11")
tab_df.12.100kb <- full.chrom.tabulator(chr12.100kb, pos_df.12, 0, 400000, 10000, "chr12")
tab_df.13.100kb <- full.chrom.tabulator(chr13.100kb, pos_df.13, 0, 400000, 10000, "chr13")
tab_df.14.100kb <- full.chrom.tabulator(chr14.100kb, pos_df.14, 0, 400000, 10000, "chr14")
tab_df.15.100kb <- full.chrom.tabulator(chr15.100kb, pos_df.15, 0, 400000, 10000, "chr15")
tab_df.16.100kb <- full.chrom.tabulator(chr16.100kb, pos_df.16, 0, 400000, 10000, "chr16")
tab_df.17.100kb <- full.chrom.tabulator(chr17.100kb, pos_df.17, 0, 400000, 10000, "chr17")
tab_df.18.100kb <- full.chrom.tabulator(chr18.100kb, pos_df.18, 0, 400000, 10000, "chr18")
tab_df.19.100kb <- full.chrom.tabulator(chr19.100kb, pos_df.19, 0, 400000, 10000, "chr19")
tab_df.20.100kb <- full.chrom.tabulator(chr20.100kb, pos_df.20, 0, 400000, 10000, "chr20")
tab_df.21.100kb <- full.chrom.tabulator(chr21.100kb, pos_df.21, 0, 400000, 10000, "chr21")
tab_df.22.100kb <- full.chrom.tabulator(chr22.100kb, pos_df.22, 0, 400000, 10000, "chr22")
tab_df.X.100kb <- full.chrom.tabulator(chrX.100kb, pos_df.X, 0, 400000, 10000, "chrX")
tab_df.Y.100kb <- full.chrom.tabulator(chrY.100kb, pos_df.Y, 0, 400000, 10000, "chrY")

tab_df.100kb.full <- rbind.fill(tab_df.1.100kb, tab_df.2.100kb, tab_df.3.100kb, tab_df.4.100kb, tab_df.5.100kb, tab_df.6.100kb, tab_df.7.100kb, tab_df.8.100kb, tab_df.9.100kb, tab_df.10.100kb, tab_df.11.100kb, tab_df.12.100kb, tab_df.13.100kb, tab_df.14.100kb, tab_df.15.100kb, tab_df.16.100kb, tab_df.17.100kb, tab_df.18.100kb, tab_df.19.100kb, tab_df.20.100kb, tab_df.21.100kb, tab_df.22.100kb, tab_df.X.100kb, tab_df.Y.100kb)
write.table(tab_df.100kb.full, file="/Users/ittaieres/Desktop/Hi-C/All_good_analysis/full_data_frames/full_data.100kb", sep="\t", col.names=T, row.names=F, quote=F)

#Get all the comparison dfs for all chromosomes at locus bins of .100kb

overlap.chr1.100kb <- signif.comparison("chr1_100kb_siglocs.txt", "chr1", tab_df.100kb.full, 100)
overlap.chr2.100kb <- signif.comparison("chr2_100kb_siglocs.txt", "chr2", tab_df.100kb.full, 100)
overlap.chr3.100kb <- signif.comparison("chr3_100kb_siglocs.txt", "chr3", tab_df.100kb.full, 100)
overlap.chr4.100kb <- signif.comparison("chr4_100kb_siglocs.txt", "chr4", tab_df.100kb.full, 100)
overlap.chr5.100kb <- signif.comparison("chr5_100kb_siglocs.txt", "chr5", tab_df.100kb.full, 100)
overlap.chr6.100kb <- signif.comparison("chr6_100kb_siglocs.txt", "chr6", tab_df.100kb.full, 100)
overlap.chr7.100kb <- signif.comparison("chr7_100kb_siglocs.txt", "chr7", tab_df.100kb.full, 100)
overlap.chr8.100kb <- signif.comparison("chr8_100kb_siglocs.txt", "chr8", tab_df.100kb.full, 100)
overlap.chr9.100kb <- signif.comparison("chr9_100kb_siglocs.txt", "chr9", tab_df.100kb.full, 100)
overlap.chr10.100kb <- signif.comparison("chr10_100kb_siglocs.txt", "chr10", tab_df.100kb.full, 100)
overlap.chr11.100kb <- signif.comparison("chr11_100kb_siglocs.txt", "chr11", tab_df.100kb.full, 100)
overlap.chr12.100kb <- signif.comparison("chr12_100kb_siglocs.txt", "chr12", tab_df.100kb.full, 100)
overlap.chr13.100kb <- signif.comparison("chr13_100kb_siglocs.txt", "chr13", tab_df.100kb.full, 100)
overlap.chr14.100kb <- signif.comparison("chr14_100kb_siglocs.txt", "chr14", tab_df.100kb.full, 100)
overlap.chr15.100kb <- signif.comparison("chr15_100kb_siglocs.txt", "chr15", tab_df.100kb.full, 100)
overlap.chr16.100kb <- signif.comparison("chr16_100kb_siglocs.txt", "chr16", tab_df.100kb.full, 100)
overlap.chr17.100kb <- signif.comparison("chr17_100kb_siglocs.txt", "chr17", tab_df.100kb.full, 100)
overlap.chr18.100kb <- signif.comparison("chr18_100kb_siglocs.txt", "chr18", tab_df.100kb.full, 100)
overlap.chr19.100kb <- signif.comparison("chr19_100kb_siglocs.txt", "chr19", tab_df.100kb.full, 100)
overlap.chr20.100kb <- signif.comparison("chr20_100kb_siglocs.txt", "chr20", tab_df.100kb.full, 100)
overlap.chr21.100kb <- signif.comparison("chr21_100kb_siglocs.txt", "chr21", tab_df.100kb.full, 100)
overlap.chr22.100kb <- signif.comparison("chr22_100kb_siglocs.txt", "chr22", tab_df.100kb.full, 100)
overlap.chrX.100kb <- signif.comparison("chrX_100kb_siglocs.txt", "chrX", tab_df.100kb.full, 100)
overlap.chrY.100kb <- signif.comparison("chrY_100kb_siglocs.txt", "chrY", tab_df.100kb.full, 100)

overlap.100kb.GW <- rbind.fill(overlap.chr1.100kb, overlap.chr2.100kb, overlap.chr3.100kb, overlap.chr4.100kb, overlap.chr5.100kb, overlap.chr6.100kb, overlap.chr7.100kb, overlap.chr8.100kb, overlap.chr9.100kb, overlap.chr10.100kb, overlap.chr11.100kb, overlap.chr12.100kb, overlap.chr13.100kb, overlap.chr14.100kb, overlap.chr15.100kb, overlap.chr16.100kb, overlap.chr17.100kb, overlap.chr18.100kb, overlap.chr19.100kb, overlap.chr20.100kb, overlap.chr21.100kb, overlap.chr22.100kb, overlap.chrX.100kb, overlap.chrY.100kb)
write.table(overlap.100kb.GW, file="/Users/ittaieres/Desktop/Hi-C/All_good_analysis/signif_comparisons/full_signif.100kb", sep="\t", col.names=T, row.names=T, quote=F)



#Get the tabulation DFs with basic count and bin factors for all the different kb "locus" sizes
tab_df.1.125kb <- full.chrom.tabulator(chr1.125kb, pos_df.1, 0, 400000, 10000, "chr1")
tab_df.2.125kb <- full.chrom.tabulator(chr2.125kb, pos_df.2, 0, 400000, 10000, "chr2")
tab_df.3.125kb <- full.chrom.tabulator(chr3.125kb, pos_df.3, 0, 400000, 10000, "chr3")
tab_df.4.125kb <- full.chrom.tabulator(chr4.125kb, pos_df.4, 0, 400000, 10000, "chr4")
tab_df.5.125kb <- full.chrom.tabulator(chr5.125kb, pos_df.5, 0, 400000, 10000, "chr5")
tab_df.6.125kb <- full.chrom.tabulator(chr6.125kb, pos_df.6, 0, 400000, 10000, "chr6")
tab_df.7.125kb <- full.chrom.tabulator(chr7.125kb, pos_df.7, 0, 400000, 10000, "chr7")
tab_df.8.125kb <- full.chrom.tabulator(chr8.125kb, pos_df.8, 0, 400000, 10000, "chr8")
tab_df.9.125kb <- full.chrom.tabulator(chr9.125kb, pos_df.9, 0, 400000, 10000, "chr9")
tab_df.10.125kb <- full.chrom.tabulator(chr10.125kb, pos_df.10, 0, 400000, 10000, "chr10")
tab_df.11.125kb <- full.chrom.tabulator(chr11.125kb, pos_df.11, 0, 400000, 10000, "chr11")
tab_df.12.125kb <- full.chrom.tabulator(chr12.125kb, pos_df.12, 0, 400000, 10000, "chr12")
tab_df.13.125kb <- full.chrom.tabulator(chr13.125kb, pos_df.13, 0, 400000, 10000, "chr13")
tab_df.14.125kb <- full.chrom.tabulator(chr14.125kb, pos_df.14, 0, 400000, 10000, "chr14")
tab_df.15.125kb <- full.chrom.tabulator(chr15.125kb, pos_df.15, 0, 400000, 10000, "chr15")
tab_df.16.125kb <- full.chrom.tabulator(chr16.125kb, pos_df.16, 0, 400000, 10000, "chr16")
tab_df.17.125kb <- full.chrom.tabulator(chr17.125kb, pos_df.17, 0, 400000, 10000, "chr17")
tab_df.18.125kb <- full.chrom.tabulator(chr18.125kb, pos_df.18, 0, 400000, 10000, "chr18")
tab_df.19.125kb <- full.chrom.tabulator(chr19.125kb, pos_df.19, 0, 400000, 10000, "chr19")
tab_df.20.125kb <- full.chrom.tabulator(chr20.125kb, pos_df.20, 0, 400000, 10000, "chr20")
tab_df.21.125kb <- full.chrom.tabulator(chr21.125kb, pos_df.21, 0, 400000, 10000, "chr21")
tab_df.22.125kb <- full.chrom.tabulator(chr22.125kb, pos_df.22, 0, 400000, 10000, "chr22")
tab_df.X.125kb <- full.chrom.tabulator(chrX.125kb, pos_df.X, 0, 400000, 10000, "chrX")
tab_df.Y.125kb <- full.chrom.tabulator(chrY.125kb, pos_df.Y, 0, 400000, 10000, "chrY")

tab_df.125kb.full <- rbind.fill(tab_df.1.125kb, tab_df.2.125kb, tab_df.3.125kb, tab_df.4.125kb, tab_df.5.125kb, tab_df.6.125kb, tab_df.7.125kb, tab_df.8.125kb, tab_df.9.125kb, tab_df.10.125kb, tab_df.11.125kb, tab_df.12.125kb, tab_df.13.125kb, tab_df.14.125kb, tab_df.15.125kb, tab_df.16.125kb, tab_df.17.125kb, tab_df.18.125kb, tab_df.19.125kb, tab_df.20.125kb, tab_df.21.125kb, tab_df.22.125kb, tab_df.X.125kb, tab_df.Y.125kb)
write.table(tab_df.125kb.full, file="/Users/ittaieres/Desktop/Hi-C/All_good_analysis/full_data_frames/full_data.125kb", sep="\t", col.names=T, row.names=F, quote=F)

#Get all the comparison dfs for all chromosomes at locus bins of .125kb

overlap.chr1.125kb <- signif.comparison("chr1_125kb_siglocs.txt", "chr1", tab_df.125kb.full, 125)
overlap.chr2.125kb <- signif.comparison("chr2_125kb_siglocs.txt", "chr2", tab_df.125kb.full, 125)
overlap.chr3.125kb <- signif.comparison("chr3_125kb_siglocs.txt", "chr3", tab_df.125kb.full, 125)
overlap.chr4.125kb <- signif.comparison("chr4_125kb_siglocs.txt", "chr4", tab_df.125kb.full, 125)
overlap.chr5.125kb <- signif.comparison("chr5_125kb_siglocs.txt", "chr5", tab_df.125kb.full, 125)
overlap.chr6.125kb <- signif.comparison("chr6_125kb_siglocs.txt", "chr6", tab_df.125kb.full, 125)
overlap.chr7.125kb <- signif.comparison("chr7_125kb_siglocs.txt", "chr7", tab_df.125kb.full, 125)
overlap.chr8.125kb <- signif.comparison("chr8_125kb_siglocs.txt", "chr8", tab_df.125kb.full, 125)
overlap.chr9.125kb <- signif.comparison("chr9_125kb_siglocs.txt", "chr9", tab_df.125kb.full, 125)
overlap.chr10.125kb <- signif.comparison("chr10_125kb_siglocs.txt", "chr10", tab_df.125kb.full, 125)
overlap.chr11.125kb <- signif.comparison("chr11_125kb_siglocs.txt", "chr11", tab_df.125kb.full, 125)
overlap.chr12.125kb <- signif.comparison("chr12_125kb_siglocs.txt", "chr12", tab_df.125kb.full, 125)
overlap.chr13.125kb <- signif.comparison("chr13_125kb_siglocs.txt", "chr13", tab_df.125kb.full, 125)
overlap.chr14.125kb <- signif.comparison("chr14_125kb_siglocs.txt", "chr14", tab_df.125kb.full, 125)
overlap.chr15.125kb <- signif.comparison("chr15_125kb_siglocs.txt", "chr15", tab_df.125kb.full, 125)
overlap.chr16.125kb <- signif.comparison("chr16_125kb_siglocs.txt", "chr16", tab_df.125kb.full, 125)
overlap.chr17.125kb <- signif.comparison("chr17_125kb_siglocs.txt", "chr17", tab_df.125kb.full, 125)
overlap.chr18.125kb <- signif.comparison("chr18_125kb_siglocs.txt", "chr18", tab_df.125kb.full, 125)
overlap.chr19.125kb <- signif.comparison("chr19_125kb_siglocs.txt", "chr19", tab_df.125kb.full, 125)
overlap.chr20.125kb <- signif.comparison("chr20_125kb_siglocs.txt", "chr20", tab_df.125kb.full, 125)
overlap.chr21.125kb <- signif.comparison("chr21_125kb_siglocs.txt", "chr21", tab_df.125kb.full, 125)
overlap.chr22.125kb <- signif.comparison("chr22_125kb_siglocs.txt", "chr22", tab_df.125kb.full, 125)
overlap.chrX.125kb <- signif.comparison("chrX_125kb_siglocs.txt", "chrX", tab_df.125kb.full, 125)
overlap.chrY.125kb <- signif.comparison("chrY_125kb_siglocs.txt", "chrY", tab_df.125kb.full, 125)

overlap.125kb.GW <- rbind.fill(overlap.chr1.125kb, overlap.chr2.125kb, overlap.chr3.125kb, overlap.chr4.125kb, overlap.chr5.125kb, overlap.chr6.125kb, overlap.chr7.125kb, overlap.chr8.125kb, overlap.chr9.125kb, overlap.chr10.125kb, overlap.chr11.125kb, overlap.chr12.125kb, overlap.chr13.125kb, overlap.chr14.125kb, overlap.chr15.125kb, overlap.chr16.125kb, overlap.chr17.125kb, overlap.chr18.125kb, overlap.chr19.125kb, overlap.chr20.125kb, overlap.chr21.125kb, overlap.chr22.125kb, overlap.chrX.125kb, overlap.chrY.125kb)
write.table(overlap.125kb.GW, file="/Users/ittaieres/Desktop/Hi-C/All_good_analysis/signif_comparisons/full_signif.125kb", sep="\t", col.names=T, row.names=T, quote=F)



#Get the tabulation DFs with basic count and bin factors for all the different kb "locus" sizes
tab_df.1.150kb <- full.chrom.tabulator(chr1.150kb, pos_df.1, 0, 400000, 10000, "chr1")
tab_df.2.150kb <- full.chrom.tabulator(chr2.150kb, pos_df.2, 0, 400000, 10000, "chr2")
tab_df.3.150kb <- full.chrom.tabulator(chr3.150kb, pos_df.3, 0, 400000, 10000, "chr3")
tab_df.4.150kb <- full.chrom.tabulator(chr4.150kb, pos_df.4, 0, 400000, 10000, "chr4")
tab_df.5.150kb <- full.chrom.tabulator(chr5.150kb, pos_df.5, 0, 400000, 10000, "chr5")
tab_df.6.150kb <- full.chrom.tabulator(chr6.150kb, pos_df.6, 0, 400000, 10000, "chr6")
tab_df.7.150kb <- full.chrom.tabulator(chr7.150kb, pos_df.7, 0, 400000, 10000, "chr7")
tab_df.8.150kb <- full.chrom.tabulator(chr8.150kb, pos_df.8, 0, 400000, 10000, "chr8")
tab_df.9.150kb <- full.chrom.tabulator(chr9.150kb, pos_df.9, 0, 400000, 10000, "chr9")
tab_df.10.150kb <- full.chrom.tabulator(chr10.150kb, pos_df.10, 0, 400000, 10000, "chr10")
tab_df.11.150kb <- full.chrom.tabulator(chr11.150kb, pos_df.11, 0, 400000, 10000, "chr11")
tab_df.12.150kb <- full.chrom.tabulator(chr12.150kb, pos_df.12, 0, 400000, 10000, "chr12")
tab_df.13.150kb <- full.chrom.tabulator(chr13.150kb, pos_df.13, 0, 400000, 10000, "chr13")
tab_df.14.150kb <- full.chrom.tabulator(chr14.150kb, pos_df.14, 0, 400000, 10000, "chr14")
tab_df.15.150kb <- full.chrom.tabulator(chr15.150kb, pos_df.15, 0, 400000, 10000, "chr15")
tab_df.16.150kb <- full.chrom.tabulator(chr16.150kb, pos_df.16, 0, 400000, 10000, "chr16")
tab_df.17.150kb <- full.chrom.tabulator(chr17.150kb, pos_df.17, 0, 400000, 10000, "chr17")
tab_df.18.150kb <- full.chrom.tabulator(chr18.150kb, pos_df.18, 0, 400000, 10000, "chr18")
tab_df.19.150kb <- full.chrom.tabulator(chr19.150kb, pos_df.19, 0, 400000, 10000, "chr19")
tab_df.20.150kb <- full.chrom.tabulator(chr20.150kb, pos_df.20, 0, 400000, 10000, "chr20")
tab_df.21.150kb <- full.chrom.tabulator(chr21.150kb, pos_df.21, 0, 400000, 10000, "chr21")
tab_df.22.150kb <- full.chrom.tabulator(chr22.150kb, pos_df.22, 0, 400000, 10000, "chr22")
tab_df.X.150kb <- full.chrom.tabulator(chrX.150kb, pos_df.X, 0, 400000, 10000, "chrX")
tab_df.Y.150kb <- full.chrom.tabulator(chrY.150kb, pos_df.Y, 0, 400000, 10000, "chrY")

tab_df.150kb.full <- rbind.fill(tab_df.1.150kb, tab_df.2.150kb, tab_df.3.150kb, tab_df.4.150kb, tab_df.5.150kb, tab_df.6.150kb, tab_df.7.150kb, tab_df.8.150kb, tab_df.9.150kb, tab_df.10.150kb, tab_df.11.150kb, tab_df.12.150kb, tab_df.13.150kb, tab_df.14.150kb, tab_df.15.150kb, tab_df.16.150kb, tab_df.17.150kb, tab_df.18.150kb, tab_df.19.150kb, tab_df.20.150kb, tab_df.21.150kb, tab_df.22.150kb, tab_df.X.150kb, tab_df.Y.150kb)
write.table(tab_df.150kb.full, file="/Users/ittaieres/Desktop/Hi-C/All_good_analysis/full_data_frames/full_data.150kb", sep="\t", col.names=T, row.names=F, quote=F)

#Get all the comparison dfs for all chromosomes at locus bins of .150kb

overlap.chr1.150kb <- signif.comparison("chr1_150kb_siglocs.txt", "chr1", tab_df.150kb.full, 150)
overlap.chr2.150kb <- signif.comparison("chr2_150kb_siglocs.txt", "chr2", tab_df.150kb.full, 150)
overlap.chr3.150kb <- signif.comparison("chr3_150kb_siglocs.txt", "chr3", tab_df.150kb.full, 150)
overlap.chr4.150kb <- signif.comparison("chr4_150kb_siglocs.txt", "chr4", tab_df.150kb.full, 150)
overlap.chr5.150kb <- signif.comparison("chr5_150kb_siglocs.txt", "chr5", tab_df.150kb.full, 150)
overlap.chr6.150kb <- signif.comparison("chr6_150kb_siglocs.txt", "chr6", tab_df.150kb.full, 150)
overlap.chr7.150kb <- signif.comparison("chr7_150kb_siglocs.txt", "chr7", tab_df.150kb.full, 150)
overlap.chr8.150kb <- signif.comparison("chr8_150kb_siglocs.txt", "chr8", tab_df.150kb.full, 150)
overlap.chr9.150kb <- signif.comparison("chr9_150kb_siglocs.txt", "chr9", tab_df.150kb.full, 150)
overlap.chr10.150kb <- signif.comparison("chr10_150kb_siglocs.txt", "chr10", tab_df.150kb.full, 150)
overlap.chr11.150kb <- signif.comparison("chr11_150kb_siglocs.txt", "chr11", tab_df.150kb.full, 150)
overlap.chr12.150kb <- signif.comparison("chr12_150kb_siglocs.txt", "chr12", tab_df.150kb.full, 150)
overlap.chr13.150kb <- signif.comparison("chr13_150kb_siglocs.txt", "chr13", tab_df.150kb.full, 150)
overlap.chr14.150kb <- signif.comparison("chr14_150kb_siglocs.txt", "chr14", tab_df.150kb.full, 150)
overlap.chr15.150kb <- signif.comparison("chr15_150kb_siglocs.txt", "chr15", tab_df.150kb.full, 150)
overlap.chr16.150kb <- signif.comparison("chr16_150kb_siglocs.txt", "chr16", tab_df.150kb.full, 150)
overlap.chr17.150kb <- signif.comparison("chr17_150kb_siglocs.txt", "chr17", tab_df.150kb.full, 150)
overlap.chr18.150kb <- signif.comparison("chr18_150kb_siglocs.txt", "chr18", tab_df.150kb.full, 150)
overlap.chr19.150kb <- signif.comparison("chr19_150kb_siglocs.txt", "chr19", tab_df.150kb.full, 150)
overlap.chr20.150kb <- signif.comparison("chr20_150kb_siglocs.txt", "chr20", tab_df.150kb.full, 150)
overlap.chr21.150kb <- signif.comparison("chr21_150kb_siglocs.txt", "chr21", tab_df.150kb.full, 150)
overlap.chr22.150kb <- signif.comparison("chr22_150kb_siglocs.txt", "chr22", tab_df.150kb.full, 150)
overlap.chrX.150kb <- signif.comparison("chrX_150kb_siglocs.txt", "chrX", tab_df.150kb.full, 150)
overlap.chrY.150kb <- signif.comparison("chrY_150kb_siglocs.txt", "chrY", tab_df.150kb.full, 150)

overlap.150kb.GW <- rbind.fill(overlap.chr1.150kb, overlap.chr2.150kb, overlap.chr3.150kb, overlap.chr4.150kb, overlap.chr5.150kb, overlap.chr6.150kb, overlap.chr7.150kb, overlap.chr8.150kb, overlap.chr9.150kb, overlap.chr10.150kb, overlap.chr11.150kb, overlap.chr12.150kb, overlap.chr13.150kb, overlap.chr14.150kb, overlap.chr15.150kb, overlap.chr16.150kb, overlap.chr17.150kb, overlap.chr18.150kb, overlap.chr19.150kb, overlap.chr20.150kb, overlap.chr21.150kb, overlap.chr22.150kb, overlap.chrX.150kb, overlap.chrY.150kb)
write.table(overlap.150kb.GW, file="/Users/ittaieres/Desktop/Hi-C/All_good_analysis/signif_comparisons/full_signif.150kb", sep="\t", col.names=T, row.names=T, quote=F)



#Get the tabulation DFs with basic count and bin factors for all the different kb "locus" sizes
tab_df.1.200kb <- full.chrom.tabulator(chr1.200kb, pos_df.1, 0, 400000, 10000, "chr1")
tab_df.2.200kb <- full.chrom.tabulator(chr2.200kb, pos_df.2, 0, 400000, 10000, "chr2")
tab_df.3.200kb <- full.chrom.tabulator(chr3.200kb, pos_df.3, 0, 400000, 10000, "chr3")
tab_df.4.200kb <- full.chrom.tabulator(chr4.200kb, pos_df.4, 0, 400000, 10000, "chr4")
tab_df.5.200kb <- full.chrom.tabulator(chr5.200kb, pos_df.5, 0, 400000, 10000, "chr5")
tab_df.6.200kb <- full.chrom.tabulator(chr6.200kb, pos_df.6, 0, 400000, 10000, "chr6")
tab_df.7.200kb <- full.chrom.tabulator(chr7.200kb, pos_df.7, 0, 400000, 10000, "chr7")
tab_df.8.200kb <- full.chrom.tabulator(chr8.200kb, pos_df.8, 0, 400000, 10000, "chr8")
tab_df.9.200kb <- full.chrom.tabulator(chr9.200kb, pos_df.9, 0, 400000, 10000, "chr9")
tab_df.10.200kb <- full.chrom.tabulator(chr10.200kb, pos_df.10, 0, 400000, 10000, "chr10")
tab_df.11.200kb <- full.chrom.tabulator(chr11.200kb, pos_df.11, 0, 400000, 10000, "chr11")
tab_df.12.200kb <- full.chrom.tabulator(chr12.200kb, pos_df.12, 0, 400000, 10000, "chr12")
tab_df.13.200kb <- full.chrom.tabulator(chr13.200kb, pos_df.13, 0, 400000, 10000, "chr13")
tab_df.14.200kb <- full.chrom.tabulator(chr14.200kb, pos_df.14, 0, 400000, 10000, "chr14")
tab_df.15.200kb <- full.chrom.tabulator(chr15.200kb, pos_df.15, 0, 400000, 10000, "chr15")
tab_df.16.200kb <- full.chrom.tabulator(chr16.200kb, pos_df.16, 0, 400000, 10000, "chr16")
tab_df.17.200kb <- full.chrom.tabulator(chr17.200kb, pos_df.17, 0, 400000, 10000, "chr17")
tab_df.18.200kb <- full.chrom.tabulator(chr18.200kb, pos_df.18, 0, 400000, 10000, "chr18")
tab_df.19.200kb <- full.chrom.tabulator(chr19.200kb, pos_df.19, 0, 400000, 10000, "chr19")
tab_df.20.200kb <- full.chrom.tabulator(chr20.200kb, pos_df.20, 0, 400000, 10000, "chr20")
tab_df.21.200kb <- full.chrom.tabulator(chr21.200kb, pos_df.21, 0, 400000, 10000, "chr21")
tab_df.22.200kb <- full.chrom.tabulator(chr22.200kb, pos_df.22, 0, 400000, 10000, "chr22")
tab_df.X.200kb <- full.chrom.tabulator(chrX.200kb, pos_df.X, 0, 400000, 10000, "chrX")
tab_df.Y.200kb <- full.chrom.tabulator(chrY.200kb, pos_df.Y, 0, 400000, 10000, "chrY")

tab_df.200kb.full <- rbind.fill(tab_df.1.200kb, tab_df.2.200kb, tab_df.3.200kb, tab_df.4.200kb, tab_df.5.200kb, tab_df.6.200kb, tab_df.7.200kb, tab_df.8.200kb, tab_df.9.200kb, tab_df.10.200kb, tab_df.11.200kb, tab_df.12.200kb, tab_df.13.200kb, tab_df.14.200kb, tab_df.15.200kb, tab_df.16.200kb, tab_df.17.200kb, tab_df.18.200kb, tab_df.19.200kb, tab_df.20.200kb, tab_df.21.200kb, tab_df.22.200kb, tab_df.X.200kb, tab_df.Y.200kb)
write.table(tab_df.200kb.full, file="/Users/ittaieres/Desktop/Hi-C/All_good_analysis/full_data_frames/full_data.200kb", sep="\t", col.names=T, row.names=F, quote=F)


#Get all the comparison dfs for all chromosomes at locus bins of .200kb

overlap.chr1.200kb <- signif.comparison("chr1_200kb_siglocs.txt", "chr1", tab_df.200kb.full, 200)
overlap.chr2.200kb <- signif.comparison("chr2_200kb_siglocs.txt", "chr2", tab_df.200kb.full, 200)
overlap.chr3.200kb <- signif.comparison("chr3_200kb_siglocs.txt", "chr3", tab_df.200kb.full, 200)
overlap.chr4.200kb <- signif.comparison("chr4_200kb_siglocs.txt", "chr4", tab_df.200kb.full, 200)
overlap.chr5.200kb <- signif.comparison("chr5_200kb_siglocs.txt", "chr5", tab_df.200kb.full, 200)
overlap.chr6.200kb <- signif.comparison("chr6_200kb_siglocs.txt", "chr6", tab_df.200kb.full, 200)
overlap.chr7.200kb <- signif.comparison("chr7_200kb_siglocs.txt", "chr7", tab_df.200kb.full, 200)
overlap.chr8.200kb <- signif.comparison("chr8_200kb_siglocs.txt", "chr8", tab_df.200kb.full, 200)
overlap.chr9.200kb <- signif.comparison("chr9_200kb_siglocs.txt", "chr9", tab_df.200kb.full, 200)
overlap.chr10.200kb <- signif.comparison("chr10_200kb_siglocs.txt", "chr10", tab_df.200kb.full, 200)
overlap.chr11.200kb <- signif.comparison("chr11_200kb_siglocs.txt", "chr11", tab_df.200kb.full, 200)
overlap.chr12.200kb <- signif.comparison("chr12_200kb_siglocs.txt", "chr12", tab_df.200kb.full, 200)
overlap.chr13.200kb <- signif.comparison("chr13_200kb_siglocs.txt", "chr13", tab_df.200kb.full, 200)
overlap.chr14.200kb <- signif.comparison("chr14_200kb_siglocs.txt", "chr14", tab_df.200kb.full, 200)
overlap.chr15.200kb <- signif.comparison("chr15_200kb_siglocs.txt", "chr15", tab_df.200kb.full, 200)
overlap.chr16.200kb <- signif.comparison("chr16_200kb_siglocs.txt", "chr16", tab_df.200kb.full, 200)
overlap.chr17.200kb <- signif.comparison("chr17_200kb_siglocs.txt", "chr17", tab_df.200kb.full, 200)
overlap.chr18.200kb <- signif.comparison("chr18_200kb_siglocs.txt", "chr18", tab_df.200kb.full, 200)
overlap.chr19.200kb <- signif.comparison("chr19_200kb_siglocs.txt", "chr19", tab_df.200kb.full, 200)
overlap.chr20.200kb <- signif.comparison("chr20_200kb_siglocs.txt", "chr20", tab_df.200kb.full, 200)
overlap.chr21.200kb <- signif.comparison("chr21_200kb_siglocs.txt", "chr21", tab_df.200kb.full, 200)
overlap.chr22.200kb <- signif.comparison("chr22_200kb_siglocs.txt", "chr22", tab_df.200kb.full, 200)
overlap.chrX.200kb <- signif.comparison("chrX_200kb_siglocs.txt", "chrX", tab_df.200kb.full, 200)
overlap.chrY.200kb <- signif.comparison("chrY_200kb_siglocs.txt", "chrY", tab_df.200kb.full, 200)

overlap.200kb.GW <- rbind.fill(overlap.chr1.200kb, overlap.chr2.200kb, overlap.chr3.200kb, overlap.chr4.200kb, overlap.chr5.200kb, overlap.chr6.200kb, overlap.chr7.200kb, overlap.chr8.200kb, overlap.chr9.200kb, overlap.chr10.200kb, overlap.chr11.200kb, overlap.chr12.200kb, overlap.chr13.200kb, overlap.chr14.200kb, overlap.chr15.200kb, overlap.chr16.200kb, overlap.chr17.200kb, overlap.chr18.200kb, overlap.chr19.200kb, overlap.chr20.200kb, overlap.chr21.200kb, overlap.chr22.200kb, overlap.chrX.200kb, overlap.chrY.200kb)
write.table(overlap.200kb.GW, file="/Users/ittaieres/Desktop/Hi-C/All_good_analysis/signif_comparisons/full_signif.200kb", sep="\t", col.names=T, row.names=T, quote=F)


```

```{r}

my.ps <- c(10, 5, 1, 0.5, 0.1)
#Get the sig.thresh for .10kb for all the chromosomes.
sig.thresh.1.10kb <- sig.thresh.finder("chr1", tab_df.1.10kb, my.ps)
sig.thresh.2.10kb <- sig.thresh.finder("chr2", tab_df.2.10kb, my.ps)
sig.thresh.3.10kb <- sig.thresh.finder("chr3", tab_df.3.10kb, my.ps)
sig.thresh.4.10kb <- sig.thresh.finder("chr4", tab_df.4.10kb, my.ps)
sig.thresh.5.10kb <- sig.thresh.finder("chr5", tab_df.5.10kb, my.ps)
sig.thresh.6.10kb <- sig.thresh.finder("chr6", tab_df.6.10kb, my.ps)
sig.thresh.7.10kb <- sig.thresh.finder("chr7", tab_df.7.10kb, my.ps)
sig.thresh.8.10kb <- sig.thresh.finder("chr8", tab_df.8.10kb, my.ps)
sig.thresh.9.10kb <- sig.thresh.finder("chr9", tab_df.9.10kb, my.ps)
sig.thresh.10.10kb <- sig.thresh.finder("chr10", tab_df.10.10kb, my.ps)
sig.thresh.11.10kb <- sig.thresh.finder("chr11", tab_df.11.10kb, my.ps)
sig.thresh.12.10kb <- sig.thresh.finder("chr12", tab_df.12.10kb, my.ps)
sig.thresh.13.10kb <- sig.thresh.finder("chr13", tab_df.13.10kb, my.ps)
sig.thresh.14.10kb <- sig.thresh.finder("chr14", tab_df.14.10kb, my.ps)
sig.thresh.15.10kb <- sig.thresh.finder("chr15", tab_df.15.10kb, my.ps)
sig.thresh.16.10kb <- sig.thresh.finder("chr16", tab_df.16.10kb, my.ps)
sig.thresh.17.10kb <- sig.thresh.finder("chr17", tab_df.17.10kb, my.ps)
sig.thresh.18.10kb <- sig.thresh.finder("chr18", tab_df.18.10kb, my.ps)
sig.thresh.19.10kb <- sig.thresh.finder("chr19", tab_df.19.10kb, my.ps)
sig.thresh.20.10kb <- sig.thresh.finder("chr20", tab_df.20.10kb, my.ps)
sig.thresh.21.10kb <- sig.thresh.finder("chr21", tab_df.21.10kb, my.ps)
sig.thresh.22.10kb <- sig.thresh.finder("chr22", tab_df.22.10kb, my.ps)
sig.thresh.X.10kb <- sig.thresh.finder("chrX", tab_df.X.10kb, my.ps)
sig.thresh.Y.10kb <- sig.thresh.finder("chrY", tab_df.Y.10kb, my.ps)
```


```{r Data assignation}
bins.1.10kb <- optimize.bin.sums(chr1.10kb)




#Get the tabulation DFs with basic count and bin factors for all the different kb "locus" sizes
tab_df.1.10kb <- full.chrom.tabulator(pos_df.1, bins.1.10kb, 0, 100000, 5000, "chr1")
tab_df.2.10kb <- full.chrom.tabulator(pos_df.2, bins.2.10kb, 0, 100000, 5000, "chr2")
tab_df.3.10kb <- full.chrom.tabulator(pos_df.3, bins.3.10kb, 0, 100000, 5000, "chr3")
tab_df.4.10kb <- full.chrom.tabulator(pos_df.4, bins.4.10kb, 0, 100000, 5000, "chr4")
tab_df.5.10kb <- full.chrom.tabulator(pos_df.5, bins.5.10kb, 0, 100000, 5000, "chr5")
tab_df.6.10kb <- full.chrom.tabulator(pos_df.6, bins.6.10kb, 0, 100000, 5000, "chr6")
tab_df.7.10kb <- full.chrom.tabulator(pos_df.7, bins.7.10kb, 0, 100000, 5000, "chr7")
tab_df.8.10kb <- full.chrom.tabulator(pos_df.8, bins.8.10kb, 0, 100000, 5000, "chr8")
tab_df.9.10kb <- full.chrom.tabulator(pos_df.9, bins.9.10kb, 0, 100000, 5000, "chr9")
tab_df.10.10kb <- full.chrom.tabulator(pos_df.10, bins.10.10kb, 0, 100000, 5000, "chr10")
tab_df.11.10kb <- full.chrom.tabulator(pos_df.11, bins.11.10kb, 0, 100000, 5000, "chr11")
tab_df.12.10kb <- full.chrom.tabulator(pos_df.12, bins.12.10kb, 0, 100000, 5000, "chr12")
tab_df.13.10kb <- full.chrom.tabulator(pos_df.13, bins.13.10kb, 0, 100000, 5000, "chr13")
tab_df.14.10kb <- full.chrom.tabulator(pos_df.14, bins.14.10kb, 0, 100000, 5000, "chr14")
tab_df.15.10kb <- full.chrom.tabulator(pos_df.15, bins.15.10kb, 0, 100000, 5000, "chr15")
tab_df.16.10kb <- full.chrom.tabulator(pos_df.16, bins.16.10kb, 0, 100000, 5000, "chr16")
tab_df.17.10kb <- full.chrom.tabulator(pos_df.17, bins.17.10kb, 0, 100000, 5000, "chr17")
tab_df.18.10kb <- full.chrom.tabulator(pos_df.18, bins.18.10kb, 0, 100000, 5000, "chr18")
tab_df.19.10kb <- full.chrom.tabulator(pos_df.19, bins.19.10kb, 0, 100000, 5000, "chr19")
tab_df.20.10kb <- full.chrom.tabulator(pos_df.20, bins.20.10kb, 0, 100000, 5000, "chr20")
tab_df.21.10kb <- full.chrom.tabulator(pos_df.21, bins.21.10kb, 0, 100000, 5000, "chr21")
tab_df.22.10kb <- full.chrom.tabulator(pos_df.22, bins.22.10kb, 0, 100000, 5000, "chr22")
tab_df.X.10kb <- full.chrom.tabulator(pos_df.X, bins.X.10kb, 0, 100000, 5000, "chrX")
tab_df.Y.10kb <- full.chrom.tabulator(pos_df.Y, bins.Y.10kb, 0, 100000, 5000, "chrY")


tab_df.1.20kb <- full.chrom.tabulator(pos_df.1, bins.1.20kb, 0, 100000, 5000, "chr1")
tab_df.2.20kb <- full.chrom.tabulator(pos_df.2, bins.2.20kb, 0, 100000, 5000, "chr2")
tab_df.3.20kb <- full.chrom.tabulator(pos_df.3, bins.3.20kb, 0, 100000, 5000, "chr3")
tab_df.4.20kb <- full.chrom.tabulator(pos_df.4, bins.4.20kb, 0, 100000, 5000, "chr4")
tab_df.5.20kb <- full.chrom.tabulator(pos_df.5, bins.5.20kb, 0, 100000, 5000, "chr5")
tab_df.6.20kb <- full.chrom.tabulator(pos_df.6, bins.6.20kb, 0, 100000, 5000, "chr6")
tab_df.7.20kb <- full.chrom.tabulator(pos_df.7, bins.7.20kb, 0, 100000, 5000, "chr7")
tab_df.8.20kb <- full.chrom.tabulator(pos_df.8, bins.8.20kb, 0, 100000, 5000, "chr8")
tab_df.9.20kb <- full.chrom.tabulator(pos_df.9, bins.9.20kb, 0, 100000, 5000, "chr9")
tab_df.10.20kb <- full.chrom.tabulator(pos_df.10, bins.10.20kb, 0, 100000, 5000, "chr10")
tab_df.11.20kb <- full.chrom.tabulator(pos_df.11, bins.11.20kb, 0, 100000, 5000, "chr11")
tab_df.12.20kb <- full.chrom.tabulator(pos_df.12, bins.12.20kb, 0, 100000, 5000, "chr12")
tab_df.13.20kb <- full.chrom.tabulator(pos_df.13, bins.13.20kb, 0, 100000, 5000, "chr13")
tab_df.14.20kb <- full.chrom.tabulator(pos_df.14, bins.14.20kb, 0, 100000, 5000, "chr14")
tab_df.15.20kb <- full.chrom.tabulator(pos_df.15, bins.15.20kb, 0, 100000, 5000, "chr15")
tab_df.16.20kb <- full.chrom.tabulator(pos_df.16, bins.16.20kb, 0, 100000, 5000, "chr16")
tab_df.17.20kb <- full.chrom.tabulator(pos_df.17, bins.17.20kb, 0, 100000, 5000, "chr17")
tab_df.18.20kb <- full.chrom.tabulator(pos_df.18, bins.18.20kb, 0, 100000, 5000, "chr18")
tab_df.19.20kb <- full.chrom.tabulator(pos_df.19, bins.19.20kb, 0, 100000, 5000, "chr19")
tab_df.20.20kb <- full.chrom.tabulator(pos_df.20, bins.20.20kb, 0, 100000, 5000, "chr20")
tab_df.21.20kb <- full.chrom.tabulator(pos_df.21, bins.21.20kb, 0, 100000, 5000, "chr21")
tab_df.22.20kb <- full.chrom.tabulator(pos_df.22, bins.22.20kb, 0, 100000, 5000, "chr22")
tab_df.X.20kb <- full.chrom.tabulator(pos_df.X, bins.X.20kb, 0, 100000, 5000, "chrX")
tab_df.Y.20kb <- full.chrom.tabulator(pos_df.Y, bins.Y.20kb, 0, 100000, 5000, "chrY")


tab_df.1.30kb <- full.chrom.tabulator(pos_df.1, bins.1.30kb, 0, 100000, 5000, "chr1")
tab_df.2.30kb <- full.chrom.tabulator(pos_df.2, bins.2.30kb, 0, 100000, 5000, "chr2")
tab_df.3.30kb <- full.chrom.tabulator(pos_df.3, bins.3.30kb, 0, 100000, 5000, "chr3")
tab_df.4.30kb <- full.chrom.tabulator(pos_df.4, bins.4.30kb, 0, 100000, 5000, "chr4")
tab_df.5.30kb <- full.chrom.tabulator(pos_df.5, bins.5.30kb, 0, 100000, 5000, "chr5")
tab_df.6.30kb <- full.chrom.tabulator(pos_df.6, bins.6.30kb, 0, 100000, 5000, "chr6")
tab_df.7.30kb <- full.chrom.tabulator(pos_df.7, bins.7.30kb, 0, 100000, 5000, "chr7")
tab_df.8.30kb <- full.chrom.tabulator(pos_df.8, bins.8.30kb, 0, 100000, 5000, "chr8")
tab_df.9.30kb <- full.chrom.tabulator(pos_df.9, bins.9.30kb, 0, 100000, 5000, "chr9")
tab_df.10.30kb <- full.chrom.tabulator(pos_df.10, bins.10.30kb, 0, 100000, 5000, "chr10")
tab_df.11.30kb <- full.chrom.tabulator(pos_df.11, bins.11.30kb, 0, 100000, 5000, "chr11")
tab_df.12.30kb <- full.chrom.tabulator(pos_df.12, bins.12.30kb, 0, 100000, 5000, "chr12")
tab_df.13.30kb <- full.chrom.tabulator(pos_df.13, bins.13.30kb, 0, 100000, 5000, "chr13")
tab_df.14.30kb <- full.chrom.tabulator(pos_df.14, bins.14.30kb, 0, 100000, 5000, "chr14")
tab_df.15.30kb <- full.chrom.tabulator(pos_df.15, bins.15.30kb, 0, 100000, 5000, "chr15")
tab_df.16.30kb <- full.chrom.tabulator(pos_df.16, bins.16.30kb, 0, 100000, 5000, "chr16")
tab_df.17.30kb <- full.chrom.tabulator(pos_df.17, bins.17.30kb, 0, 100000, 5000, "chr17")
tab_df.18.30kb <- full.chrom.tabulator(pos_df.18, bins.18.30kb, 0, 100000, 5000, "chr18")
tab_df.19.30kb <- full.chrom.tabulator(pos_df.19, bins.19.30kb, 0, 100000, 5000, "chr19")
tab_df.20.30kb <- full.chrom.tabulator(pos_df.20, bins.20.30kb, 0, 100000, 5000, "chr20")
tab_df.21.30kb <- full.chrom.tabulator(pos_df.21, bins.21.30kb, 0, 100000, 5000, "chr21")
tab_df.22.30kb <- full.chrom.tabulator(pos_df.22, bins.22.30kb, 0, 100000, 5000, "chr22")
tab_df.X.30kb <- full.chrom.tabulator(pos_df.X, bins.X.30kb, 0, 100000, 5000, "chrX")
tab_df.Y.30kb <- full.chrom.tabulator(pos_df.Y, bins.Y.30kb, 0, 100000, 5000, "chrY")


tab_df.1.40kb <- full.chrom.tabulator(pos_df.1, bins.1.40kb, 0, 100000, 5000, "chr1")
tab_df.2.40kb <- full.chrom.tabulator(pos_df.2, bins.2.40kb, 0, 100000, 5000, "chr2")
tab_df.3.40kb <- full.chrom.tabulator(pos_df.3, bins.3.40kb, 0, 100000, 5000, "chr3")
tab_df.4.40kb <- full.chrom.tabulator(pos_df.4, bins.4.40kb, 0, 100000, 5000, "chr4")
tab_df.5.40kb <- full.chrom.tabulator(pos_df.5, bins.5.40kb, 0, 100000, 5000, "chr5")
tab_df.6.40kb <- full.chrom.tabulator(pos_df.6, bins.6.40kb, 0, 100000, 5000, "chr6")
tab_df.7.40kb <- full.chrom.tabulator(pos_df.7, bins.7.40kb, 0, 100000, 5000, "chr7")
tab_df.8.40kb <- full.chrom.tabulator(pos_df.8, bins.8.40kb, 0, 100000, 5000, "chr8")
tab_df.9.40kb <- full.chrom.tabulator(pos_df.9, bins.9.40kb, 0, 100000, 5000, "chr9")
tab_df.10.40kb <- full.chrom.tabulator(pos_df.10, bins.10.40kb, 0, 100000, 5000, "chr10")
tab_df.11.40kb <- full.chrom.tabulator(pos_df.11, bins.11.40kb, 0, 100000, 5000, "chr11")
tab_df.12.40kb <- full.chrom.tabulator(pos_df.12, bins.12.40kb, 0, 100000, 5000, "chr12")
tab_df.13.40kb <- full.chrom.tabulator(pos_df.13, bins.13.40kb, 0, 100000, 5000, "chr13")
tab_df.14.40kb <- full.chrom.tabulator(pos_df.14, bins.14.40kb, 0, 100000, 5000, "chr14")
tab_df.15.40kb <- full.chrom.tabulator(pos_df.15, bins.15.40kb, 0, 100000, 5000, "chr15")
tab_df.16.40kb <- full.chrom.tabulator(pos_df.16, bins.16.40kb, 0, 100000, 5000, "chr16")
tab_df.17.40kb <- full.chrom.tabulator(pos_df.17, bins.17.40kb, 0, 100000, 5000, "chr17")
tab_df.18.40kb <- full.chrom.tabulator(pos_df.18, bins.18.40kb, 0, 100000, 5000, "chr18")
tab_df.19.40kb <- full.chrom.tabulator(pos_df.19, bins.19.40kb, 0, 100000, 5000, "chr19")
tab_df.20.40kb <- full.chrom.tabulator(pos_df.20, bins.20.40kb, 0, 100000, 5000, "chr20")
tab_df.21.40kb <- full.chrom.tabulator(pos_df.21, bins.21.40kb, 0, 100000, 5000, "chr21")
tab_df.22.40kb <- full.chrom.tabulator(pos_df.22, bins.22.40kb, 0, 100000, 5000, "chr22")
tab_df.X.40kb <- full.chrom.tabulator(pos_df.X, bins.X.40kb, 0, 100000, 5000, "chrX")
tab_df.Y.40kb <- full.chrom.tabulator(pos_df.Y, bins.Y.40kb, 0, 100000, 5000, "chrY")


tab_df.1.50kb <- full.chrom.tabulator(pos_df.1, bins.1.50kb, 0, 200000, 10000, "chr1")
tab_df.2.50kb <- full.chrom.tabulator(pos_df.2, bins.2.50kb, 0, 200000, 10000, "chr2")
tab_df.3.50kb <- full.chrom.tabulator(pos_df.3, bins.3.50kb, 0, 200000, 10000, "chr3")
tab_df.4.50kb <- full.chrom.tabulator(pos_df.4, bins.4.50kb, 0, 200000, 10000, "chr4")
tab_df.5.50kb <- full.chrom.tabulator(pos_df.5, bins.5.50kb, 0, 200000, 10000, "chr5")
tab_df.6.50kb <- full.chrom.tabulator(pos_df.6, bins.6.50kb, 0, 200000, 10000, "chr6")
tab_df.7.50kb <- full.chrom.tabulator(pos_df.7, bins.7.50kb, 0, 200000, 10000, "chr7")
tab_df.8.50kb <- full.chrom.tabulator(pos_df.8, bins.8.50kb, 0, 200000, 10000, "chr8")
tab_df.9.50kb <- full.chrom.tabulator(pos_df.9, bins.9.50kb, 0, 200000, 10000, "chr9")
tab_df.10.50kb <- full.chrom.tabulator(pos_df.10, bins.10.50kb, 0, 200000, 10000, "chr10")
tab_df.11.50kb <- full.chrom.tabulator(pos_df.11, bins.11.50kb, 0, 200000, 10000, "chr11")
tab_df.12.50kb <- full.chrom.tabulator(pos_df.12, bins.12.50kb, 0, 200000, 10000, "chr12")
tab_df.13.50kb <- full.chrom.tabulator(pos_df.13, bins.13.50kb, 0, 200000, 10000, "chr13")
tab_df.14.50kb <- full.chrom.tabulator(pos_df.14, bins.14.50kb, 0, 200000, 10000, "chr14")
tab_df.15.50kb <- full.chrom.tabulator(pos_df.15, bins.15.50kb, 0, 200000, 10000, "chr15")
tab_df.16.50kb <- full.chrom.tabulator(pos_df.16, bins.16.50kb, 0, 200000, 10000, "chr16")
tab_df.17.50kb <- full.chrom.tabulator(pos_df.17, bins.17.50kb, 0, 200000, 10000, "chr17")
tab_df.18.50kb <- full.chrom.tabulator(pos_df.18, bins.18.50kb, 0, 200000, 10000, "chr18")
tab_df.19.50kb <- full.chrom.tabulator(pos_df.19, bins.19.50kb, 0, 200000, 10000, "chr19")
tab_df.20.50kb <- full.chrom.tabulator(pos_df.20, bins.20.50kb, 0, 200000, 10000, "chr20")
tab_df.21.50kb <- full.chrom.tabulator(pos_df.21, bins.21.50kb, 0, 200000, 10000, "chr21")
tab_df.22.50kb <- full.chrom.tabulator(pos_df.22, bins.22.50kb, 0, 200000, 10000, "chr22")
tab_df.X.50kb <- full.chrom.tabulator(pos_df.X, bins.X.50kb, 0, 200000, 10000, "chrX")
tab_df.Y.50kb <- full.chrom.tabulator(pos_df.Y, bins.Y.50kb, 0, 200000, 10000, "chrY")


tab_df.1.60kb <- full.chrom.tabulator(pos_df.1, bins.1.60kb, 0, 250000, 10000, "chr1")
tab_df.2.60kb <- full.chrom.tabulator(pos_df.2, bins.2.60kb, 0, 250000, 10000, "chr2")
tab_df.3.60kb <- full.chrom.tabulator(pos_df.3, bins.3.60kb, 0, 250000, 10000, "chr3")
tab_df.4.60kb <- full.chrom.tabulator(pos_df.4, bins.4.60kb, 0, 250000, 10000, "chr4")
tab_df.5.60kb <- full.chrom.tabulator(pos_df.5, bins.5.60kb, 0, 250000, 10000, "chr5")
tab_df.6.60kb <- full.chrom.tabulator(pos_df.6, bins.6.60kb, 0, 250000, 10000, "chr6")
tab_df.7.60kb <- full.chrom.tabulator(pos_df.7, bins.7.60kb, 0, 250000, 10000, "chr7")
tab_df.8.60kb <- full.chrom.tabulator(pos_df.8, bins.8.60kb, 0, 250000, 10000, "chr8")
tab_df.9.60kb <- full.chrom.tabulator(pos_df.9, bins.9.60kb, 0, 250000, 10000, "chr9")
tab_df.10.60kb <- full.chrom.tabulator(pos_df.10, bins.10.60kb, 0, 250000, 10000, "chr10")
tab_df.11.60kb <- full.chrom.tabulator(pos_df.11, bins.11.60kb, 0, 250000, 10000, "chr11")
tab_df.12.60kb <- full.chrom.tabulator(pos_df.12, bins.12.60kb, 0, 250000, 10000, "chr12")
tab_df.13.60kb <- full.chrom.tabulator(pos_df.13, bins.13.60kb, 0, 250000, 10000, "chr13")
tab_df.14.60kb <- full.chrom.tabulator(pos_df.14, bins.14.60kb, 0, 250000, 10000, "chr14")
tab_df.15.60kb <- full.chrom.tabulator(pos_df.15, bins.15.60kb, 0, 250000, 10000, "chr15")
tab_df.16.60kb <- full.chrom.tabulator(pos_df.16, bins.16.60kb, 0, 250000, 10000, "chr16")
tab_df.17.60kb <- full.chrom.tabulator(pos_df.17, bins.17.60kb, 0, 250000, 10000, "chr17")
tab_df.18.60kb <- full.chrom.tabulator(pos_df.18, bins.18.60kb, 0, 250000, 10000, "chr18")
tab_df.19.60kb <- full.chrom.tabulator(pos_df.19, bins.19.60kb, 0, 250000, 10000, "chr19")
tab_df.20.60kb <- full.chrom.tabulator(pos_df.20, bins.20.60kb, 0, 250000, 10000, "chr20")
tab_df.21.60kb <- full.chrom.tabulator(pos_df.21, bins.21.60kb, 0, 250000, 10000, "chr21")
tab_df.22.60kb <- full.chrom.tabulator(pos_df.22, bins.22.60kb, 0, 250000, 10000, "chr22")
tab_df.X.60kb <- full.chrom.tabulator(pos_df.X, bins.X.60kb, 0, 250000, 10000, "chrX")
tab_df.Y.60kb <- full.chrom.tabulator(pos_df.Y, bins.Y.60kb, 0, 250000, 10000, "chrY")


tab_df.1.70kb <- full.chrom.tabulator(pos_df.1, bins.1.70kb, 0, 300000, 10000, "chr1")
tab_df.2.70kb <- full.chrom.tabulator(pos_df.2, bins.2.70kb, 0, 300000, 10000, "chr2")
tab_df.3.70kb <- full.chrom.tabulator(pos_df.3, bins.3.70kb, 0, 300000, 10000, "chr3")
tab_df.4.70kb <- full.chrom.tabulator(pos_df.4, bins.4.70kb, 0, 300000, 10000, "chr4")
tab_df.5.70kb <- full.chrom.tabulator(pos_df.5, bins.5.70kb, 0, 300000, 10000, "chr5")
tab_df.6.70kb <- full.chrom.tabulator(pos_df.6, bins.6.70kb, 0, 300000, 10000, "chr6")
tab_df.7.70kb <- full.chrom.tabulator(pos_df.7, bins.7.70kb, 0, 300000, 10000, "chr7")
tab_df.8.70kb <- full.chrom.tabulator(pos_df.8, bins.8.70kb, 0, 300000, 10000, "chr8")
tab_df.9.70kb <- full.chrom.tabulator(pos_df.9, bins.9.70kb, 0, 300000, 10000, "chr9")
tab_df.10.70kb <- full.chrom.tabulator(pos_df.10, bins.10.70kb, 0, 300000, 10000, "chr10")
tab_df.11.70kb <- full.chrom.tabulator(pos_df.11, bins.11.70kb, 0, 300000, 10000, "chr11")
tab_df.12.70kb <- full.chrom.tabulator(pos_df.12, bins.12.70kb, 0, 300000, 10000, "chr12")
tab_df.13.70kb <- full.chrom.tabulator(pos_df.13, bins.13.70kb, 0, 300000, 10000, "chr13")
tab_df.14.70kb <- full.chrom.tabulator(pos_df.14, bins.14.70kb, 0, 300000, 10000, "chr14")
tab_df.15.70kb <- full.chrom.tabulator(pos_df.15, bins.15.70kb, 0, 300000, 10000, "chr15")
tab_df.16.70kb <- full.chrom.tabulator(pos_df.16, bins.16.70kb, 0, 300000, 10000, "chr16")
tab_df.17.70kb <- full.chrom.tabulator(pos_df.17, bins.17.70kb, 0, 300000, 10000, "chr17")
tab_df.18.70kb <- full.chrom.tabulator(pos_df.18, bins.18.70kb, 0, 300000, 10000, "chr18")
tab_df.19.70kb <- full.chrom.tabulator(pos_df.19, bins.19.70kb, 0, 300000, 10000, "chr19")
tab_df.20.70kb <- full.chrom.tabulator(pos_df.20, bins.20.70kb, 0, 300000, 10000, "chr20")
tab_df.21.70kb <- full.chrom.tabulator(pos_df.21, bins.21.70kb, 0, 300000, 10000, "chr21")
tab_df.22.70kb <- full.chrom.tabulator(pos_df.22, bins.22.70kb, 0, 300000, 10000, "chr22")
tab_df.X.70kb <- full.chrom.tabulator(pos_df.X, bins.X.70kb, 0, 300000, 10000, "chrX")
tab_df.Y.70kb <- full.chrom.tabulator(pos_df.Y, bins.Y.70kb, 0, 300000, 10000, "chrY")


tab_df.1.80kb <- full.chrom.tabulator(pos_df.1, bins.1.80kb, 0, 350000, 10000, "chr1")
tab_df.2.80kb <- full.chrom.tabulator(pos_df.2, bins.2.80kb, 0, 350000, 10000, "chr2")
tab_df.3.80kb <- full.chrom.tabulator(pos_df.3, bins.3.80kb, 0, 350000, 10000, "chr3")
tab_df.4.80kb <- full.chrom.tabulator(pos_df.4, bins.4.80kb, 0, 350000, 10000, "chr4")
tab_df.5.80kb <- full.chrom.tabulator(pos_df.5, bins.5.80kb, 0, 350000, 10000, "chr5")
tab_df.6.80kb <- full.chrom.tabulator(pos_df.6, bins.6.80kb, 0, 350000, 10000, "chr6")
tab_df.7.80kb <- full.chrom.tabulator(pos_df.7, bins.7.80kb, 0, 350000, 10000, "chr7")
tab_df.8.80kb <- full.chrom.tabulator(pos_df.8, bins.8.80kb, 0, 350000, 10000, "chr8")
tab_df.9.80kb <- full.chrom.tabulator(pos_df.9, bins.9.80kb, 0, 350000, 10000, "chr9")
tab_df.10.80kb <- full.chrom.tabulator(pos_df.10, bins.10.80kb, 0, 350000, 10000, "chr10")
tab_df.11.80kb <- full.chrom.tabulator(pos_df.11, bins.11.80kb, 0, 350000, 10000, "chr11")
tab_df.12.80kb <- full.chrom.tabulator(pos_df.12, bins.12.80kb, 0, 350000, 10000, "chr12")
tab_df.13.80kb <- full.chrom.tabulator(pos_df.13, bins.13.80kb, 0, 350000, 10000, "chr13")
tab_df.14.80kb <- full.chrom.tabulator(pos_df.14, bins.14.80kb, 0, 350000, 10000, "chr14")
tab_df.15.80kb <- full.chrom.tabulator(pos_df.15, bins.15.80kb, 0, 350000, 10000, "chr15")
tab_df.16.80kb <- full.chrom.tabulator(pos_df.16, bins.16.80kb, 0, 350000, 10000, "chr16")
tab_df.17.80kb <- full.chrom.tabulator(pos_df.17, bins.17.80kb, 0, 350000, 10000, "chr17")
tab_df.18.80kb <- full.chrom.tabulator(pos_df.18, bins.18.80kb, 0, 350000, 10000, "chr18")
tab_df.19.80kb <- full.chrom.tabulator(pos_df.19, bins.19.80kb, 0, 350000, 10000, "chr19")
tab_df.20.80kb <- full.chrom.tabulator(pos_df.20, bins.20.80kb, 0, 350000, 10000, "chr20")
tab_df.21.80kb <- full.chrom.tabulator(pos_df.21, bins.21.80kb, 0, 350000, 10000, "chr21")
tab_df.22.80kb <- full.chrom.tabulator(pos_df.22, bins.22.80kb, 0, 350000, 10000, "chr22")
tab_df.X.80kb <- full.chrom.tabulator(pos_df.X, bins.X.80kb, 0, 350000, 10000, "chrX")
tab_df.Y.80kb <- full.chrom.tabulator(pos_df.Y, bins.Y.80kb, 0, 350000, 10000, "chrY")


tab_df.1.90kb <- full.chrom.tabulator(pos_df.1, bins.1.90kb, 0, 400000, 10000, "chr1")
tab_df.2.90kb <- full.chrom.tabulator(pos_df.2, bins.2.90kb, 0, 400000, 10000, "chr2")
tab_df.3.90kb <- full.chrom.tabulator(pos_df.3, bins.3.90kb, 0, 400000, 10000, "chr3")
tab_df.4.90kb <- full.chrom.tabulator(pos_df.4, bins.4.90kb, 0, 400000, 10000, "chr4")
tab_df.5.90kb <- full.chrom.tabulator(pos_df.5, bins.5.90kb, 0, 400000, 10000, "chr5")
tab_df.6.90kb <- full.chrom.tabulator(pos_df.6, bins.6.90kb, 0, 400000, 10000, "chr6")
tab_df.7.90kb <- full.chrom.tabulator(pos_df.7, bins.7.90kb, 0, 400000, 10000, "chr7")
tab_df.8.90kb <- full.chrom.tabulator(pos_df.8, bins.8.90kb, 0, 400000, 10000, "chr8")
tab_df.9.90kb <- full.chrom.tabulator(pos_df.9, bins.9.90kb, 0, 400000, 10000, "chr9")
tab_df.10.90kb <- full.chrom.tabulator(pos_df.10, bins.10.90kb, 0, 400000, 10000, "chr10")
tab_df.11.90kb <- full.chrom.tabulator(pos_df.11, bins.11.90kb, 0, 400000, 10000, "chr11")
tab_df.12.90kb <- full.chrom.tabulator(pos_df.12, bins.12.90kb, 0, 400000, 10000, "chr12")
tab_df.13.90kb <- full.chrom.tabulator(pos_df.13, bins.13.90kb, 0, 400000, 10000, "chr13")
tab_df.14.90kb <- full.chrom.tabulator(pos_df.14, bins.14.90kb, 0, 400000, 10000, "chr14")
tab_df.15.90kb <- full.chrom.tabulator(pos_df.15, bins.15.90kb, 0, 400000, 10000, "chr15")
tab_df.16.90kb <- full.chrom.tabulator(pos_df.16, bins.16.90kb, 0, 400000, 10000, "chr16")
tab_df.17.90kb <- full.chrom.tabulator(pos_df.17, bins.17.90kb, 0, 400000, 10000, "chr17")
tab_df.18.90kb <- full.chrom.tabulator(pos_df.18, bins.18.90kb, 0, 400000, 10000, "chr18")
tab_df.19.90kb <- full.chrom.tabulator(pos_df.19, bins.19.90kb, 0, 400000, 10000, "chr19")
tab_df.20.90kb <- full.chrom.tabulator(pos_df.20, bins.20.90kb, 0, 400000, 10000, "chr20")
tab_df.21.90kb <- full.chrom.tabulator(pos_df.21, bins.21.90kb, 0, 400000, 10000, "chr21")
tab_df.22.90kb <- full.chrom.tabulator(pos_df.22, bins.22.90kb, 0, 400000, 10000, "chr22")
tab_df.X.90kb <- full.chrom.tabulator(pos_df.X, bins.X.90kb, 0, 400000, 10000, "chrX")
tab_df.Y.90kb <- full.chrom.tabulator(pos_df.Y, bins.Y.90kb, 0, 400000, 10000, "chrY")


tab_df.1.100kb <- full.chrom.tabulator(pos_df.1, bins.1.100kb, 0, 500000, 25000, "chr1")
tab_df.2.100kb <- full.chrom.tabulator(pos_df.2, bins.2.100kb, 0, 500000, 25000, "chr2")
tab_df.3.100kb <- full.chrom.tabulator(pos_df.3, bins.3.100kb, 0, 500000, 25000, "chr3")
tab_df.4.100kb <- full.chrom.tabulator(pos_df.4, bins.4.100kb, 0, 500000, 25000, "chr4")
tab_df.5.100kb <- full.chrom.tabulator(pos_df.5, bins.5.100kb, 0, 500000, 25000, "chr5")
tab_df.6.100kb <- full.chrom.tabulator(pos_df.6, bins.6.100kb, 0, 500000, 25000, "chr6")
tab_df.7.100kb <- full.chrom.tabulator(pos_df.7, bins.7.100kb, 0, 500000, 25000, "chr7")
tab_df.8.100kb <- full.chrom.tabulator(pos_df.8, bins.8.100kb, 0, 500000, 25000, "chr8")
tab_df.9.100kb <- full.chrom.tabulator(pos_df.9, bins.9.100kb, 0, 500000, 25000, "chr9")
tab_df.10.100kb <- full.chrom.tabulator(pos_df.10, bins.10.100kb, 0, 500000, 25000, "chr10")
tab_df.11.100kb <- full.chrom.tabulator(pos_df.11, bins.11.100kb, 0, 500000, 25000, "chr11")
tab_df.12.100kb <- full.chrom.tabulator(pos_df.12, bins.12.100kb, 0, 500000, 25000, "chr12")
tab_df.13.100kb <- full.chrom.tabulator(pos_df.13, bins.13.100kb, 0, 500000, 25000, "chr13")
tab_df.14.100kb <- full.chrom.tabulator(pos_df.14, bins.14.100kb, 0, 500000, 25000, "chr14")
tab_df.15.100kb <- full.chrom.tabulator(pos_df.15, bins.15.100kb, 0, 500000, 25000, "chr15")
tab_df.16.100kb <- full.chrom.tabulator(pos_df.16, bins.16.100kb, 0, 500000, 25000, "chr16")
tab_df.17.100kb <- full.chrom.tabulator(pos_df.17, bins.17.100kb, 0, 500000, 25000, "chr17")
tab_df.18.100kb <- full.chrom.tabulator(pos_df.18, bins.18.100kb, 0, 500000, 25000, "chr18")
tab_df.19.100kb <- full.chrom.tabulator(pos_df.19, bins.19.100kb, 0, 500000, 25000, "chr19")
tab_df.20.100kb <- full.chrom.tabulator(pos_df.20, bins.20.100kb, 0, 500000, 25000, "chr20")
tab_df.21.100kb <- full.chrom.tabulator(pos_df.21, bins.21.100kb, 0, 500000, 25000, "chr21")
tab_df.22.100kb <- full.chrom.tabulator(pos_df.22, bins.22.100kb, 0, 500000, 25000, "chr22")
tab_df.X.100kb <- full.chrom.tabulator(pos_df.X, bins.X.100kb, 0, 500000, 25000, "chrX")
tab_df.Y.100kb <- full.chrom.tabulator(pos_df.Y, bins.Y.100kb, 0, 500000, 25000, "chrY")


tab_df.1.125kb <- full.chrom.tabulator(pos_df.1, bins.1.125kb, 0, 600000, 25000, "chr1")
tab_df.2.125kb <- full.chrom.tabulator(pos_df.2, bins.2.125kb, 0, 600000, 25000, "chr2")
tab_df.3.125kb <- full.chrom.tabulator(pos_df.3, bins.3.125kb, 0, 600000, 25000, "chr3")
tab_df.4.125kb <- full.chrom.tabulator(pos_df.4, bins.4.125kb, 0, 600000, 25000, "chr4")
tab_df.5.125kb <- full.chrom.tabulator(pos_df.5, bins.5.125kb, 0, 600000, 25000, "chr5")
tab_df.6.125kb <- full.chrom.tabulator(pos_df.6, bins.6.125kb, 0, 600000, 25000, "chr6")
tab_df.7.125kb <- full.chrom.tabulator(pos_df.7, bins.7.125kb, 0, 600000, 25000, "chr7")
tab_df.8.125kb <- full.chrom.tabulator(pos_df.8, bins.8.125kb, 0, 600000, 25000, "chr8")
tab_df.9.125kb <- full.chrom.tabulator(pos_df.9, bins.9.125kb, 0, 600000, 25000, "chr9")
tab_df.10.125kb <- full.chrom.tabulator(pos_df.10, bins.10.125kb, 0, 600000, 25000, "chr10")
tab_df.11.125kb <- full.chrom.tabulator(pos_df.11, bins.11.125kb, 0, 600000, 25000, "chr11")
tab_df.12.125kb <- full.chrom.tabulator(pos_df.12, bins.12.125kb, 0, 600000, 25000, "chr12")
tab_df.13.125kb <- full.chrom.tabulator(pos_df.13, bins.13.125kb, 0, 600000, 25000, "chr13")
tab_df.14.125kb <- full.chrom.tabulator(pos_df.14, bins.14.125kb, 0, 600000, 25000, "chr14")
tab_df.15.125kb <- full.chrom.tabulator(pos_df.15, bins.15.125kb, 0, 600000, 25000, "chr15")
tab_df.16.125kb <- full.chrom.tabulator(pos_df.16, bins.16.125kb, 0, 600000, 25000, "chr16")
tab_df.17.125kb <- full.chrom.tabulator(pos_df.17, bins.17.125kb, 0, 600000, 25000, "chr17")
tab_df.18.125kb <- full.chrom.tabulator(pos_df.18, bins.18.125kb, 0, 600000, 25000, "chr18")
tab_df.19.125kb <- full.chrom.tabulator(pos_df.19, bins.19.125kb, 0, 600000, 25000, "chr19")
tab_df.20.125kb <- full.chrom.tabulator(pos_df.20, bins.20.125kb, 0, 600000, 25000, "chr20")
tab_df.21.125kb <- full.chrom.tabulator(pos_df.21, bins.21.125kb, 0, 600000, 25000, "chr21")
tab_df.22.125kb <- full.chrom.tabulator(pos_df.22, bins.22.125kb, 0, 600000, 25000, "chr22")
tab_df.X.125kb <- full.chrom.tabulator(pos_df.X, bins.X.125kb, 0, 600000, 25000, "chrX")
tab_df.Y.125kb <- full.chrom.tabulator(pos_df.Y, bins.Y.125kb, 0, 600000, 25000, "chrY")


tab_df.1.150kb <- full.chrom.tabulator(pos_df.1, bins.1.150kb, 0, 700000, 25000, "chr1")
tab_df.2.150kb <- full.chrom.tabulator(pos_df.2, bins.2.150kb, 0, 700000, 25000, "chr2")
tab_df.3.150kb <- full.chrom.tabulator(pos_df.3, bins.3.150kb, 0, 700000, 25000, "chr3")
tab_df.4.150kb <- full.chrom.tabulator(pos_df.4, bins.4.150kb, 0, 700000, 25000, "chr4")
tab_df.5.150kb <- full.chrom.tabulator(pos_df.5, bins.5.150kb, 0, 700000, 25000, "chr5")
tab_df.6.150kb <- full.chrom.tabulator(pos_df.6, bins.6.150kb, 0, 700000, 25000, "chr6")
tab_df.7.150kb <- full.chrom.tabulator(pos_df.7, bins.7.150kb, 0, 700000, 25000, "chr7")
tab_df.8.150kb <- full.chrom.tabulator(pos_df.8, bins.8.150kb, 0, 700000, 25000, "chr8")
tab_df.9.150kb <- full.chrom.tabulator(pos_df.9, bins.9.150kb, 0, 700000, 25000, "chr9")
tab_df.10.150kb <- full.chrom.tabulator(pos_df.10, bins.10.150kb, 0, 700000, 25000, "chr10")
tab_df.11.150kb <- full.chrom.tabulator(pos_df.11, bins.11.150kb, 0, 700000, 25000, "chr11")
tab_df.12.150kb <- full.chrom.tabulator(pos_df.12, bins.12.150kb, 0, 700000, 25000, "chr12")
tab_df.13.150kb <- full.chrom.tabulator(pos_df.13, bins.13.150kb, 0, 700000, 25000, "chr13")
tab_df.14.150kb <- full.chrom.tabulator(pos_df.14, bins.14.150kb, 0, 700000, 25000, "chr14")
tab_df.15.150kb <- full.chrom.tabulator(pos_df.15, bins.15.150kb, 0, 700000, 25000, "chr15")
tab_df.16.150kb <- full.chrom.tabulator(pos_df.16, bins.16.150kb, 0, 700000, 25000, "chr16")
tab_df.17.150kb <- full.chrom.tabulator(pos_df.17, bins.17.150kb, 0, 700000, 25000, "chr17")
tab_df.18.150kb <- full.chrom.tabulator(pos_df.18, bins.18.150kb, 0, 700000, 25000, "chr18")
tab_df.19.150kb <- full.chrom.tabulator(pos_df.19, bins.19.150kb, 0, 700000, 25000, "chr19")
tab_df.20.150kb <- full.chrom.tabulator(pos_df.20, bins.20.150kb, 0, 700000, 25000, "chr20")
tab_df.21.150kb <- full.chrom.tabulator(pos_df.21, bins.21.150kb, 0, 700000, 25000, "chr21")
tab_df.22.150kb <- full.chrom.tabulator(pos_df.22, bins.22.150kb, 0, 700000, 25000, "chr22")
tab_df.X.150kb <- full.chrom.tabulator(pos_df.X, bins.X.150kb, 0, 700000, 25000, "chrX")
tab_df.Y.150kb <- full.chrom.tabulator(pos_df.Y, bins.Y.150kb, 0, 700000, 25000, "chrY")


tab_df.1.200kb <- full.chrom.tabulator(pos_df.1, bins.1.200kb, 0, 1000000, 50000, "chr1")
tab_df.2.200kb <- full.chrom.tabulator(pos_df.2, bins.2.200kb, 0, 1000000, 50000, "chr2")
tab_df.3.200kb <- full.chrom.tabulator(pos_df.3, bins.3.200kb, 0, 1000000, 50000, "chr3")
tab_df.4.200kb <- full.chrom.tabulator(pos_df.4, bins.4.200kb, 0, 1000000, 50000, "chr4")
tab_df.5.200kb <- full.chrom.tabulator(pos_df.5, bins.5.200kb, 0, 1000000, 50000, "chr5")
tab_df.6.200kb <- full.chrom.tabulator(pos_df.6, bins.6.200kb, 0, 1000000, 50000, "chr6")
tab_df.7.200kb <- full.chrom.tabulator(pos_df.7, bins.7.200kb, 0, 1000000, 50000, "chr7")
tab_df.8.200kb <- full.chrom.tabulator(pos_df.8, bins.8.200kb, 0, 1000000, 50000, "chr8")
tab_df.9.200kb <- full.chrom.tabulator(pos_df.9, bins.9.200kb, 0, 1000000, 50000, "chr9")
tab_df.10.200kb <- full.chrom.tabulator(pos_df.10, bins.10.200kb, 0, 1000000, 50000, "chr10")
tab_df.11.200kb <- full.chrom.tabulator(pos_df.11, bins.11.200kb, 0, 1000000, 50000, "chr11")
tab_df.12.200kb <- full.chrom.tabulator(pos_df.12, bins.12.200kb, 0, 1000000, 50000, "chr12")
tab_df.13.200kb <- full.chrom.tabulator(pos_df.13, bins.13.200kb, 0, 1000000, 50000, "chr13")
tab_df.14.200kb <- full.chrom.tabulator(pos_df.14, bins.14.200kb, 0, 1000000, 50000, "chr14")
tab_df.15.200kb <- full.chrom.tabulator(pos_df.15, bins.15.200kb, 0, 1000000, 50000, "chr15")
tab_df.16.200kb <- full.chrom.tabulator(pos_df.16, bins.16.200kb, 0, 1000000, 50000, "chr16")
tab_df.17.200kb <- full.chrom.tabulator(pos_df.17, bins.17.200kb, 0, 1000000, 50000, "chr17")
tab_df.18.200kb <- full.chrom.tabulator(pos_df.18, bins.18.200kb, 0, 1000000, 50000, "chr18")
tab_df.19.200kb <- full.chrom.tabulator(pos_df.19, bins.19.200kb, 0, 1000000, 50000, "chr19")
tab_df.20.200kb <- full.chrom.tabulator(pos_df.20, bins.20.200kb, 0, 1000000, 50000, "chr20")
tab_df.21.200kb <- full.chrom.tabulator(pos_df.21, bins.21.200kb, 0, 1000000, 50000, "chr21")
tab_df.22.200kb <- full.chrom.tabulator(pos_df.22, bins.22.200kb, 0, 1000000, 50000, "chr22")
tab_df.X.200kb <- full.chrom.tabulator(pos_df.X, bins.X.200kb, 0, 1000000, 50000, "chrX")
tab_df.Y.200kb <- full.chrom.tabulator(pos_df.Y, bins.Y.200kb, 0, 1000000, 50000, "chrY")

#Get the sig.thresh for .10kb for all the chromosomes.
sig.thresh.1.10kb <- chrom.analyzer("chr1", tab_df.1.10kb, my.ps)
sig.thresh.2.10kb <- chrom.analyzer("chr2", tab_df.2.10kb, my.ps)
sig.thresh.3.10kb <- chrom.analyzer("chr3", tab_df.3.10kb, my.ps)
sig.thresh.4.10kb <- chrom.analyzer("chr4", tab_df.4.10kb, my.ps)
sig.thresh.5.10kb <- chrom.analyzer("chr5", tab_df.5.10kb, my.ps)
sig.thresh.6.10kb <- chrom.analyzer("chr6", tab_df.6.10kb, my.ps)
sig.thresh.7.10kb <- chrom.analyzer("chr7", tab_df.7.10kb, my.ps)
sig.thresh.8.10kb <- chrom.analyzer("chr8", tab_df.8.10kb, my.ps)
sig.thresh.9.10kb <- chrom.analyzer("chr9", tab_df.9.10kb, my.ps)
sig.thresh.10.10kb <- chrom.analyzer("chr10", tab_df.10.10kb, my.ps)
sig.thresh.11.10kb <- chrom.analyzer("chr11", tab_df.11.10kb, my.ps)
sig.thresh.12.10kb <- chrom.analyzer("chr12", tab_df.12.10kb, my.ps)
sig.thresh.13.10kb <- chrom.analyzer("chr13", tab_df.13.10kb, my.ps)
sig.thresh.14.10kb <- chrom.analyzer("chr14", tab_df.14.10kb, my.ps)
sig.thresh.15.10kb <- chrom.analyzer("chr15", tab_df.15.10kb, my.ps)
sig.thresh.16.10kb <- chrom.analyzer("chr16", tab_df.16.10kb, my.ps)
sig.thresh.17.10kb <- chrom.analyzer("chr17", tab_df.17.10kb, my.ps)
sig.thresh.18.10kb <- chrom.analyzer("chr18", tab_df.18.10kb, my.ps)
sig.thresh.19.10kb <- chrom.analyzer("chr19", tab_df.19.10kb, my.ps)
sig.thresh.20.10kb <- chrom.analyzer("chr20", tab_df.20.10kb, my.ps)
sig.thresh.21.10kb <- chrom.analyzer("chr21", tab_df.21.10kb, my.ps)
sig.thresh.22.10kb <- chrom.analyzer("chr22", tab_df.22.10kb, my.ps)
sig.thresh.X.10kb <- chrom.analyzer("chrX", tab_df.X.10kb, my.ps)
sig.thresh.Y.10kb <- chrom.analyzer("chrY", tab_df.Y.10kb, my.ps)

#Get the sig.thresh for .20kb for all the chromosomes.
sig.thresh.1.20kb <- chrom.analyzer("chr1", tab_df.1.20kb, my.ps)
sig.thresh.2.20kb <- chrom.analyzer("chr2", tab_df.2.20kb, my.ps)
sig.thresh.3.20kb <- chrom.analyzer("chr3", tab_df.3.20kb, my.ps)
sig.thresh.4.20kb <- chrom.analyzer("chr4", tab_df.4.20kb, my.ps)
sig.thresh.5.20kb <- chrom.analyzer("chr5", tab_df.5.20kb, my.ps)
sig.thresh.6.20kb <- chrom.analyzer("chr6", tab_df.6.20kb, my.ps)
sig.thresh.7.20kb <- chrom.analyzer("chr7", tab_df.7.20kb, my.ps)
sig.thresh.8.20kb <- chrom.analyzer("chr8", tab_df.8.20kb, my.ps)
sig.thresh.9.20kb <- chrom.analyzer("chr9", tab_df.9.20kb, my.ps)
sig.thresh.10.20kb <- chrom.analyzer("chr10", tab_df.10.20kb, my.ps)
sig.thresh.11.20kb <- chrom.analyzer("chr11", tab_df.11.20kb, my.ps)
sig.thresh.12.20kb <- chrom.analyzer("chr12", tab_df.12.20kb, my.ps)
sig.thresh.13.20kb <- chrom.analyzer("chr13", tab_df.13.20kb, my.ps)
sig.thresh.14.20kb <- chrom.analyzer("chr14", tab_df.14.20kb, my.ps)
sig.thresh.15.20kb <- chrom.analyzer("chr15", tab_df.15.20kb, my.ps)
sig.thresh.16.20kb <- chrom.analyzer("chr16", tab_df.16.20kb, my.ps)
sig.thresh.17.20kb <- chrom.analyzer("chr17", tab_df.17.20kb, my.ps)
sig.thresh.18.20kb <- chrom.analyzer("chr18", tab_df.18.20kb, my.ps)
sig.thresh.19.20kb <- chrom.analyzer("chr19", tab_df.19.20kb, my.ps)
sig.thresh.20.20kb <- chrom.analyzer("chr20", tab_df.20.20kb, my.ps)
sig.thresh.21.20kb <- chrom.analyzer("chr21", tab_df.21.20kb, my.ps)
sig.thresh.22.20kb <- chrom.analyzer("chr22", tab_df.22.20kb, my.ps)
sig.thresh.X.20kb <- chrom.analyzer("chrX", tab_df.X.20kb, my.ps)
sig.thresh.Y.20kb <- chrom.analyzer("chrY", tab_df.Y.20kb, my.ps)

#Get the sig.thresh for .30kb for all the chromosomes.
sig.thresh.1.30kb <- chrom.analyzer("chr1", tab_df.1.30kb, my.ps)
sig.thresh.2.30kb <- chrom.analyzer("chr2", tab_df.2.30kb, my.ps)
sig.thresh.3.30kb <- chrom.analyzer("chr3", tab_df.3.30kb, my.ps)
sig.thresh.4.30kb <- chrom.analyzer("chr4", tab_df.4.30kb, my.ps)
sig.thresh.5.30kb <- chrom.analyzer("chr5", tab_df.5.30kb, my.ps)
sig.thresh.6.30kb <- chrom.analyzer("chr6", tab_df.6.30kb, my.ps)
sig.thresh.7.30kb <- chrom.analyzer("chr7", tab_df.7.30kb, my.ps)
sig.thresh.8.30kb <- chrom.analyzer("chr8", tab_df.8.30kb, my.ps)
sig.thresh.9.30kb <- chrom.analyzer("chr9", tab_df.9.30kb, my.ps)
sig.thresh.10.30kb <- chrom.analyzer("chr10", tab_df.10.30kb, my.ps)
sig.thresh.11.30kb <- chrom.analyzer("chr11", tab_df.11.30kb, my.ps)
sig.thresh.12.30kb <- chrom.analyzer("chr12", tab_df.12.30kb, my.ps)
sig.thresh.13.30kb <- chrom.analyzer("chr13", tab_df.13.30kb, my.ps)
sig.thresh.14.30kb <- chrom.analyzer("chr14", tab_df.14.30kb, my.ps)
sig.thresh.15.30kb <- chrom.analyzer("chr15", tab_df.15.30kb, my.ps)
sig.thresh.16.30kb <- chrom.analyzer("chr16", tab_df.16.30kb, my.ps)
sig.thresh.17.30kb <- chrom.analyzer("chr17", tab_df.17.30kb, my.ps)
sig.thresh.18.30kb <- chrom.analyzer("chr18", tab_df.18.30kb, my.ps)
sig.thresh.19.30kb <- chrom.analyzer("chr19", tab_df.19.30kb, my.ps)
sig.thresh.20.30kb <- chrom.analyzer("chr20", tab_df.20.30kb, my.ps)
sig.thresh.21.30kb <- chrom.analyzer("chr21", tab_df.21.30kb, my.ps)
sig.thresh.22.30kb <- chrom.analyzer("chr22", tab_df.22.30kb, my.ps)
sig.thresh.X.30kb <- chrom.analyzer("chrX", tab_df.X.30kb, my.ps)
sig.thresh.Y.30kb <- chrom.analyzer("chrY", tab_df.Y.30kb, my.ps)

#Get the sig.thresh for .40kb for all the chromosomes.
sig.thresh.1.40kb <- chrom.analyzer("chr1", tab_df.1.40kb, my.ps)
sig.thresh.2.40kb <- chrom.analyzer("chr2", tab_df.2.40kb, my.ps)
sig.thresh.3.40kb <- chrom.analyzer("chr3", tab_df.3.40kb, my.ps)
sig.thresh.4.40kb <- chrom.analyzer("chr4", tab_df.4.40kb, my.ps)
sig.thresh.5.40kb <- chrom.analyzer("chr5", tab_df.5.40kb, my.ps)
sig.thresh.6.40kb <- chrom.analyzer("chr6", tab_df.6.40kb, my.ps)
sig.thresh.7.40kb <- chrom.analyzer("chr7", tab_df.7.40kb, my.ps)
sig.thresh.8.40kb <- chrom.analyzer("chr8", tab_df.8.40kb, my.ps)
sig.thresh.9.40kb <- chrom.analyzer("chr9", tab_df.9.40kb, my.ps)
sig.thresh.10.40kb <- chrom.analyzer("chr10", tab_df.10.40kb, my.ps)
sig.thresh.11.40kb <- chrom.analyzer("chr11", tab_df.11.40kb, my.ps)
sig.thresh.12.40kb <- chrom.analyzer("chr12", tab_df.12.40kb, my.ps)
sig.thresh.13.40kb <- chrom.analyzer("chr13", tab_df.13.40kb, my.ps)
sig.thresh.14.40kb <- chrom.analyzer("chr14", tab_df.14.40kb, my.ps)
sig.thresh.15.40kb <- chrom.analyzer("chr15", tab_df.15.40kb, my.ps)
sig.thresh.16.40kb <- chrom.analyzer("chr16", tab_df.16.40kb, my.ps)
sig.thresh.17.40kb <- chrom.analyzer("chr17", tab_df.17.40kb, my.ps)
sig.thresh.18.40kb <- chrom.analyzer("chr18", tab_df.18.40kb, my.ps)
sig.thresh.19.40kb <- chrom.analyzer("chr19", tab_df.19.40kb, my.ps)
sig.thresh.20.40kb <- chrom.analyzer("chr20", tab_df.20.40kb, my.ps)
sig.thresh.21.40kb <- chrom.analyzer("chr21", tab_df.21.40kb, my.ps)
sig.thresh.22.40kb <- chrom.analyzer("chr22", tab_df.22.40kb, my.ps)
sig.thresh.X.40kb <- chrom.analyzer("chrX", tab_df.X.40kb, my.ps)
sig.thresh.Y.40kb <- chrom.analyzer("chrY", tab_df.Y.40kb, my.ps)
```

```{r Garbage/backup}
do.it.all <- function(locus.bin.size, start=0, end=400000, by=10000, variable.list=c("chr1", "chr2", "chr3", "chr4", "chr5", "chr6", "chr7", "chr8", "chr9", "chr10", "chr11", "chr12", "chr13", "chr14", "chr15", "chr16", "chr17", "chr18", "chr19", "chr20", "chr21", "chr22", "chrX", "chrY")){
  setwd("/Users/ittaieres/Desktop/Hi-C/All_good_analysis/cis_only/cis_only_coverage") #Set to correct WD for grabbing coverage files
	coverage <- fread(paste("coverage_", locus.bin.size, "kb_5kb.bed", sep=""), sep="\t", header=FALSE, stringsAsFactors=FALSE) #Pull out coverage for the given size
	colnames(coverage) <- c("chr", "start", "end", "count") #Rename the columns of the coverage df to work with other functions
	
	setwd("/Users/ittaieres/Desktop/Hi-C/All_good_analysis/cis_only/paired_read_midpoint_positions") #Set to correct WD for grabbing read pair positions in loop below
	
	#Begin iterating through all chromosomes!
	for(chromosome in variable.list){
	
		#Create all the coverage dfs for all the chromosomes at this bin size. Form: coverage.Xkb.chrX
		assign(paste("coverage.", locus.bin.size, "kb.", chromosome, sep=""), coverage[coverage$chr==chromosome,])
		
		#Create all the read pair position dfs for all chromosomes. Form: pos_df.chrX
		assign(paste("pos_df.", chromosome, sep=""), fread(paste("midpoint.pairs.", substr(chromosome, 4, 5), sep=""), header=FALSE, stringsAsFactors=FALSE))
		
		#Get all the tabulation dfs for all chromosomes at this bin size. Form: tab_df.Xkb.chrX
		assign(paste("tab_df.", locus.bin.size, "kb.", chromsome, sep=""), full.chrom.tabulator(get(paste("coverage.", locus.bin.size, "kb.", chromosome, sep=""))), get(paste("pos_df.", chromosome, sep="")), start, end, by, chromosome)
	}
}


#Get the sig.thresh for .10kb for all the chromosomes.
sig.thresh.1.10kb <- sig.thresh.finder("chr1", tab_df.1.10kb, my.ps)
sig.thresh.2.10kb <- sig.thresh.finder("chr2", tab_df.2.10kb, my.ps)
sig.thresh.3.10kb <- sig.thresh.finder("chr3", tab_df.3.10kb, my.ps)
sig.thresh.4.10kb <- sig.thresh.finder("chr4", tab_df.4.10kb, my.ps)
sig.thresh.5.10kb <- sig.thresh.finder("chr5", tab_df.5.10kb, my.ps)
sig.thresh.6.10kb <- sig.thresh.finder("chr6", tab_df.6.10kb, my.ps)
sig.thresh.7.10kb <- sig.thresh.finder("chr7", tab_df.7.10kb, my.ps)
sig.thresh.8.10kb <- sig.thresh.finder("chr8", tab_df.8.10kb, my.ps)
sig.thresh.9.10kb <- sig.thresh.finder("chr9", tab_df.9.10kb, my.ps)
sig.thresh.10.10kb <- sig.thresh.finder("chr10", tab_df.10.10kb, my.ps)
sig.thresh.11.10kb <- sig.thresh.finder("chr11", tab_df.11.10kb, my.ps)
sig.thresh.12.10kb <- sig.thresh.finder("chr12", tab_df.12.10kb, my.ps)
sig.thresh.13.10kb <- sig.thresh.finder("chr13", tab_df.13.10kb, my.ps)
sig.thresh.14.10kb <- sig.thresh.finder("chr14", tab_df.14.10kb, my.ps)
sig.thresh.15.10kb <- sig.thresh.finder("chr15", tab_df.15.10kb, my.ps)
sig.thresh.16.10kb <- sig.thresh.finder("chr16", tab_df.16.10kb, my.ps)
sig.thresh.17.10kb <- sig.thresh.finder("chr17", tab_df.17.10kb, my.ps)
sig.thresh.18.10kb <- sig.thresh.finder("chr18", tab_df.18.10kb, my.ps)
sig.thresh.19.10kb <- sig.thresh.finder("chr19", tab_df.19.10kb, my.ps)
sig.thresh.20.10kb <- sig.thresh.finder("chr20", tab_df.20.10kb, my.ps)
sig.thresh.21.10kb <- sig.thresh.finder("chr21", tab_df.21.10kb, my.ps)
sig.thresh.22.10kb <- sig.thresh.finder("chr22", tab_df.22.10kb, my.ps)
sig.thresh.X.10kb <- sig.thresh.finder("chrX", tab_df.X.10kb, my.ps)
sig.thresh.Y.10kb <- sig.thresh.finder("chrY", tab_df.Y.10kb, my.ps)

chrom.analyzer <- function(chromosome, df, percents){
  
  #chr.df <- filter(df, chr==chromosome) #Filter out to get the chromosome of interest
  #ggplot(chr.df, aes(x=bin, y=count)) + geom_boxplot(aes(group=bin)) #For actually visualizing the boxplots
  
  bins <- unique(df$bin) #Grab the bins
  percents <- percents*.01 #Re-scale the percents to work with quantile in the loop below.
  
  values.df <- data.frame(matrix(nrow=length(bins), ncol=length(percents))) #Initialize the df to be returned
  colnames(values.df) <- percents #Initialize the df's colnames for iterating through
  rownames(values.df) <- bins #Initialize the df's rownames for iterating through
  
  for(bin in bins){
    for(percent in percents){
      values.df[as.character(bin), as.character(percent)] <- quantile(df$count[df$bin==bin], 1-percent)[[1]]#To pull out the value above which are the top percent % of loci
    }
  }
  return(values.df)
}

head(homer.10kb.chr22)
colnames(homer.10kb.chr22) <- c("start1", "end1", "start2", "end2")
start1_new <- ifelse(homer.10kb.chr22$start1<homer.10kb.chr22$start2, homer.10kb.chr22$start1, homer.10kb.chr22$start2) #reorders columns so pos1 < pos2. important for not repeating bin pairs
start2_new <- ifelse(homer.10kb.chr22$start1<homer.10kb.chr22$start2, homer.10kb.chr22$start2, homer.10kb.chr22$start1) #If true, returns second arg; if false, returns third.
end1_new <-ifelse(homer.10kb.chr22$start1<homer.10kb.chr22$start2, homer.10kb.chr22$end1, homer.10kb.chr22$end2)
end2_new <- ifelse(homer.10kb.chr22$start1<homer.10kb.chr22$start2, homer.10kb.chr22$end2, homer.10kb.chr22$end1)
  
#Assign the new columns.
homer.10kb.chr22$start1 <- start1_new
homer.10kb.chr22$start2 <- start2_new
homer.10kb.chr22$end1 <- end1_new
homer.10kb.chr22$end2 <- end2_new


homer.10kb.chr22$loc1 <- rowMeans(homer.10kb.chr22[,1:2])
homer.10kb.chr22$loc2 <- rowMeans(homer.10kb.chr22[,3:4])
homer.10kb.chr22$distance <- abs(homer.10kb.chr22$loc1-homer.10kb.chr22$loc2)
homer.10kb.chr22 <- homer.10kb.chr22 %>% mutate(loc1_bin=cut2(loc1, bins.vec.22, onlycuts=TRUE, oneval=FALSE, digits=9)) %>% mutate(loc2_bin=cut2(loc2, bins.vec.22, onlycuts=TRUE, oneval=FALSE, digits=9))


homer.10kb.chr22 <- homer.10kb.chr22[,-1:-6] #Finally have what we're looking for

#Now, subset down to just the relevant dataframes (with pairs at the distance desired) to check for overlap.
homer.10kb.sigs <- homer.10kb.chr22[homer.10kb.chr22$distance<20000&homer.10kb.chr22$distance>=10000,2:3]

data.22 <- tab_df.10kb.full[tab_df.10kb.full$chr=="chr22",]
my.10kb.sigs <- data.22[data.22$bin==10000&data.22$count>=sig.22["10000", "0.05"], 3:4]
colnames(my.10kb.sigs) <- c("loc1_bin", "loc2_bin")

nrow(merge(homer.10kb.sigs, my.10kb.sigs)) #YES! Works well :)

homer.10kb.chr22.bins1 <- as.factor(paste("[", homer.10kb.chr22$'start(1)', ",", homer.10kb.chr22$'end(1)', ")", sep=""))

#Function that takes a file of significant hits in homer, a file of coverage from data, and a positions file from data, finding the overlap of significant hits between homer and my method.
signif.comparison <- function(coverage.file, homer.file, chr, start=0, end=50000, by=5000, percents=c(5, 2.5, 1, 0.5, 0.1)){
  
  #Read in the coverage.file, rename its columns, and subset down to the specific chromosome of interest!
  setwd("/Users/ittaieres/Desktop/Hi-C/All_good_analysis/cis_only/cis_only_coverage")
  full_coverage <- fread("coverage_10kb_5kb.bed", sep="\t", header=F, stringsAsFactors = FALSE)
  colnames(full_coverage) <- c("chr", "start", "end", "count")
  coverage.df <- full_coverage[full_coverage$chr==chr,]
  
  #Find the optimum bins for this chromosome at this "locus bin" size.
  chr.bins <- optimize.bin.sums(coverage.df)
  chr.bins <- unique(sort(c(bins$start, bins$end)))

  #Read in the pos.file 
  setwd("/Users/ittaieres/Desktop/Hi-C/All_good_analysis/cis_only/paired_read_midpoint_positions")
  pos.df <- fread(paste("midpoint.pairs.", substr(chr, 4, nchar(chr)), sep=""), header=FALSE, stringsAsFactors = FALSE)
  
  #Get long-form dataframe of all the data I could possibly want. Dictated by distance bins defined by start, end, and by.
  full_info <- full.chrom.tabulator(coverage.df, pos.df, start, end, by)
  
  #Get a data frame of the values above which we would consider hits "significant" in the different distance bins.
  sig.values <- sig.thresh.finder(full_info, percents)
  
  #Read in homer file, change column names to work with next lines
  homer.sig <- fread(homer.file, header=TRUE, stringsAsFactors = FALSE)
  colnames(homer.sig) <- c("start1", "end1", "start2", "end2")
  
  #Reformat the homer file to be easier to work with for matching hereafter.
  start1_new <- ifelse(homer.sig$start1<homer.sig$start2, homer.sig$start1, homer.sig$start2) #reorders columns so pos1 < pos2. important for not repeating bin pairs
  start2_new <- ifelse(homer.sig$start1<homer.sig$start2, homer.sig$start2, homer.sig$start1) #If true, returns second arg; if false, returns third.
  end1_new <-ifelse(homer.sig$start1<homer.sig$start2, homer.sig$end1, homer.sig$end2)
  end2_new <- ifelse(homer.sig$start1<homer.sig$start2, homer.sig$end2, homer.sig$end1)
  
  #Assign the new columns.
  homer.sig$start1 <- start1_new
  homer.sig$start2 <- start2_new
  homer.sig$end1 <- end1_new
  homer.sig$end2 <- end2_new
  
  #Get individual locations as midpoints of loci identified as significant by homer, so I can assign them to my set of bins for the given chr.
  homer.sig$loc1 <- rowMeans(homer.sig[,1:2]) #Get midpoint of start1 and end1, for loc1
  homer.sig$loc2 <- rowMeans(homer.sig[,3:4]) #Same for loc2 (start2 and end2)
  
  #Get another column that is distance between the two loci, for subsetting down later on.
  homer.sig$distance <- abs(homer.sig$loc1-homer.sig$loc2)
  
  #Assign the individual locations of homer-significant loci to my set of bins for the given chr.
  homer.sig <- homer.sig %>% mutate(loc1_bin=cut2(loc1, chr.bins, onlycuts=TRUE, oneval=FALSE, digits=9)) %>% mutate(loc2_bin=cut2(loc2, chr.bins, onlycuts=TRUE, oneval=FALSE, digits=9))
  
  #Subset down to just the pairs of significant hits bins from the homer data.
  homer.sig <- homer.sig[,-1:-6] #Gets rid of columns 1-6, column 7 with distance between loci and columns 8 and 9 with position bin assignations are all that matter.
  
  
  
  #Make a properly formatted DF of my significant hits.
  tab_df <- tab_df[tab_df$count>=sig_thresh&tab_df$bin==bin,] #Pull out only values above the significance threshold in the right bin.
  characters <- (unique(nchar(as.character(tab_df$loc1)))-3)/2
  start1 <- as.numeric(substr(as.character(tab_df$loc1), 2, characters+1))
  end1 <- as.numeric(substr(as.character(tab_df$loc1), characters+3, characters*2+2))
  start2 <- as.numeric(substr(as.character(tab_df$loc2), 2, characters+1))
  end2 <- as.numeric(substr(as.character(tab_df$loc2), characters+3, characters*2+2))
  my.signifs <- data.frame(start1=start1, end1=end1, start2=start2, end2=end2)
  
  #Find the matching indices, i.e. the overlap of significant hits!
  matching.indices <- which(my.signifs$start1 %in% homer.sig$start.1. & my.signifs$start2 %in% homer.sig$start.2.)
  print(matching.indices)
  return(homer.sig)
}

setwd("/Users/ittaieres/Desktop/Hi-C/Hi-C Basic Analyze/homer.signif/sig_10kb")
test <- signif.comparison("chr1_10kb_siglocs.txt", tab_df.1.10kb, 30000, 77)

##Was a part of sig.thresh.finder, but is no longer needed:
  #Remove rows that are duplicate bins with each other!
  # characters <- (unique(nchar(as.character(full_tab_df$loc1)))-3)/2
  # start1 <- as.numeric(substr(as.character(full_tab_df$loc1), 2, characters+1))
  # start2 <- as.numeric(substr(as.character(full_tab_df$loc2), 2, characters+1))
  # full_tab_df <- full_tab_df[-which(start1==start2),]

#Here's a set of tests for checking that optimize.bin.sums and the full chrom tabulator both come back with the same (or very similar) results. Sometimes the full chrom tabulator might have a few bins less than the optimize.bin.sums, because, while both work on the coverage df produced by bedtools, optimize bin sums does so without regard for the position DF, whereas the full chrom tabulator assigns the pos df positions to individual bins, so some things that appear as bins from optimize.bin.sums based on the coverage DF are actually irrelevant in making locus bins for tabulation because they represent reads that span multiple bins (since the coverage includes a 100-bp read in each of the bins it may overlap)
coverage.22 <- full_coverage[full_coverage$chr=="chr22",] #Get coverage df for chr22.
setwd("/Users/ittaieres/Desktop/Hi-C/All_good_analysis/cis_only/paired_read_midpoint_positions")
pos.22 <- fread("midpoint.pairs.22", header=FALSE, stringsAsFactors = FALSE) #Get pos df for chr22.
#Good to here!

tab.22 <- basic_tabulation(coverage.22, pos.22, 0, 35000000) #Tabulate all contact counts for chr22, just to check! But the input to basic_tabulation needs to change

data.22 <- full.chrom.tabulator(coverage.22, pos.22, 0, 341500000, 1000000, "chr22") #Get EVERYTHING! Do this all the way up to the max value with increment distance bin sizes of 1 million just to make computationally efficient. Then check against optimize bin sums result to see all bins are the same (or they should be!)

#Grab bins from this data table!
data.22.bins <- unique(sort(c(as.numeric(substr(data.22$loc1, 2, 9)), as.numeric(substr(data.22$loc1, 11, 18)), as.numeric(substr(data.22$loc2, 2, 9)), as.numeric(substr(data.22$loc2, 11, 18)))))

#Grab bins more manually with optimize bin sums.
obs.bins.22 <- optimize.bin.sums(coverage.22)
bins.vec.22 <- unique(sort(c(obs.bins.22$start, obs.bins.22$end)))

#Compare the two.
bins.vec.22 %in% data.22.bins

#It is OK for the bins from optimize bin sums and the bins from cutting out parts of a tabulation DF are not all identical--some of the bins created from optimize bin sums may be for contacts that are between loci which are at a distance I don't examine in the creation of the full DF (this is another issue--that really, the DF should encapsulate everything. I need to find a way to set the max distance it'll check to be the max distance for pairs on that chromosome...and to skip over the many sparse distance bins that should exist in between! Add in something to full.chrom.tabulator that finds max distance between pairs and then sets that as the "to" of the seq for pair distance bins--that way, only have to specify "by" in call to function (from should still be 0 always). On the flip side this also creates the issue of having an absurd number of bins and this taking too long computationally!)
sig.22 <- sig.thresh.finder(data.22, percents=c(20, 15, 10, 5))

bins.22 <- optimize.bin.sums(coverage.22)
bins.vec.22 <- unique(sort(c(bins.22$start, bins.22$end)))

# To check that the bins in the tabulation all make sense--they do!
unique(substr(levels(tab.22$pos1_bin), 15, 18))
```


```{r See what's going on w/ ChrY in analysis w/ homer signif comparison}
#Read in all the chrY data!
setwd("/Users/ittaieres/Desktop/Hi-C/All_good_analysis/cis_only/cis_only_coverage")
full_coverage <- fread("coverage_10kb_5kb.bed", sep="\t", header=F, stringsAsFactors = FALSE)
colnames(full_coverage) <- c("chr", "start", "end", "count")
coverage.Y <- full_coverage[full_coverage$chr=="chrY",] #Get coverage df for chr22.
setwd("/Users/ittaieres/Desktop/Hi-C/All_good_analysis/cis_only/paired_read_midpoint_positions")
pos.Y <- fread("midpoint.pairs.Y", header=FALSE, stringsAsFactors = FALSE) #Get pos df for chr22.
contacts.Y <- full.chrom.tabulator(coverage.Y, pos.Y, 0, 100000, 5000, "chr22") #Get full tab df for chr22.
sig.Y <- sig.thresh.finder(contacts.Y, c(20, 15, 10, 5)) #Get significance threshold values for this df.

#Read in homer significance data for chrY!
setwd("/Users/ittaieres/Desktop/Hi-C/All_good_analysis/homer.signif/sig_10kb")
homer.10kb.chrY <- fread("chrY_10kb_siglocs.txt", header=TRUE, stringsAsFactors = FALSE)

####Reformat homer to be sure it works.
colnames(homer.10kb.chrY) <- c("start1", "end1", "start2", "end2")
start1_new <- ifelse(homer.10kb.chrY$start1<homer.10kb.chrY$start2, homer.10kb.chrY$start1, homer.10kb.chrY$start2) #reorders columns so pos1 < pos2. important for not repeating bin pairs
start2_new <- ifelse(homer.10kb.chrY$start1<homer.10kb.chrY$start2, homer.10kb.chrY$start2, homer.10kb.chrY$start1) #If true, returns second arg; if false, returns third.
end1_new <-ifelse(homer.10kb.chrY$start1<homer.10kb.chrY$start2, homer.10kb.chrY$end1, homer.10kb.chrY$end2)
end2_new <- ifelse(homer.10kb.chrY$start1<homer.10kb.chrY$start2, homer.10kb.chrY$end2, homer.10kb.chrY$end1)
homer.10kb.chrY$start1 <- start1_new
homer.10kb.chrY$start2 <- start2_new
homer.10kb.chrY$end1 <- end1_new
homer.10kb.chrY$end2 <- end2_new
homer.10kb.chrY$loc1 <- rowMeans(homer.10kb.chrY[,1:2]) #Get midpoint of start1 and end1, for loc1
homer.10kb.chrY$loc2 <- rowMeans(homer.10kb.chrY[,3:4]) #Same for loc2 (start2 and end2)
homer.10kb.chrY$distance <- abs(homer.10kb.chrY$loc1-homer.10kb.chrY$loc2)
  
##Now, obtain bins from chr_df and use those to assign my bins to the homer-sig hits:
characters <- (unique(nchar(as.character(contacts.Y$loc1_bin)))-3)/2
chr.bins <- unique(sort(c(as.numeric(substr(contacts.Y$loc1_bin, 2, characters+1)), as.numeric(substr(contacts.Y$loc1_bin, characters+3, 2*characters+2)), as.numeric(substr(contacts.Y$loc2_bin, 2, characters+1)), as.numeric(substr(contacts.Y$loc2_bin, characters+3, 2*characters+2)))))

homer.sig <- homer.10kb.chrY %>% mutate(loc1_bin=cut2(loc1, chr.bins, onlycuts=TRUE, oneval=FALSE, digits=9)) %>% mutate(loc2_bin=cut2(loc2, chr.bins, onlycuts=TRUE, oneval=FALSE, digits=9))

#Subset down to just the distance and pair bins from the homer.sig data
homer.sig <- homer.sig[,7:9]

#Remove duplicates, if any
homer.sig <- homer.sig[homer.sig$loc1_bin!=homer.sig$loc2_bin,] 

#Obtain distance b/t bins
slide <- as.numeric(unique(contacts.22$dist.bin)[2]-unique(contacts.22$dist.bin)[1])

#Testing!
my.sigs <- contacts.22[(contacts.22$dist.bin=="10000")&contacts.22$count>=(sig.22["10000", "0.2"]),] #Get my significant contacts
tmp.homer.sig <- homer.sig[(homer.sig$distance)>=10000&homer.sig$distance<(10000+slide), 2:3] #Pull out homer's significant contacts at this range

tmp.overlap <- merge(my.sigs[,3:4], tmp.homer.sig)
nrow(merge(my.sigs[,3:4], tmp.homer.sig))

indices <- my.sigs[(my.sigs$loc1_bin %in% tmp.overlap$loc1_bin)&(my.sigs$loc2_bin %in% tmp.overlap$loc2_bin),] #Use this, appears to work 
indices.test <- my.sigs[(my.sigs$loc1_bin %in% tmp.homer.sig$loc1_bin) & (my.sigs$loc2_bin %in% tmp.homer.sig$loc2_bin),] #This could (and does) accidentally pull out some rows that are not truly matching, but that simply had loc1 match in 1 row and loc2 in another (not the same)

test <- variable.assign(coverage.extractor, "coverage_10kb", c("this is a test", "uh-oh spagghetios"))

##Test it all
coverage <- fread("/project/gilad/ittai/roach/cis_only_cov_pos/coverage_cis_only/coverage_10kb_5kb.bed", sep="\t", header=FALSE, stringsAsFactors=FALSE, data.table=FALSE)
colnames(coverage) <- c("chr", "start", "end", "count")
coverage.Y <- coverage[coverage$chr=="chrY",]
pos.Y <- fread("/project/gilad/ittai/roach/cis_only_cov_pos/paired_read_midpoint_positions/midpoint.pairs.Y", header=FALSE, stringsAsFactors=FALSE, data.table=FALSE)

contacts.Y <- full.chrom.tabulator(coverage.Y, pos.Y, 0, 100000, 10000, "chrY")
print(head(contacts.Y))
print(nrow(contacts.Y))

basic.tab <- basic_tabulation(coverage.Y, pos.Y, 0, 500000)
print(head(basic.tab))
print(nrow(basic.tab))
```


